<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="This R package analyzes high-throughput sequencing of T
        and B cell receptor complementarity determining region 3 (CDR3)
        sequences generated by Adaptive Biotechnologies' ImmunoSEQ
        assay.  Its input comes from tab-separated value (.tsv) files
        exported from the ImmunoSEQ analyzer.">
<title>Analyze high-throughput sequencing of T and B cell receptors • LymphoSeq2</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Analyze high-throughput sequencing of T and B cell receptors">
<meta property="og:description" content="This R package analyzes high-throughput sequencing of T
        and B cell receptor complementarity determining region 3 (CDR3)
        sequences generated by Adaptive Biotechnologies' ImmunoSEQ
        assay.  Its input comes from tab-separated value (.tsv) files
        exported from the ImmunoSEQ analyzer.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">LymphoSeq2</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="articles/LymphoSeq2.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/shashidhar22/LymphoSeq2/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="lymphoseq2">LymphoSeq2<a class="anchor" aria-label="anchor" href="#lymphoseq2"></a>
</h1></div>
<!-- badges: start -->

<p>Adaptive Immune Receptor Repertoire Sequencing (AIRR-seq) provides a unique opportunity to interrogate the adaptive immune repertoire under various clinical conditions. The utility offered by this technology has quickly garnered interest from a community of clinicians and researchers investigating the immunological landscapes of a large spectrum of health and disease states. LymphoSeq2 is a toolkit that allows users to import, manipulate and visualize AIRR-Seq data from various AIRR-Seq assays such as Adaptive ImmunoSEQ and BGI-IRSeq, with support for 10X VDJ sequencing coming soon. The platform also supports the importing of AIRR-seq data processed using the MiXCR pipeline. The vignette highlights some of the key features of LymphoSeq2.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>To install the latest version of LymphoSeq2 you can use the <code>devtools</code> package and install LymphoSeq2 from GitHub</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("devtools")</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"shashidhar22/LymphoSeq2"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="getting-started">Getting started<a class="anchor" aria-label="anchor" href="#getting-started"></a>
</h2>
<p>To import AIRR-Seq data using <code>LymphoSeq2</code> we can use the <code>readImmunoSeq</code> function. Currently the function can import data from MiXCR, Adaptive ImmunoSEQ, BGI IR-SEQ, and 10X Genomic single cell VDJ rearrangements.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/shashidhar22/LymphoSeq2" class="external-link">LymphoSeq2</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: data.table</span></span>
<span><span class="co">#&gt; Registered S3 method overwritten by 'ggtree':</span></span>
<span><span class="co">#&gt;   method      from </span></span>
<span><span class="co">#&gt;   identify.gg ggfun</span></span>
<span><span class="va">study_files</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"TCRB_sequencing"</span>, package <span class="op">=</span> <span class="st">"LymphoSeq2"</span><span class="op">)</span> </span>
<span><span class="va">study_table</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="reference/readImmunoSeq.html">readImmunoSeq</a></span><span class="op">(</span><span class="va">study_files</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: One or more parsing issues, call `problems()` on your data frame for details,</span></span>
<span><span class="co">#&gt; e.g.:</span></span>
<span><span class="co">#&gt;   dat &lt;- vroom(...)</span></span>
<span><span class="co">#&gt;   problems(dat)</span></span>
<span><span class="co">#&gt; One or more parsing issues, call `problems()` on your data frame for details,</span></span>
<span><span class="co">#&gt; e.g.:</span></span>
<span><span class="co">#&gt;   dat &lt;- vroom(...)</span></span>
<span><span class="co">#&gt;   problems(dat)</span></span>
<span><span class="co">#&gt; One or more parsing issues, call `problems()` on your data frame for details,</span></span>
<span><span class="co">#&gt; e.g.:</span></span>
<span><span class="co">#&gt;   dat &lt;- vroom(...)</span></span>
<span><span class="co">#&gt;   problems(dat)</span></span>
<span><span class="co">#&gt; One or more parsing issues, call `problems()` on your data frame for details,</span></span>
<span><span class="co">#&gt; e.g.:</span></span>
<span><span class="co">#&gt;   dat &lt;- vroom(...)</span></span>
<span><span class="co">#&gt;   problems(dat)</span></span>
<span><span class="co">#&gt; One or more parsing issues, call `problems()` on your data frame for details,</span></span>
<span><span class="co">#&gt; e.g.:</span></span>
<span><span class="co">#&gt;   dat &lt;- vroom(...)</span></span>
<span><span class="co">#&gt;   problems(dat)</span></span>
<span><span class="co">#&gt; One or more parsing issues, call `problems()` on your data frame for details,</span></span>
<span><span class="co">#&gt; e.g.:</span></span>
<span><span class="co">#&gt;   dat &lt;- vroom(...)</span></span>
<span><span class="co">#&gt;   problems(dat)</span></span></code></pre></div>
<p>To get a quick summary of repertoire characteristics, use the <code>clonality</code> function. This will calculate many standard repertoire diversity metrics such <code>clonality</code>, <code>gini coefficient</code>, <code>convergence</code>, and <code>unique productive sequence</code> for each of the repertoires in the input dataset.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">summary_table</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="reference/clonality.html">clonality</a></span><span class="op">(</span><span class="va">study_table</span><span class="op">)</span></span>
<span><span class="va">summary_table</span></span>
<span><span class="co">#&gt; # A tibble: 10 × 8</span></span>
<span><span class="co">#&gt;    repertoire_id     total_seq…¹ uniqu…² total…³ clona…⁴ gini_…⁵ top_p…⁶ conve…⁷</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;                   &lt;int&gt;   &lt;int&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;</span></span>
<span><span class="co">#&gt;  1 TRB_CD4_949              1000     845   25769   0.443   0.867   30.1     1   </span></span>
<span><span class="co">#&gt;  2 TRB_CD8_949              1000     794   26239   0.431   0.903   19.3     1.01</span></span>
<span><span class="co">#&gt;  3 TRB_CD8_CMV_369           414     281    1794   0.332   0.761   16.5     1.12</span></span>
<span><span class="co">#&gt;  4 TRB_Unsorted_0           1000     838   18161   0.281   0.818    5.77    1   </span></span>
<span><span class="co">#&gt;  5 TRB_Unsorted_1320        1000     838  178190   0.422   0.902   14.6     1   </span></span>
<span><span class="co">#&gt;  6 TRB_Unsorted_1496        1000     832   33669   0.389   0.881   14.2     1   </span></span>
<span><span class="co">#&gt;  7 TRB_Unsorted_32           920     767   31078   0.134   0.601    4.87    1.01</span></span>
<span><span class="co">#&gt;  8 TRB_Unsorted_369         1000     830  339413   0.426   0.845   29.7     1   </span></span>
<span><span class="co">#&gt;  9 TRB_Unsorted_83          1000     823  236732   0.338   0.777   23.6     1   </span></span>
<span><span class="co">#&gt; 10 TRB_Unsorted_949         1000     831    6549   0.306   0.765   13.8     1   </span></span>
<span><span class="co">#&gt; # … with abbreviated variable names ¹​total_sequences,</span></span>
<span><span class="co">#&gt; #   ²​unique_productive_sequences, ³​total_count, ⁴​clonality, ⁵​gini_coefficient,</span></span>
<span><span class="co">#&gt; #   ⁶​top_productive_sequence, ⁷​convergence</span></span></code></pre></div>
<p>To compare samples with varying depth of sequencing, you can use the <code>clonality</code> function to sample down all repertoires to a minimum number of sequences. Since we randomly sample sequences from each repertoire, in this mode the <code>clonality</code> function will repeat this operation for a user specified number of <code>iterations</code> and caculate the average value for all the diversity metrics.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sampled_summary</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="reference/clonality.html">clonality</a></span><span class="op">(</span><span class="va">study_table</span>, rarefy <span class="op">=</span> <span class="cn">TRUE</span>, iterations <span class="op">=</span> <span class="fl">5</span>, min_count <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span></span>
<span><span class="va">sampled_summary</span></span>
<span><span class="co">#&gt; # A tibble: 10 × 8</span></span>
<span><span class="co">#&gt;    repertoire_id     total_seq…¹ uniqu…² total…³ clona…⁴ gini_…⁵ top_p…⁶ conve…⁷</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;                   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;</span></span>
<span><span class="co">#&gt;  1 TRB_CD4_949              163.    137.    1000  0.310    0.723   30.8     1   </span></span>
<span><span class="co">#&gt;  2 TRB_CD8_949              192.    155.    1000  0.291    0.731   18.9     1.01</span></span>
<span><span class="co">#&gt;  3 TRB_CD8_CMV_369          272.    189.    1000  0.299    0.719   16.7     1.08</span></span>
<span><span class="co">#&gt;  4 TRB_Unsorted_0           254     209.    1000  0.160    0.621    6.18    1.00</span></span>
<span><span class="co">#&gt;  5 TRB_Unsorted_1320        183     151.    1000  0.274    0.729   14.2     1.01</span></span>
<span><span class="co">#&gt;  6 TRB_Unsorted_1496        211     175.    1000  0.257    0.705   14.0     1   </span></span>
<span><span class="co">#&gt;  7 TRB_Unsorted_32          415.    351.    1000  0.0872   0.455    4.42    1.01</span></span>
<span><span class="co">#&gt;  8 TRB_Unsorted_369         247.    204.    1000  0.345    0.714   31.6     1   </span></span>
<span><span class="co">#&gt;  9 TRB_Unsorted_83          314.    256.    1000  0.272    0.653   23.7     1   </span></span>
<span><span class="co">#&gt; 10 TRB_Unsorted_949         303.    249.    1000  0.230    0.635   13.5     1   </span></span>
<span><span class="co">#&gt; # … with abbreviated variable names ¹​total_sequences,</span></span>
<span><span class="co">#&gt; #   ²​unique_productive_sequences, ³​total_count, ⁴​clonality, ⁵​gini_coefficient,</span></span>
<span><span class="co">#&gt; #   ⁶​top_productive_sequence, ⁷​convergence</span></span></code></pre></div>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/shashidhar22/LymphoSeq2/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/shashidhar22/LymphoSeq2/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/Artistic-2.0" class="external-link">Artistic-2.0</a></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing LymphoSeq2</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Wu Elena <br><small class="roles"> Author </small>  </li>
<li>Ravishankar Shashidhar <br><small class="roles"> Maintainer, author </small>  </li>
<li>Coffey David <br><small class="roles"> Author </small>  </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/shashidhar22/LymphoSeq2/actions" class="external-link"><img src="https://github.com/shashidhar22/LymphoSeq2/workflows/R-CMD-check/badge.svg" alt="R-CMD-check"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Wu Elena, Ravishankar Shashidhar, Coffey David.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
