<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Perform multiple sequence alignment using one of three methods and output results to the
console or as a pdf file.  One may perform the alignment of all amino acid or junction
sequences in a single repertoire_id.  Alternatively, one may search for a given sequence
within a list of samples using an edit distance threshold."><title>Align multiple sequences — alignSeq • LymphoSeq2</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Align multiple sequences — alignSeq"><meta property="og:description" content="Perform multiple sequence alignment using one of three methods and output results to the
console or as a pdf file.  One may perform the alignment of all amino acid or junction
sequences in a single repertoire_id.  Alternatively, one may search for a given sequence
within a list of samples using an edit distance threshold."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">LymphoSeq2</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/LymphoSeq2.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/shashidhar22/LymphoSeq2/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Align multiple sequences</h1>
      <small class="dont-index">Source: <a href="https://github.com/shashidhar22/LymphoSeq2/blob/HEAD/R/alignSeq.R" class="external-link"><code>R/alignSeq.R</code></a></small>
      <div class="d-none name"><code>alignSeq.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Perform multiple sequence alignment using one of three methods and output results to the
console or as a pdf file.  One may perform the alignment of all amino acid or junction
sequences in a single repertoire_id.  Alternatively, one may search for a given sequence
within a list of samples using an edit distance threshold.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">alignSeq</span><span class="op">(</span></span>
<span>  <span class="va">study_table</span>,</span>
<span>  repertoire_ids <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  sequence_list <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  edit_distance <span class="op">=</span> <span class="fl">15</span>,</span>
<span>  type <span class="op">=</span> <span class="st">"junction"</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"ClustalOmega"</span>,</span>
<span>  top <span class="op">=</span> <span class="fl">150</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>study_table</dt>
<dd><p>A tibble consisting of antigen receptor sequences
imported by the LymphoSeq function readImmunoSeq.</p></dd>


<dt>sequence_list</dt>
<dd><p>A character vector of one ore more amino acid or junction
CDR3 sequences to search.</p></dd>


<dt>edit_distance</dt>
<dd><p>An integer giving the minimum edit distance that the
sequence must be less than or equal to.  See details below.</p></dd>


<dt>type</dt>
<dd><p>A character vector indicating whether "junction_aa" or "junction" sequences
should be aligned.  If "junction_aa" is specified, then run productiveSeqs first.</p></dd>


<dt>method</dt>
<dd><p>A character vector indicating the multiple sequence alignment method to
be used.  Refer to the Bioconductor msa package for more details.  Options include
"ClustalW", "ClustalOmega", and "Muscle".</p></dd>


<dt>top</dt>
<dd><p>The number of top sequences to perform alignment.</p></dd>


<dt>repertoire_id</dt>
<dd><p>A character vector indicating the name of the repertoire_id in the productive
sequence list.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>Performs a multiple sequence alignment object.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>Edit distance is a way of quantifying how dissimilar two sequences
are to one another by counting the minimum number of operations required to
transform one sequence into the other.  For example, an edit distance of 0
means the sequences are identical and an edit distance of 1 indicates that
the sequences different by a single amino acid or junction.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p>If having trouble saving pdf files, refer to Bioconductor package msa for
installation instructions
<a href="http://bioconductor.org/packages/release/bioc/vignettes/msa/inst/doc/msa.pdf" class="external-link">http://bioconductor.org/packages/release/bioc/vignettes/msa/inst/doc/msa.pdf</a></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">file_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"IGH_sequencing"</span>, package <span class="op">=</span> <span class="st">"LymphoSeq2"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">stable</span> <span class="op">&lt;-</span> <span class="fu"><a href="readImmunoSeq.html">readImmunoSeq</a></span><span class="op">(</span>path <span class="op">=</span> <span class="va">file_path</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Registered S3 methods overwritten by 'readr':</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   method                    from </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   as.data.frame.spec_tbl_df vroom</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   as_tibble.spec_tbl_df     vroom</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   format.col_spec           vroom</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   print.col_spec            vroom</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   print.collector           vroom</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   print.date_names          vroom</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   print.locale              vroom</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   str.col_spec              vroom</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">ntable</span> <span class="op">&lt;-</span> <span class="fu"><a href="productiveSeq.html">productiveSeq</a></span><span class="op">(</span><span class="va">stable</span>, aggregate <span class="op">=</span> <span class="st">"junction"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">alignSeq</span><span class="op">(</span><span class="va">ntable</span>, repertoire_id <span class="op">=</span> <span class="st">"IGH_MVQ92552A_BL"</span>, type <span class="op">=</span> <span class="st">"junction"</span>, </span></span>
<span class="r-in"><span>         method <span class="op">=</span> <span class="st">"ClustalW"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> use default substitution matrix</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> CLUSTAL 2.1  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Call:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    msa::msa(string_list, method = method)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MsaDNAMultipleAlignment with 65 rows and 186 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      aln                                                   names</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [1] -------------------------...CTTTTGATATCTGGGGCCAAGGG-- IGH_MVQ92552A_BL_16</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [2] ------------------------G...GTATGGACGTCTGGGGCCAAGGG-- IGH_MVQ92552A_BL_55</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [3] ------------------GACAACA...CTTTTGATTTTTGGGGCCAAGGG-- IGH_MVQ92552A_BL_48</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [4] ---------------------CGCG...ACATGGACGTCTGGGGCAAAGGG-- IGH_MVQ92552A_BL_35</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [5] ------------------------A...CTTTTGATGTTTGGGGCCAAGGG-- IGH_MVQ92552A_BL_5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [6] -------------------------...ACATGGACGTCTGGGGCAAAGGG-- IGH_MVQ92552A_BL_34</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [7] -------------------------...CTATGGACGTCTGGGGCCAAGGG-- IGH_MVQ92552A_BL_51</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [8] -------------------------...ACATGGACGTCTGGGGCAAAGGG-- IGH_MVQ92552A_BL_59</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  [9] -------------------------...ACATGGACGTCTGGGGCAAAGGG-- IGH_MVQ92552A_BL_9 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  ... ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [58] ------CAGGGTCACCATGACCAGG...----CCTTA-CTGGGGCCAGGGA-- IGH_MVQ92552A_BL_27</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [59] ---------------CATGACCAGG...----TGACTACTGGGGCCAGGGA-- IGH_MVQ92552A_BL_31</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [60] ---------------------CGCG...--TTGACCTACTGGGGCCAGGGA-- IGH_MVQ92552A_BL_36</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [61] -------------------------...GGTTCAGATACTGGGGCCAGGGA-- IGH_MVQ92552A_BL_60</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [62] ---------------CATCTCCAGA...A--TTGACTACTGGGGCCAGGGA-- IGH_MVQ92552A_BL_29</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [63] ------------------CTCCAGA...ACATGGACGTCTGGGGCAAAGGG-- IGH_MVQ92552A_BL_40</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [64] ----------------------GCC...ACATGGACGTCTGGGGCAAAGGG-- IGH_MVQ92552A_BL_53</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [65] -------------------------...TGTGCCCCCTAGGAGGCCACACTGG IGH_MVQ92552A_BL_64</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  Con ------------------------?...?T????AC??CTGGGGCCAGGGA-- Consensus </span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Wu Elena, Ravishankar Shashidhar, Coffey David.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

