<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Use a Fisher exact test to calculate differential abdunance of each sequence in
two samples and reports the log2 transformed fold change, P value and adjusted P value."><title>Differential abundance analysis — differentialAbundance • LymphoSeq2</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Differential abundance analysis — differentialAbundance"><meta property="og:description" content="Use a Fisher exact test to calculate differential abdunance of each sequence in
two samples and reports the log2 transformed fold change, P value and adjusted P value."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">LymphoSeq2</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/LymphoSeq2.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/elulu3/LymphoSeq2/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Differential abundance analysis</h1>
      <small class="dont-index">Source: <a href="https://github.com/elulu3/LymphoSeq2/blob/HEAD/R/differentialAbundance.R" class="external-link"><code>R/differentialAbundance.R</code></a></small>
      <div class="d-none name"><code>differentialAbundance.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Use a Fisher exact test to calculate differential abdunance of each sequence in
two samples and reports the log2 transformed fold change, P value and adjusted P value.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">differentialAbundance</span><span class="op">(</span>
  <span class="va">study_table</span>,
  repertoire_ids <span class="op">=</span> <span class="cn">NULL</span>,
  abundance <span class="op">=</span> <span class="st">"duplicate_count"</span>,
  type <span class="op">=</span> <span class="st">"junction_aa"</span>,
  q <span class="op">=</span> <span class="fl">1</span>,
  zero <span class="op">=</span> <span class="fl">1</span>,
  parallel <span class="op">=</span> <span class="cn">FALSE</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>study_table</dt>
<dd><p>A tibble consisting of antigen receptor sequences
imported by the LymphoSeq function readImmunoSeq.</p></dd>
<dt>repertoire_ids</dt>
<dd><p>A character vector of two repertoire_ids in study_table
to be compared. If NULL, the first two repertoire_ids from study_table will be used.</p></dd>
<dt>abundance</dt>
<dd><p>The input value for the Fisher exact test.  "duplicate_count"
is recommend but "duplicate_count" may also be used.</p></dd>
<dt>type</dt>
<dd><p>A character vector indicating whether "junction_aa" or "junction" sequences
should be used.  If "junction_aa" is specified, then run productiveSeqs first.</p></dd>
<dt>q</dt>
<dd><p>A numeric value between 0.0 and 1.0 indicating the threshold Holms adjusted
P value (also knowns as the false discovery rate or q value) to subset the results with.
Any sequences with a q value greater than this value will not be shown.</p></dd>
<dt>zero</dt>
<dd><p>A numeric value to set all zero values to when calculating the log2
transformed fold change between samples 1 and 2.  This does not apply to the
p and q value calculations.</p></dd>
<dt>parallel</dt>
<dd><p>A boolean indicating wheter parallel processing should be used or not.</p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>Returns a data frame with columns corresponding to the frequency of the abudance
measure in samples 1 and 2, the P value, Q value (Holms adjusted P value, also knowns as
the false discovery rate), and log2 transformed fold change.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="va">file_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"TCRB_sequencing"</span>, package <span class="op">=</span> <span class="st">"LymphoSeq2"</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">stable</span> <span class="op">&lt;-</span> <span class="fu"><a href="readImmunoSeq.html">readImmunoSeq</a></span><span class="op">(</span>path <span class="op">=</span> <span class="va">file_path</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">1</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">144</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Delimiter:</span> ","</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">chr</span> (69): sequence_id, sequence, sequence_aa, locus, v_call, d_call, d2_call...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">dbl</span> (70): v_score, v_identity, v_support, d_score, d_identity, d_support, d2...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">lgl</span>  (5): rev_comp, productive, vj_in_frame, stop_codon, complete_vdj</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">1000</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">52</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Delimiter:</span> "\t"</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">chr</span> (33): nucleotide, aminoAcid, vMaxResolved, vFamilyName, vGeneName, vGene...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">dbl</span> (15): count (templates/reads), frequencyCount (%), cdr3Length, vDeletion...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">lgl</span>  (4): vFamilyTies, jFamilyTies, jGeneNameTies, jGeneAlleleTies</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Joining, by = c("sequence", "sequence_aa", "v_call", "d_call", "d2_call",</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> "j_call", "junction", "junction_aa", "duplicate_count", "clone_id",</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> "repertoire_id")</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">1000</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">52</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Delimiter:</span> "\t"</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">chr</span> (34): nucleotide, aminoAcid, vMaxResolved, vFamilyName, vGeneName, vGene...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">dbl</span> (15): count (templates/reads), frequencyCount (%), cdr3Length, vDeletion...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">lgl</span>  (3): jFamilyTies, jGeneNameTies, jGeneAlleleTies</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Joining, by = c("sequence", "sequence_aa", "v_call", "d_call", "d2_call",</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> "j_call", "junction", "junction_aa", "duplicate_count", "clone_id",</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> "repertoire_id")</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">414</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">52</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Delimiter:</span> "\t"</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">chr</span> (34): nucleotide, aminoAcid, vMaxResolved, vFamilyName, vGeneName, vGene...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">dbl</span> (15): count (templates/reads), frequencyCount (%), cdr3Length, vDeletion...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">lgl</span>  (3): jFamilyTies, jGeneNameTies, jGeneAlleleTies</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Joining, by = c("sequence", "sequence_aa", "v_call", "d_call", "d2_call",</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> "j_call", "junction", "junction_aa", "duplicate_count", "clone_id",</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> "repertoire_id")</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">1000</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">52</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Delimiter:</span> "\t"</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">chr</span> (34): nucleotide, aminoAcid, vMaxResolved, vFamilyName, vGeneName, vGene...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">dbl</span> (15): count (templates/reads), frequencyCount (%), cdr3Length, vDeletion...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">lgl</span>  (3): jFamilyTies, jGeneNameTies, jGeneAlleleTies</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Joining, by = c("sequence", "sequence_aa", "v_call", "d_call", "d2_call",</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> "j_call", "junction", "junction_aa", "duplicate_count", "clone_id",</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> "repertoire_id")</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">1000</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">52</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Delimiter:</span> "\t"</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">chr</span> (34): nucleotide, aminoAcid, vMaxResolved, vFamilyName, vGeneName, vGene...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">dbl</span> (15): count (templates/reads), frequencyCount (%), cdr3Length, vDeletion...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">lgl</span>  (3): jFamilyTies, jGeneNameTies, jGeneAlleleTies</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Joining, by = c("sequence", "sequence_aa", "v_call", "d_call", "d2_call",</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> "j_call", "junction", "junction_aa", "duplicate_count", "clone_id",</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> "repertoire_id")</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">1000</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">52</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Delimiter:</span> "\t"</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">chr</span> (35): nucleotide, aminoAcid, vMaxResolved, vFamilyName, vGeneName, vGene...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">dbl</span> (15): count (templates/reads), frequencyCount (%), cdr3Length, vDeletion...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">lgl</span>  (2): jFamilyTies, jGeneAlleleTies</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Joining, by = c("sequence", "sequence_aa", "v_call", "d_call", "d2_call",</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> "j_call", "junction", "junction_aa", "duplicate_count", "clone_id",</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> "repertoire_id")</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">920</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">52</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Delimiter:</span> "\t"</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">chr</span> (29): nucleotide, aminoAcid, vMaxResolved, vFamilyName, vGeneName, vFami...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">dbl</span> (14): count (templates/reads), frequencyCount (%), cdr3Length, vDeletion...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">lgl</span>  (9): vGeneAllele, vGeneAlleleTies, dGeneAllele, dFamilyTies, dGeneAllel...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Joining, by = c("sequence", "sequence_aa", "v_call", "d_call", "d2_call",</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> "j_call", "junction", "junction_aa", "duplicate_count", "clone_id",</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> "repertoire_id")</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">1000</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">52</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Delimiter:</span> "\t"</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">chr</span> (29): nucleotide, aminoAcid, vMaxResolved, vFamilyName, vGeneName, vFami...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">dbl</span> (14): count (templates/reads), frequencyCount (%), cdr3Length, vDeletion...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">lgl</span>  (9): vGeneAllele, vGeneAlleleTies, dGeneAllele, dFamilyTies, dGeneAllel...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Joining, by = c("sequence", "sequence_aa", "v_call", "d_call", "d2_call",</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> "j_call", "junction", "junction_aa", "duplicate_count", "clone_id",</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> "repertoire_id")</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">1000</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">52</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Delimiter:</span> "\t"</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">chr</span> (29): nucleotide, aminoAcid, vMaxResolved, vFamilyName, vGeneName, vFami...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">dbl</span> (14): count (templates/reads), frequencyCount (%), cdr3Length, vDeletion...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">lgl</span>  (9): vGeneAllele, vGeneAlleleTies, dGeneAllele, dFamilyTies, dGeneAllel...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Joining, by = c("sequence", "sequence_aa", "v_call", "d_call", "d2_call",</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> "j_call", "junction", "junction_aa", "duplicate_count", "clone_id",</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> "repertoire_id")</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">1000</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">52</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Delimiter:</span> "\t"</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">chr</span> (34): nucleotide, aminoAcid, vMaxResolved, vFamilyName, vGeneName, vGene...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">dbl</span> (15): count (templates/reads), frequencyCount (%), cdr3Length, vDeletion...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">lgl</span>  (3): jFamilyTies, jGeneNameTies, jGeneAlleleTies</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Joining, by = c("sequence", "sequence_aa", "v_call", "d_call", "d2_call",</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> "j_call", "junction", "junction_aa", "duplicate_count", "clone_id",</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> "repertoire_id")</span>
<span class="r-in"></span>
<span class="r-in"><span class="va">atable</span> <span class="op">&lt;-</span> <span class="fu"><a href="productiveSeq.html">productiveSeq</a></span><span class="op">(</span>study_table <span class="op">=</span> <span class="va">stable</span>, aggregate <span class="op">=</span> <span class="st">"junction_aa"</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="fu">differentialAbundance</span><span class="op">(</span>study_table <span class="op">=</span> <span class="va">atable</span>, repertoire_ids <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TRB_Unsorted_949"</span>, <span class="st">"TRB_Unsorted_1320"</span><span class="op">)</span>, </span>
<span class="r-in">                      type <span class="op">=</span> <span class="st">"junction_aa"</span>, q <span class="op">=</span> <span class="fl">0.01</span>, zero <span class="op">=</span> <span class="fl">0.001</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 1,332 × 6</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    junction_aa       TRB_Unsorted_949 TRB_Unsorted_1320        p       q    l2fc</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> CACALGDGYTF                      0                 2  1   <span style="color: #949494;">e</span>+0 1   <span style="color: #949494;">e</span>+0 -<span style="color: #BB0000;">Inf</span>   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> CACQRTGSSYEQYF                   0                 3  1   <span style="color: #949494;">e</span>+0 1   <span style="color: #949494;">e</span>+0 -<span style="color: #BB0000;">Inf</span>   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> CAIGLSNQPQHF                     2                67  1   <span style="color: #949494;">e</span>+0 1   <span style="color: #949494;">e</span>+0   -<span style="color: #BB0000;">5.07</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> CAIKMETPNGEQYF                  29               326  2.66<span style="color: #949494;">e</span><span style="color: #BB0000;">-5</span> 2.66<span style="color: #949494;">e</span><span style="color: #BB0000;">-5</span>   -<span style="color: #BB0000;">3.49</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> CAIRGTEDNNSPLHF                  0                16  1   <span style="color: #949494;">e</span>+0 1   <span style="color: #949494;">e</span>+0 -<span style="color: #BB0000;">Inf</span>   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> CAISDSSYEQYF                     1                23  5.71<span style="color: #949494;">e</span><span style="color: #BB0000;">-1</span> 5.71<span style="color: #949494;">e</span><span style="color: #BB0000;">-1</span>   -<span style="color: #BB0000;">4.52</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> CAISDTGELFF                      8                83  1.39<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> 1.39<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span>   -<span style="color: #BB0000;">3.38</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> CAISEFGLMAREYGYTF                0                 1  1   <span style="color: #949494;">e</span>+0 1   <span style="color: #949494;">e</span>+0 -<span style="color: #BB0000;">Inf</span>   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> CAISEGQGVKPQHF                   0               167  4.92<span style="color: #949494;">e</span><span style="color: #BB0000;">-3</span> 4.92<span style="color: #949494;">e</span><span style="color: #BB0000;">-3</span> -<span style="color: #BB0000;">Inf</span>   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> CAISESGVLNEKLFF                 13               150  4.69<span style="color: #949494;">e</span><span style="color: #BB0000;">-3</span> 4.69<span style="color: #949494;">e</span><span style="color: #BB0000;">-3</span>   -<span style="color: #BB0000;">3.53</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># … with 1,322 more rows</span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by First Last.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

    </footer></div>

  

  

  </body></html>

