<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Plots a Lorenz curve derived from the frequency of the amino acid sequences."><title>Lorenz curve — lorenzCurve • LymphoSeq2</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Lorenz curve — lorenzCurve"><meta property="og:description" content="Plots a Lorenz curve derived from the frequency of the amino acid sequences."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">LymphoSeq2</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/elulu3/LymphoSeq2/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Lorenz curve</h1>
      <small class="dont-index">Source: <a href="https://github.com/elulu3/LymphoSeq2/blob/HEAD/R/plotLorenzCurve.R" class="external-link"><code>R/plotLorenzCurve.R</code></a></small>
      <div class="d-none name"><code>lorenzCurve.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Plots a Lorenz curve derived from the frequency of the amino acid sequences.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">lorenzCurve</span><span class="op">(</span><span class="va">repertoire_ids</span>, <span class="va">study_table</span><span class="op">)</span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>repertoire_ids</dt>
<dd><p>A character vector of repertoire_id names in list.</p></dd>
<dt>study_table</dt>
<dd><p>A tibble generated using the LymphoSeq function readImmunoSeq
or productiveSeq.  "duplicate_frequency" is a required column.</p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>Returns a Lorenz curve.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>The Gini coefficient is an alternative metric used to calculate
repertoire diversity and is derived from the Lorenz curve.  The Lorenz curve
is drawn such that x-axis represents the cumulative percentage of unique
sequences and the y-axis represents the cumulative percentage of reads.  A
line passing through the origin with a slope of 1 reflects equal frequencies
of all sequences.  The Gini coefficient is the ratio of the area between the
line of equality and the observed Lorenz curve over the total area under the
line of equality.</p>
<p>The plot is made using the package ggplot2 and can be reformatted
using ggplot2 functions.  See examples below.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p>An excellent resource for examples on how to reformat a ggplot can
be found in the R Graphics Cookbook online (<a href="http://www.cookbook-r.com/Graphs/" class="external-link">http://www.cookbook-r.com/Graphs/</a>).</p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="va">file_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"TCRB_sequencing"</span>, package <span class="op">=</span> <span class="st">"LymphoSeq2"</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">stable</span> <span class="op">&lt;-</span> <span class="fu"><a href="readImmunoSeq.html">readImmunoSeq</a></span><span class="op">(</span>path <span class="op">=</span> <span class="va">file_path</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">1</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">144</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Delimiter:</span> ","</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">chr</span> (69): sequence_id, sequence, sequence_aa, locus, v_call, d_call, d2_call...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">dbl</span> (70): v_score, v_identity, v_support, d_score, d_identity, d_support, d2...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">lgl</span>  (5): rev_comp, productive, vj_in_frame, stop_codon, complete_vdj</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">1000</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">52</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Delimiter:</span> "\t"</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">chr</span> (33): nucleotide, aminoAcid, vMaxResolved, vFamilyName, vGeneName, vGene...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">dbl</span> (15): count (templates/reads), frequencyCount (%), cdr3Length, vDeletion...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">lgl</span>  (4): vFamilyTies, jFamilyTies, jGeneNameTies, jGeneAlleleTies</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Joining, by = c("sequence", "sequence_aa", "v_call", "d_call", "d2_call",</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> "j_call", "junction", "junction_aa", "duplicate_count", "clone_id",</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> "repertoire_id")</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">1000</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">52</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Delimiter:</span> "\t"</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">chr</span> (34): nucleotide, aminoAcid, vMaxResolved, vFamilyName, vGeneName, vGene...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">dbl</span> (15): count (templates/reads), frequencyCount (%), cdr3Length, vDeletion...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">lgl</span>  (3): jFamilyTies, jGeneNameTies, jGeneAlleleTies</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Joining, by = c("sequence", "sequence_aa", "v_call", "d_call", "d2_call",</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> "j_call", "junction", "junction_aa", "duplicate_count", "clone_id",</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> "repertoire_id")</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">414</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">52</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Delimiter:</span> "\t"</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">chr</span> (34): nucleotide, aminoAcid, vMaxResolved, vFamilyName, vGeneName, vGene...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">dbl</span> (15): count (templates/reads), frequencyCount (%), cdr3Length, vDeletion...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">lgl</span>  (3): jFamilyTies, jGeneNameTies, jGeneAlleleTies</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Joining, by = c("sequence", "sequence_aa", "v_call", "d_call", "d2_call",</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> "j_call", "junction", "junction_aa", "duplicate_count", "clone_id",</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> "repertoire_id")</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">1000</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">52</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Delimiter:</span> "\t"</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">chr</span> (34): nucleotide, aminoAcid, vMaxResolved, vFamilyName, vGeneName, vGene...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">dbl</span> (15): count (templates/reads), frequencyCount (%), cdr3Length, vDeletion...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">lgl</span>  (3): jFamilyTies, jGeneNameTies, jGeneAlleleTies</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Joining, by = c("sequence", "sequence_aa", "v_call", "d_call", "d2_call",</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> "j_call", "junction", "junction_aa", "duplicate_count", "clone_id",</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> "repertoire_id")</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">1000</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">52</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Delimiter:</span> "\t"</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">chr</span> (34): nucleotide, aminoAcid, vMaxResolved, vFamilyName, vGeneName, vGene...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">dbl</span> (15): count (templates/reads), frequencyCount (%), cdr3Length, vDeletion...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">lgl</span>  (3): jFamilyTies, jGeneNameTies, jGeneAlleleTies</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Joining, by = c("sequence", "sequence_aa", "v_call", "d_call", "d2_call",</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> "j_call", "junction", "junction_aa", "duplicate_count", "clone_id",</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> "repertoire_id")</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">1000</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">52</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Delimiter:</span> "\t"</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">chr</span> (35): nucleotide, aminoAcid, vMaxResolved, vFamilyName, vGeneName, vGene...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">dbl</span> (15): count (templates/reads), frequencyCount (%), cdr3Length, vDeletion...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">lgl</span>  (2): jFamilyTies, jGeneAlleleTies</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Joining, by = c("sequence", "sequence_aa", "v_call", "d_call", "d2_call",</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> "j_call", "junction", "junction_aa", "duplicate_count", "clone_id",</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> "repertoire_id")</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">920</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">52</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Delimiter:</span> "\t"</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">chr</span> (29): nucleotide, aminoAcid, vMaxResolved, vFamilyName, vGeneName, vFami...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">dbl</span> (14): count (templates/reads), frequencyCount (%), cdr3Length, vDeletion...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">lgl</span>  (9): vGeneAllele, vGeneAlleleTies, dGeneAllele, dFamilyTies, dGeneAllel...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Joining, by = c("sequence", "sequence_aa", "v_call", "d_call", "d2_call",</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> "j_call", "junction", "junction_aa", "duplicate_count", "clone_id",</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> "repertoire_id")</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">1000</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">52</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Delimiter:</span> "\t"</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">chr</span> (29): nucleotide, aminoAcid, vMaxResolved, vFamilyName, vGeneName, vFami...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">dbl</span> (14): count (templates/reads), frequencyCount (%), cdr3Length, vDeletion...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">lgl</span>  (9): vGeneAllele, vGeneAlleleTies, dGeneAllele, dFamilyTies, dGeneAllel...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Joining, by = c("sequence", "sequence_aa", "v_call", "d_call", "d2_call",</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> "j_call", "junction", "junction_aa", "duplicate_count", "clone_id",</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> "repertoire_id")</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">1000</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">52</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Delimiter:</span> "\t"</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">chr</span> (29): nucleotide, aminoAcid, vMaxResolved, vFamilyName, vGeneName, vFami...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">dbl</span> (14): count (templates/reads), frequencyCount (%), cdr3Length, vDeletion...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">lgl</span>  (9): vGeneAllele, vGeneAlleleTies, dGeneAllele, dFamilyTies, dGeneAllel...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Joining, by = c("sequence", "sequence_aa", "v_call", "d_call", "d2_call",</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> "j_call", "junction", "junction_aa", "duplicate_count", "clone_id",</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> "repertoire_id")</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">1000</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">52</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Delimiter:</span> "\t"</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">chr</span> (34): nucleotide, aminoAcid, vMaxResolved, vFamilyName, vGeneName, vGene...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">dbl</span> (15): count (templates/reads), frequencyCount (%), cdr3Length, vDeletion...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">lgl</span>  (3): jFamilyTies, jGeneNameTies, jGeneAlleleTies</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Joining, by = c("sequence", "sequence_aa", "v_call", "d_call", "d2_call",</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> "j_call", "junction", "junction_aa", "duplicate_count", "clone_id",</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> "repertoire_id")</span>
<span class="r-in"></span>
<span class="r-in"><span class="va">repertoire_ids</span> <span class="op">&lt;-</span> <span class="va">stable</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">repertoire_id</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="op">)</span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in stable %&gt;% dplyr::pull(repertoire_id) %&gt;% unique():</span> could not find function "%&gt;%"</span>
<span class="r-in"></span>
<span class="r-in"><span class="fu">lorenzCurve</span><span class="op">(</span>repertoire_ids <span class="op">=</span> <span class="va">repertoire_ids</span>, study_table <span class="op">=</span> <span class="va">stable</span><span class="op">)</span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in .approxfun(x, y, v, method, yleft, yright, f, na.rm):</span> object 'repertoire_ids' not found</span>
<span class="r-in"></span>
<span class="r-in"><span class="va">atable</span> <span class="op">&lt;-</span> <span class="fu"><a href="productiveSeq.html">productiveSeq</a></span><span class="op">(</span>study_table <span class="op">=</span> <span class="va">stable</span>, aggregate <span class="op">=</span> <span class="st">"junction_aa"</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">repertoire_ids</span> <span class="op">&lt;-</span> <span class="va">atable</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">repertoire_id</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="op">)</span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in atable %&gt;% dplyr::pull(repertoire_id) %&gt;% unique():</span> could not find function "%&gt;%"</span>
<span class="r-in"></span>
<span class="r-in"><span class="fu">lorenzCurve</span><span class="op">(</span>repertoire_ids <span class="op">=</span> <span class="va">repertoire_ids</span>, study_table <span class="op">=</span> <span class="va">atable</span><span class="op">)</span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in .approxfun(x, y, v, method, yleft, yright, f, na.rm):</span> object 'repertoire_ids' not found</span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># Change the legend labels, line colors, and add a title</span></span>
<span class="r-in"><span class="va">repertoire_ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TRB_Unsorted_0"</span>, <span class="st">"TRB_Unsorted_32"</span>, </span>
<span class="r-in">   <span class="st">"TRB_Unsorted_83"</span>, <span class="st">"TRB_Unsorted_949"</span>, <span class="st">"TRB_Unsorted_1320"</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">lorenz_curve</span> <span class="op">&lt;-</span> <span class="fu">lorenzCurve</span><span class="op">(</span>repertoire_ids <span class="op">=</span> <span class="va">repertoire_ids</span>, study_table <span class="op">=</span> <span class="va">atable</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">labels</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Day 0"</span>, <span class="st">"Day 32"</span>, <span class="st">"Day 83"</span>, <span class="st">"Day 949"</span>, <span class="st">"Day 1320"</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">colors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"navyblue"</span>, <span class="st">"red"</span>, <span class="st">"darkgreen"</span>, <span class="st">"orange"</span>, <span class="st">"purple"</span>,</span>
<span class="r-in">             <span class="st">"yellow"</span>, <span class="st">"pink"</span>, <span class="st">"lightgreen"</span>, <span class="st">"cyan"</span>, <span class="st">"maroon"</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">lorenz_curve</span> <span class="op">+</span> </span>
<span class="r-in"><span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">"repertoire_ids"</span>, </span>
<span class="r-in">                            breaks <span class="op">=</span> <span class="va">repertoire_ids</span>, </span>
<span class="r-in">                            labels <span class="op">=</span> <span class="va">labels</span>, values <span class="op">=</span> <span class="va">colors</span><span class="op">)</span> <span class="op">+</span> </span>
<span class="r-in"><span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Figure Title"</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Scale for 'colour' is already present. Adding another scale for 'colour',</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> which will replace the existing scale.</span>
<span class="r-plt img"><img src="lorenzCurve-1.png" alt="" width="700" height="433"></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by First Last.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

    </footer></div>

  

  

  </body></html>

