<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Create a phylogenetic tree using neighbor joining tree estimation for amino
acid or junction CDR3 sequences in a list of data frames."><title>Create phylogenetic tree — phyloTree • LymphoSeq2</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Create phylogenetic tree — phyloTree"><meta property="og:description" content="Create a phylogenetic tree using neighbor joining tree estimation for amino
acid or junction CDR3 sequences in a list of data frames."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">LymphoSeq2</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/LymphoSeq2.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/shashidhar22/LymphoSeq2/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Create phylogenetic tree</h1>
      <small class="dont-index">Source: <a href="https://github.com/shashidhar22/LymphoSeq2/blob/HEAD/R/phyloTree.R" class="external-link"><code>R/phyloTree.R</code></a></small>
      <div class="d-none name"><code>phyloTree.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Create a phylogenetic tree using neighbor joining tree estimation for amino
acid or junction CDR3 sequences in a list of data frames.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">phyloTree</span><span class="op">(</span></span>
<span>  <span class="va">study_table</span>,</span>
<span>  <span class="va">repertoire_ids</span>,</span>
<span>  type <span class="op">=</span> <span class="st">"junction"</span>,</span>
<span>  layout <span class="op">=</span> <span class="st">"rectangular"</span>,</span>
<span>  label <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>study_table</dt>
<dd><p>A tibble of unproductive junction sequences or productive
junction sequences generated by the LymphoSeq2 function <code><a href="productiveSeq.html">productiveSeq()</a></code>.
"v_family", "d_family", "j_family", and "duplicate_count" are required
columns.</p></dd>


<dt>repertoire_ids</dt>
<dd><p>A character vector indicating the name of the
repertoire_id in the study table.</p></dd>


<dt>type</dt>
<dd><p>A character vector indicating whether <code>"junction_aa"</code> or
<code>"junction"</code> (the default) sequences should be compared.</p></dd>


<dt>layout</dt>
<dd><p>A character vector indicating the tree layout. Options include
<code>"rectangular"</code> (the default), <code>"slanted"</code>, <code>"fan"</code>, <code>"circular"</code>, <code>"radial"</code>
and <code>"unrooted"</code>.</p></dd>


<dt>label</dt>
<dd><p>A Boolean value</p><ul><li><p><code>TRUE</code> (the default): use sequencing duplicate_count to label the tree</p></li>
<li><p><code>FALSE</code> : do not show duplicate_count</p></li>
</ul></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>Returns a phylogenetic tree where each leaf represents a sequence
color coded by the V, D, and J gene usage. The number next to each leaf
refers to the sequence duplicate_count. A triangle shaped leaf indicates the
dominant sequence. Refer to the ggtree Bioconductor package documentation for
details on how to manipulate the tree.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">file_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"IGH_sequencing"</span>, package <span class="op">=</span> <span class="st">"LymphoSeq2"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">study_table</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="readImmunoSeq.html">readImmunoSeq</a></span><span class="op">(</span>path <span class="op">=</span> <span class="va">file_path</span>, threads <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">study_table</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="topSeqs.html">topSeqs</a></span><span class="op">(</span><span class="va">study_table</span>, top <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">nucleotide_table</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="productiveSeq.html">productiveSeq</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  study_table <span class="op">=</span> <span class="va">study_table</span>,</span></span>
<span class="r-in"><span>  aggregate <span class="op">=</span> <span class="st">"junction"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu">phyloTree</span><span class="op">(</span></span></span>
<span class="r-in"><span>  study_table <span class="op">=</span> <span class="va">nucleotide_table</span>, repertoire_ids <span class="op">=</span> <span class="st">"IGH_MVQ92552A_BL"</span>,</span></span>
<span class="r-in"><span>  type <span class="op">=</span> <span class="st">"junction"</span>, layout <span class="op">=</span> <span class="st">"rectangular"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The `&lt;scale&gt;` argument of `guides()` cannot be `FALSE`. Use "none" instead as</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> of ggplot2 3.3.4.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">LymphoSeq2</span> package.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span>   Please report the issue at</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span>   <span style="color: #0000BB; font-style: italic;">&lt;https://github.com/shashidhar22/LymphoSeq2/issues&gt;</span>.</span>
<span class="r-plt img"><img src="phyloTree-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span><span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu">phyloTree</span><span class="op">(</span></span></span>
<span class="r-in"><span>  study_table <span class="op">=</span> <span class="va">nucleotide_table</span>, repertoire_ids <span class="op">=</span> <span class="st">"IGH_MVQ92552A_BL"</span>,</span></span>
<span class="r-in"><span>  type <span class="op">=</span> <span class="st">"junction_aa"</span>, layout <span class="op">=</span> <span class="st">"circular"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="phyloTree-2.png" alt="" width="700" height="433"></span>
<span class="r-in"><span><span class="co"># Add scale and title to figure</span></span></span>
<span class="r-in"><span><span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu">phyloTree</span><span class="op">(</span></span></span>
<span class="r-in"><span>  study_table <span class="op">=</span> <span class="va">nucleotide_table</span>, repertoire_ids <span class="op">=</span> <span class="st">"IGH_MVQ92552A_BL"</span>,</span></span>
<span class="r-in"><span>  type <span class="op">=</span> <span class="st">"junction_aa"</span>, layout <span class="op">=</span> <span class="st">"rectangular"</span></span></span>
<span class="r-in"><span><span class="op">)</span> <span class="op">+</span></span></span>
<span class="r-in"><span>  <span class="fu">ggtree</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ggtree/man/theme_tree2.html" class="external-link">theme_tree2</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span></span>
<span class="r-in"><span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    legend.position <span class="op">=</span> <span class="st">"right"</span>,</span></span>
<span class="r-in"><span>    legend.key <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_rect</a></span><span class="op">(</span>colour <span class="op">=</span> <span class="st">"white"</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="op">)</span> <span class="op">+</span></span></span>
<span class="r-in"><span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Title"</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="phyloTree-3.png" alt="" width="700" height="433"></span>
<span class="r-in"><span><span class="co"># Hide legend and leaf labels</span></span></span>
<span class="r-in"><span><span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu">phyloTree</span><span class="op">(</span></span></span>
<span class="r-in"><span>  study_table <span class="op">=</span> <span class="va">nucleotide_table</span>, repertoire_ids <span class="op">=</span> <span class="st">"IGH_MVQ92552A_BL"</span>,</span></span>
<span class="r-in"><span>  type <span class="op">=</span> <span class="st">"junction"</span>, layout <span class="op">=</span> <span class="st">"rectangular"</span>, label <span class="op">=</span> <span class="cn">FALSE</span></span></span>
<span class="r-in"><span><span class="op">)</span> <span class="op">+</span></span></span>
<span class="r-in"><span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="phyloTree-4.png" alt="" width="700" height="433"></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Wu Elena, Ravishankar Shashidhar, Coffey David.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

