<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="LymphoSeq2">
<title>Analysis of Adaptive Immune Receptor Repertoire Sequencing data with LymphoSeq2 • LymphoSeq2</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Analysis of Adaptive Immune Receptor Repertoire Sequencing data with LymphoSeq2">
<meta property="og:description" content="LymphoSeq2">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">LymphoSeq2</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/LymphoSeq2.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/shashidhar22/LymphoSeq2/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">



<script src="LymphoSeq2_files/htmlwidgets-1.6.2/htmlwidgets.js"></script><link href="LymphoSeq2_files/wordcloud2-0.0.1/wordcloud.css" rel="stylesheet">
<script src="LymphoSeq2_files/wordcloud2-0.0.1/wordcloud2-all.js"></script><script src="LymphoSeq2_files/wordcloud2-0.0.1/hover.js"></script><script src="LymphoSeq2_files/wordcloud2-binding-0.2.1/wordcloud2.js"></script><div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Analysis of Adaptive Immune Receptor Repertoire Sequencing data with LymphoSeq2</h1>
                        <h4 data-toc-skip class="author">Elena Wu</h4>
                        <h4 data-toc-skip class="author">Shashidhar
Ravishankar</h4>
                        <h4 data-toc-skip class="author">David
Coffey</h4>
            
            <h4 data-toc-skip class="date">2023-03-21</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/shashidhar22/LymphoSeq2/blob/HEAD/vignettes/LymphoSeq2.Rmd" class="external-link"><code>vignettes/LymphoSeq2.Rmd</code></a></small>
      <div class="d-none name"><code>LymphoSeq2.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/shashidhar22/LymphoSeq2" class="external-link">LymphoSeq2</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: data.table</span></span>
<span><span class="co">#&gt; Registered S3 methods overwritten by 'ggalt':</span></span>
<span><span class="co">#&gt;   method                  from   </span></span>
<span><span class="co">#&gt;   grid.draw.absoluteGrob  ggplot2</span></span>
<span><span class="co">#&gt;   grobHeight.absoluteGrob ggplot2</span></span>
<span><span class="co">#&gt;   grobWidth.absoluteGrob  ggplot2</span></span>
<span><span class="co">#&gt;   grobX.absoluteGrob      ggplot2</span></span>
<span><span class="co">#&gt;   grobY.absoluteGrob      ggplot2</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">RColorBrewer</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">grDevices</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/lchiffon/wordcloud2" class="external-link">wordcloud2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Attaching core tidyverse packages</span> ──────────────────────── tidyverse 2.0.0 ──</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">dplyr    </span> 1.1.0     <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">readr    </span> 2.1.4</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">forcats  </span> 1.0.0     <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">stringr  </span> 1.5.0</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">ggplot2  </span> 3.4.1     <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">tibble   </span> 3.2.1</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">lubridate</span> 1.9.2     <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">tidyr    </span> 1.3.0</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">purrr    </span> 1.0.1</span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Conflicts</span> ────────────────────────────────────────── tidyverse_conflicts() ──</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">dplyr</span>::<span style="color: #00BB00;">between()</span>     masks <span style="color: #0000BB;">data.table</span>::between()</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">dplyr</span>::<span style="color: #00BB00;">filter()</span>      masks <span style="color: #0000BB;">stats</span>::filter()</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">dplyr</span>::<span style="color: #00BB00;">first()</span>       masks <span style="color: #0000BB;">data.table</span>::first()</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">lubridate</span>::<span style="color: #00BB00;">hour()</span>    masks <span style="color: #0000BB;">data.table</span>::hour()</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">lubridate</span>::<span style="color: #00BB00;">isoweek()</span> masks <span style="color: #0000BB;">data.table</span>::isoweek()</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">dplyr</span>::<span style="color: #00BB00;">lag()</span>         masks <span style="color: #0000BB;">stats</span>::lag()</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">dplyr</span>::<span style="color: #00BB00;">last()</span>        masks <span style="color: #0000BB;">data.table</span>::last()</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">lubridate</span>::<span style="color: #00BB00;">mday()</span>    masks <span style="color: #0000BB;">data.table</span>::mday()</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">lubridate</span>::<span style="color: #00BB00;">minute()</span>  masks <span style="color: #0000BB;">data.table</span>::minute()</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">lubridate</span>::<span style="color: #00BB00;">month()</span>   masks <span style="color: #0000BB;">data.table</span>::month()</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">lubridate</span>::<span style="color: #00BB00;">quarter()</span> masks <span style="color: #0000BB;">data.table</span>::quarter()</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">lubridate</span>::<span style="color: #00BB00;">second()</span>  masks <span style="color: #0000BB;">data.table</span>::second()</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">purrr</span>::<span style="color: #00BB00;">transpose()</span>   masks <span style="color: #0000BB;">data.table</span>::transpose()</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">lubridate</span>::<span style="color: #00BB00;">wday()</span>    masks <span style="color: #0000BB;">data.table</span>::wday()</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">lubridate</span>::<span style="color: #00BB00;">week()</span>    masks <span style="color: #0000BB;">data.table</span>::week()</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">lubridate</span>::<span style="color: #00BB00;">yday()</span>    masks <span style="color: #0000BB;">data.table</span>::yday()</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">lubridate</span>::<span style="color: #00BB00;">year()</span>    masks <span style="color: #0000BB;">data.table</span>::year()</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use the conflicted package (<span style="color: #0000BB; font-style: italic;">&lt;http://conflicted.r-lib.org/&gt;</span>) to force all conflicts to become errors</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://vroom.r-lib.org" class="external-link">vroom</a></span><span class="op">)</span></span></code></pre></div>
<p>Adaptive Immune Receptor Repertoire Sequencing (AIRR-seq) provides a
unique opportunity to interrogate the adaptive immune repertoire under
various clinical conditions. The utility offered by this technology has
quickly garnered interest from a community of clinicians and researchers
investigating the immunological landscapes of a large spectrum of health
and disease states. LymphoSeq2 is a toolkit that allows users to import,
manipulate and visualize AIRR-Seq data from various AIRR-Seq assays such
as Adaptive ImmunoSEQ and BGI-IRSeq, with support for 10X VDJ sequencing
coming soon. The platform also supports the importing of AIRR-seq data
processed using the MiXCR pipeline. The vignette highlights some of the
key features of LymphoSeq2.</p>
<div class="section level2">
<h2 id="importing-data">Importing data<a class="anchor" aria-label="anchor" href="#importing-data"></a>
</h2>
<p>The function <code>readImmunoSeq</code> imports AIRR-seq receptor
files from Adaptive ImmunoSEQ assay as well well as BGI-IRSeq assay. The
sequences can be (.tsv) files processed using one of the three following
platforms: Adaptive Biotechnologies ImmunoSEQ analyzer, BGI IR-SEQ
iMonitor platform, and the MiXCR pipeline for AIRR-seq data analysis.
The function has the ability to identify file type based on the headers
provided in the (.tsv) file, accordingly the data is transformed into a
format that is compatible AIRR-Community guidelines (<a href="https://github.com/airr-community/airr-standards" class="external-link uri">https://github.com/airr-community/airr-standards</a>).</p>
<p>To explore the features of LymphoSeq2, this package includes 2
example data sets. The first is a data set of T cell receptor beta
(TCRB) sequencing from 10 blood samples acquired serially from a single
patient who underwent a bone marrow transplant (Kanakry, C.G., et
al. JCI Insight 2016;1(5):pii: e86252). The second, is a data set of B
cell receptor immunoglobulin heavy (IGH) chain sequencing from Burkitt
lymphoma tumor biopsies acquired from 10 different individuals
(Lombardo, K.A., et al. Blood Advances 2017 1:535-544). To improve
performance, both data sets contain only the top 1,000 most frequent
sequences. The complete data sets are publicly available through
Adapatives’ immuneACCESS portal. As shown in the example below, you can
specify the path to the example data sets using the command</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"TCRB_sequencing"</span>, package <span class="op">=</span> <span class="st">"LymphoSeq2"</span><span class="op">)</span> <span class="co">#For the TCRB files</span></span>
<span><span class="co">#&gt; [1] "/home/runner/work/_temp/Library/LymphoSeq2/extdata/TCRB_sequencing"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"IGH_sequencing"</span>, package <span class="op">=</span> <span class="st">"LymphoSeq2"</span><span class="op">)</span> <span class="co">#For the IGH files.</span></span>
<span><span class="co">#&gt; [1] "/home/runner/work/_temp/Library/LymphoSeq2/extdata/IGH_sequencing"</span></span></code></pre></div>
<p><code>readImmunoSeq</code> can take as input, a single file name, a
list of files or the path to a directory containing AIRR-seq data. The
columns are renamed to follow AIRR-community guidelines based on the
input file type. The function returns a tibble with individual file
names set as the <code>repertoire_id</code>. The CDR3 nucleotide and
amino acid sequences are denoted by the <code>junction</code> and
<code>junction_aa</code> fields respectively. The counts of the CDR3
sequences observed, and their frequency in each individual repertoire is
denoted by the <code>duplicate_count</code> and
<code>duplicate_frequency</code> field respectively.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">study_files</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"TCRB_sequencing"</span>, package <span class="op">=</span> <span class="st">"LymphoSeq2"</span><span class="op">)</span> </span>
<span><span class="va">study_table</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/readImmunoSeq.html">readImmunoSeq</a></span><span class="op">(</span><span class="va">study_files</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="../reference/topSeqs.html">topSeqs</a></span><span class="op">(</span>top <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></span></code></pre></div>
<p>Looking at the <code>study_table</code> we see a tibble with 145
columns and 1000 rows</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">study_table</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 1,000 × 145</span></span></span>
<span><span class="co">#&gt;    sequen…¹ seque…² seque…³ rev_c…⁴ produ…⁵ vj_in…⁶ stop_…⁷ compl…⁸ locus v_call</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> TRB_CD4… GAGTCA… CASSES… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>    TRBV1…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> TRB_CD4… GCCCTC… <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>      TRUE    FALSE   <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> TRB_CD4… ATTCCC… <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>      TRUE    FALSE   <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> TRB_CD4… GTGACA… CASSPR… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>    TRBV1…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> TRB_CD4… ACCTTG… CASSLD… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>    TRBV5…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> TRB_CD4… GTGACC… CSAKTS… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   FALSE   <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> TRB_CD4… ACCCTG… CASSQD… FALSE   TRUE    <span style="color: #BB0000;">NA</span>      TRUE    FALSE   <span style="color: #BB0000;">NA</span>    TRBV4…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> TRB_CD4… CTCCTT… CAWSDF… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>    TRBV3…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> TRB_CD4… CTGACG… CASSPD… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>    TRBV7…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> TRB_CD4… TCAGAA… CASSFR… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   FALSE   <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 990 more rows, 135 more variables: d_call &lt;chr&gt;, d2_call &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   j_call &lt;chr&gt;, c_call &lt;chr&gt;, sequence_alignment &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   sequence_alignment_aa &lt;chr&gt;, germline_alignment &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   germline_alignment_aa &lt;chr&gt;, junction &lt;chr&gt;, junction_aa &lt;chr&gt;, np1 &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   np1_aa &lt;chr&gt;, np2 &lt;chr&gt;, np2_aa &lt;chr&gt;, np3 &lt;chr&gt;, np3_aa &lt;chr&gt;, cdr1 &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   cdr1_aa &lt;chr&gt;, cdr2 &lt;chr&gt;, cdr2_aa &lt;chr&gt;, cdr3 &lt;chr&gt;, cdr3_aa &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   fwr1 &lt;chr&gt;, fwr1_aa &lt;chr&gt;, fwr2 &lt;chr&gt;, fwr2_aa &lt;chr&gt;, fwr3 &lt;chr&gt;, …</span></span></span></code></pre></div>
<p>Since the study table is a tibble, we can use tidyverse syntax to
extract a list of sample names</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">study_table</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">repertoire_id</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "TRB_CD4_949"       "TRB_CD8_949"       "TRB_CD8_CMV_369"  </span></span>
<span><span class="co">#&gt;  [4] "TRB_Unsorted_0"    "TRB_Unsorted_1320" "TRB_Unsorted_1496"</span></span>
<span><span class="co">#&gt;  [7] "TRB_Unsorted_32"   "TRB_Unsorted_369"  "TRB_Unsorted_83"  </span></span>
<span><span class="co">#&gt; [10] "TRB_Unsorted_949"</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="subsetting-data">Subsetting Data<a class="anchor" aria-label="anchor" href="#subsetting-data"></a>
</h2>
<p>The tibble structure of the TCR data allows for easy subsampling of
data. To select the TCR sequences from any given samples in the dataset,
the <code>filter</code> function from the <code>dplyr</code> package can
be used.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">TRB_Unsorted_0</span> <span class="op">&lt;-</span> <span class="va">study_table</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">repertoire_id</span> <span class="op">==</span> <span class="st">"TRB_Unsorted_0"</span><span class="op">)</span></span>
<span><span class="va">TRB_Unsorted_0</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 100 × 145</span></span></span>
<span><span class="co">#&gt;    sequen…¹ seque…² seque…³ rev_c…⁴ produ…⁵ vj_in…⁶ stop_…⁷ compl…⁸ locus v_call</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> TRB_Uns… TCAATT… <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>      TRUE    FALSE   <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> TRB_Uns… CTGATT… CASSPV… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>    TRBV2…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> TRB_Uns… ATCAAT… CASSQE… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   FALSE   <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> TRB_Uns… CACACC… CASSQE… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>    TRBV4…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> TRB_Uns… TGCCAT… <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>      TRUE    FALSE   <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> TRB_Uns… GCCAGC… CASSLE… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>    TRBV2…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> TRB_Uns… CCCCTG… CASSPG… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>    TRBV2…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> TRB_Uns… AGTGCC… CSARSP… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   FALSE   <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> TRB_Uns… GGAGCT… <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>      TRUE    FALSE   <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> TRB_Uns… CTGTAG… CASSEK… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>    TRBV1…</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 90 more rows, 135 more variables: d_call &lt;chr&gt;, d2_call &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   j_call &lt;chr&gt;, c_call &lt;chr&gt;, sequence_alignment &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   sequence_alignment_aa &lt;chr&gt;, germline_alignment &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   germline_alignment_aa &lt;chr&gt;, junction &lt;chr&gt;, junction_aa &lt;chr&gt;, np1 &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   np1_aa &lt;chr&gt;, np2 &lt;chr&gt;, np2_aa &lt;chr&gt;, np3 &lt;chr&gt;, np3_aa &lt;chr&gt;, cdr1 &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   cdr1_aa &lt;chr&gt;, cdr2 &lt;chr&gt;, cdr2_aa &lt;chr&gt;, cdr3 &lt;chr&gt;, cdr3_aa &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   fwr1 &lt;chr&gt;, fwr1_aa &lt;chr&gt;, fwr2 &lt;chr&gt;, fwr2_aa &lt;chr&gt;, fwr3 &lt;chr&gt;, …</span></span></span></code></pre></div>
<p>The <code>str_detect</code> function from <code>stringr</code>
package can be used in conjunction with the <code>filter</code> to find
samples using a pattern</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">CMV</span> <span class="op">&lt;-</span> <span class="va">study_table</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>       <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_detect.html" class="external-link">str_detect</a></span><span class="op">(</span><span class="va">repertoire_id</span>, <span class="st">"CMV"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">CMV</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 100 × 145</span></span></span>
<span><span class="co">#&gt;    sequen…¹ seque…² seque…³ rev_c…⁴ produ…⁵ vj_in…⁶ stop_…⁷ compl…⁸ locus v_call</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> TRB_CD8… CAGCGC… CASSPP… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>    TRBV7…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> TRB_CD8… CAGCCC… CASSPA… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   FALSE   <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> TRB_CD8… CAGCCT… CASSQD… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>    TRBV1…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> TRB_CD8… TCGGCC… CASSQD… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>    TRBV4…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> TRB_CD8… ATCCTG… CASSLQ… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>    TRBV2…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> TRB_CD8… GAGGAT… <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>      TRUE    TRUE    <span style="color: #BB0000;">NA</span>    TRBV1…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> TRB_CD8… ACCCTG… CASSQD… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>    TRBV4…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> TRB_CD8… GAGTCC… CASSLA… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   FALSE   <span style="color: #BB0000;">NA</span>    TRBV2…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> TRB_CD8… CTCCTC… CAISDT… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   FALSE   <span style="color: #BB0000;">NA</span>    TRBV1…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> TRB_CD8… TCCAGC… <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>      TRUE    TRUE    <span style="color: #BB0000;">NA</span>    TRBV1…</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 90 more rows, 135 more variables: d_call &lt;chr&gt;, d2_call &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   j_call &lt;chr&gt;, c_call &lt;chr&gt;, sequence_alignment &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   sequence_alignment_aa &lt;chr&gt;, germline_alignment &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   germline_alignment_aa &lt;chr&gt;, junction &lt;chr&gt;, junction_aa &lt;chr&gt;, np1 &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   np1_aa &lt;chr&gt;, np2 &lt;chr&gt;, np2_aa &lt;chr&gt;, np3 &lt;chr&gt;, np3_aa &lt;chr&gt;, cdr1 &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   cdr1_aa &lt;chr&gt;, cdr2 &lt;chr&gt;, cdr2_aa &lt;chr&gt;, cdr3 &lt;chr&gt;, cdr3_aa &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   fwr1 &lt;chr&gt;, fwr1_aa &lt;chr&gt;, fwr2 &lt;chr&gt;, fwr2_aa &lt;chr&gt;, fwr3 &lt;chr&gt;, …</span></span></span></code></pre></div>
<p>A metadata file for the TCR sequencing samples can easily be combined
with the <code>study_table</code> by reading in the metadata file as a
tibble and using the <code><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">dplyr::left_join</a></code> function to merge the
two tables. In the example below, a metadata file is imported for the
example TCRB data set which contains information on the number of days
post bone marrow transplant the sample was collected and the cellular
phenotype the blood sample was sorted for prior to sequencing.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">TCRB_metadata</span> <span class="op">&lt;-</span> <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link">read_csv</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"TCRB_metadata.csv"</span>, package <span class="op">=</span> <span class="st">"LymphoSeq2"</span><span class="op">)</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">TCRB_metadata</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 10 × 3</span></span></span>
<span><span class="co">#&gt;    samples             day phenotype</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> TRB_Unsorted_0        0 Unsorted </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> TRB_Unsorted_32      32 Unsorted </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> TRB_Unsorted_83      82 Unsorted </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> TRB_CD8_CMV_369     369 CD8+CMV+ </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> TRB_Unsorted_369    369 Unsorted </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> TRB_CD4_949         949 CD4+     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> TRB_CD8_949         949 CD8+     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> TRB_Unsorted_949    949 Unsorted </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> TRB_Unsorted_1320  <span style="text-decoration: underline;">1</span>320 Unsorted </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> TRB_Unsorted_1496  <span style="text-decoration: underline;">1</span>496 Unsorted</span></span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">study_table</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">study_table</span>, <span class="va">TCRB_metadata</span>, by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"repertoire_id"</span> <span class="op">=</span> <span class="st">"samples"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">study_table</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 1,000 × 147</span></span></span>
<span><span class="co">#&gt;    sequen…¹ seque…² seque…³ rev_c…⁴ produ…⁵ vj_in…⁶ stop_…⁷ compl…⁸ locus v_call</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> TRB_CD4… GAGTCA… CASSES… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>    TRBV1…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> TRB_CD4… GCCCTC… <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>      TRUE    FALSE   <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> TRB_CD4… ATTCCC… <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>      TRUE    FALSE   <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> TRB_CD4… GTGACA… CASSPR… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>    TRBV1…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> TRB_CD4… ACCTTG… CASSLD… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>    TRBV5…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> TRB_CD4… GTGACC… CSAKTS… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   FALSE   <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> TRB_CD4… ACCCTG… CASSQD… FALSE   TRUE    <span style="color: #BB0000;">NA</span>      TRUE    FALSE   <span style="color: #BB0000;">NA</span>    TRBV4…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> TRB_CD4… CTCCTT… CAWSDF… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>    TRBV3…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> TRB_CD4… CTGACG… CASSPD… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>    TRBV7…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> TRB_CD4… TCAGAA… CASSFR… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   FALSE   <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 990 more rows, 137 more variables: d_call &lt;chr&gt;, d2_call &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   j_call &lt;chr&gt;, c_call &lt;chr&gt;, sequence_alignment &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   sequence_alignment_aa &lt;chr&gt;, germline_alignment &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   germline_alignment_aa &lt;chr&gt;, junction &lt;chr&gt;, junction_aa &lt;chr&gt;, np1 &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   np1_aa &lt;chr&gt;, np2 &lt;chr&gt;, np2_aa &lt;chr&gt;, np3 &lt;chr&gt;, np3_aa &lt;chr&gt;, cdr1 &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   cdr1_aa &lt;chr&gt;, cdr2 &lt;chr&gt;, cdr2_aa &lt;chr&gt;, cdr3 &lt;chr&gt;, cdr3_aa &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   fwr1 &lt;chr&gt;, fwr1_aa &lt;chr&gt;, fwr2 &lt;chr&gt;, fwr2_aa &lt;chr&gt;, fwr3 &lt;chr&gt;, …</span></span></span></code></pre></div>
<p>Now the metadata information can be used to further subset the data.
For instance to select all “Unsorted” samples collected more than 300
days after bone marrow transplant, we would use the following code</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">unsorted_300</span> <span class="op">&lt;-</span> <span class="va">study_table</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>                <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">day</span> <span class="op">&gt;</span> <span class="fl">300</span> <span class="op">&amp;</span> <span class="va">phenotype</span> <span class="op">==</span> <span class="st">"Unsorted"</span><span class="op">)</span></span>
<span><span class="va">unsorted_300</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 400 × 147</span></span></span>
<span><span class="co">#&gt;    sequen…¹ seque…² seque…³ rev_c…⁴ produ…⁵ vj_in…⁶ stop_…⁷ compl…⁸ locus v_call</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> TRB_Uns… CAGCCC… CASSPA… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   FALSE   <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> TRB_Uns… CAGCGC… CASSPP… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>    TRBV7…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> TRB_Uns… GAGGAT… <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>      TRUE    TRUE    <span style="color: #BB0000;">NA</span>    TRBV1…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> TRB_Uns… ATCCTG… CASSLQ… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>    TRBV2…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> TRB_Uns… GAGTCA… CASSES… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>    TRBV1…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> TRB_Uns… CAGCCT… CASSQD… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>    TRBV1…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> TRB_Uns… GAGTCC… CASSLA… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   FALSE   <span style="color: #BB0000;">NA</span>    TRBV2…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> TRB_Uns… GCCCTC… <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>      TRUE    FALSE   <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> TRB_Uns… TCGGCC… CASSQD… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>    TRBV4…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> TRB_Uns… CTCAGG… CASSYV… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   FALSE   <span style="color: #BB0000;">NA</span>    TRBV6…</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 390 more rows, 137 more variables: d_call &lt;chr&gt;, d2_call &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   j_call &lt;chr&gt;, c_call &lt;chr&gt;, sequence_alignment &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   sequence_alignment_aa &lt;chr&gt;, germline_alignment &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   germline_alignment_aa &lt;chr&gt;, junction &lt;chr&gt;, junction_aa &lt;chr&gt;, np1 &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   np1_aa &lt;chr&gt;, np2 &lt;chr&gt;, np2_aa &lt;chr&gt;, np3 &lt;chr&gt;, np3_aa &lt;chr&gt;, cdr1 &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   cdr1_aa &lt;chr&gt;, cdr2 &lt;chr&gt;, cdr2_aa &lt;chr&gt;, cdr3 &lt;chr&gt;, cdr3_aa &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   fwr1 &lt;chr&gt;, fwr1_aa &lt;chr&gt;, fwr2 &lt;chr&gt;, fwr2_aa &lt;chr&gt;, fwr3 &lt;chr&gt;, …</span></span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="extracting-productive-sequences">Extracting productive sequences<a class="anchor" aria-label="anchor" href="#extracting-productive-sequences"></a>
</h2>
<p>When AIRR-seq samples are derived from genomic DNA rather than
complimentary DNA made from RNA, then you will find productive and
unproductive sequences. Productive sequences are defined as in-frame
sequences without any early stop codons. To filter out these productive
sequences, you can use the <code>productiveSeq</code> to remove
unproductive sequences and recompute the duplicate_frequency to reflect
the productive amino acid or nucleotide sequence frequencies.</p>
<p>If you are interested in just the complementarity determining region
3 (CDR3) amino acid sequences, then set aggregate to
<code>junction_aa</code> and the <code>duplicate_count</code> for
duplicate amino acid sequences will be summed. The resulting tibble will
have <code>junction_aa</code>, <code>duplicate_count</code>,
<code>duplicate_frequency</code>, <code>reading_frame</code>, and the
most frequent VDJ gene combinations for each of the duplicated amino
acid sequences and the corresponding gene family names. These gene names
are only kept for consistency of the tibble structure, but since a
single amino acid sequence can be generated from different VDJ
combinations, it is inadvisable to use these values for downstream
analysis</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">aa_table</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/productiveSeq.html">productiveSeq</a></span><span class="op">(</span>study_table <span class="op">=</span> <span class="va">study_table</span>, aggregate <span class="op">=</span> <span class="st">"junction_aa"</span>, prevalence <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">aa_table</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 810 × 11</span></span></span>
<span><span class="co">#&gt;    repert…¹ junct…² v_call d_call j_call v_fam…³ d_fam…⁴ j_fam…⁵ readi…⁶ dupli…⁷</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> TRB_CD4… CAISVG… TRBV1… TRBD2… TRBJ1… TRBV10  TRBD2   TRBJ1   in-fra…      10</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> TRB_CD4… CASDGG… TRBV1… TRBD2… TRBJ1… TRBV19  TRBD2   TRBJ1   in-fra…     458</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> TRB_CD4… CASGGL… <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>     TRBJ1… <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>      TRBJ1   in-fra…      23</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> TRB_CD4… CASGLV… TRBV1… TRBD2… TRBJ2… TRBV12  TRBD2   TRBJ2   in-fra…      29</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> TRB_CD4… CASGTG… TRBV6… TRBD2… TRBJ2… TRBV6   TRBD2   TRBJ2   in-fra…      21</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> TRB_CD4… CASHSS… TRBV6… <span style="color: #BB0000;">NA</span>     TRBJ1… TRBV6   <span style="color: #BB0000;">NA</span>      TRBJ1   in-fra…      11</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> TRB_CD4… CASKPP… TRBV6… TRBD1… TRBJ1… TRBV6   TRBD1   TRBJ1   in-fra…     119</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> TRB_CD4… CASMID… TRBV5… <span style="color: #BB0000;">NA</span>     TRBJ1… TRBV5   <span style="color: #BB0000;">NA</span>      TRBJ1   in-fra…      11</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> TRB_CD4… CASNAR… TRBV6… TRBD1… TRBJ1… TRBV6   TRBD1   TRBJ1   in-fra…      12</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> TRB_CD4… CASRLG… <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>     TRBJ1… <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>      TRBJ1   in-fra…      24</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 800 more rows, 1 more variable: duplicate_frequency &lt;dbl&gt;, and</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   abbreviated variable names ¹​repertoire_id, ²​junction_aa, ³​v_family,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ⁴​d_family, ⁵​j_family, ⁶​reading_frame, ⁷​duplicate_count</span></span></span></code></pre></div>
<p>Alternatively you can aggregate by <code>junction</code> to group
sequences by CDR3 nucleotide sequences. This option will produce a
tibble similar to the output of <code>readImmunoSeq</code>. Many of the
functions within LymphoSeq2 use the results from
<code>productiveSeq</code> function. Please be sure to check the
function documentation.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nuc_table</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/productiveSeq.html">productiveSeq</a></span><span class="op">(</span>study_table <span class="op">=</span> <span class="va">study_table</span>, aggregate <span class="op">=</span> <span class="st">"junction"</span>, </span>
<span>                          prevalence <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">nuc_table</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 812 × 12</span></span></span>
<span><span class="co">#&gt;    repert…¹ junct…² junct…³ v_call d_call j_call v_fam…⁴ d_fam…⁵ j_fam…⁶ readi…⁷</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> TRB_CD4… AACCTG… CASSVE… TRBV9… <span style="color: #BB0000;">NA</span>     TRBJ1… TRBV9   <span style="color: #BB0000;">NA</span>      TRBJ1   in-fra…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> TRB_CD4… AACCTG… CASSVM… TRBV9… <span style="color: #BB0000;">NA</span>     TRBJ1… TRBV9   <span style="color: #BB0000;">NA</span>      TRBJ1   in-fra…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> TRB_CD4… AACGCC… CASSDS… TRBV5… TRBD1… TRBJ1… TRBV5   TRBD1   TRBJ1   in-fra…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> TRB_CD4… AACGCC… CASSSQ… TRBV5… TRBD1… TRBJ2… TRBV5   TRBD1   TRBJ2   in-fra…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> TRB_CD4… AACGCC… CASSLT… TRBV5… <span style="color: #BB0000;">NA</span>     TRBJ2… TRBV5   <span style="color: #BB0000;">NA</span>      TRBJ2   in-fra…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> TRB_CD4… AAGATC… CASSSN… <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>     TRBJ1… <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>      TRBJ1   in-fra…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> TRB_CD4… AATCTT… CASSQG… <span style="color: #BB0000;">NA</span>     TRBD2… TRBJ1… <span style="color: #BB0000;">NA</span>      TRBD2   TRBJ1   in-fra…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> TRB_CD4… AATGTG… CASSLA… TRBV5… TRBD2… TRBJ1… TRBV5   TRBD2   TRBJ1   in-fra…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> TRB_CD4… AATTCC… CASSQP… <span style="color: #BB0000;">NA</span>     TRBD1… TRBJ1… <span style="color: #BB0000;">NA</span>      TRBD1   TRBJ1   in-fra…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> TRB_CD4… AATTCC… CASSQG… <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>     TRBJ1… <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>      TRBJ1   in-fra…</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 802 more rows, 2 more variables: duplicate_count &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   duplicate_frequency &lt;dbl&gt;, and abbreviated variable names ¹​repertoire_id,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ²​junction, ³​junction_aa, ⁴​v_family, ⁵​d_family, ⁶​j_family, ⁷​reading_frame</span></span></span></code></pre></div>
<p>If the parameter prevalence is set to TRUE, then a new column is
added to each of the data frames giving the prevalence of each TCR beta
CDR3 amino acid sequence in 55 healthy donor peripheral blood samples.
Values range from 0 to 100 percent where 100 percent means the sequence
appeared in the blood of all 55 individuals.</p>
<p>Notice in the example below that there are no amino acid sequences
given in the first and fourth row of the study_table table for sample
“TRB_Unsorted_949”. This is because the nucleotide sequence is out of
frame and does not produce a productively transcribed amino acid
sequence. If an asterisk (*) appears in the amino acid sequences, this
would indicate an early stop codon.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">study_table</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">repertoire_id</span> <span class="op">==</span> <span class="st">"TRB_Unsorted_0"</span><span class="op">)</span> </span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 100 × 147</span></span></span>
<span><span class="co">#&gt;    sequen…¹ seque…² seque…³ rev_c…⁴ produ…⁵ vj_in…⁶ stop_…⁷ compl…⁸ locus v_call</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> TRB_Uns… TCAATT… <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>      TRUE    FALSE   <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> TRB_Uns… CTGATT… CASSPV… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>    TRBV2…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> TRB_Uns… ATCAAT… CASSQE… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   FALSE   <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> TRB_Uns… CACACC… CASSQE… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>    TRBV4…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> TRB_Uns… TGCCAT… <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>      TRUE    FALSE   <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> TRB_Uns… GCCAGC… CASSLE… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>    TRBV2…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> TRB_Uns… CCCCTG… CASSPG… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>    TRBV2…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> TRB_Uns… AGTGCC… CSARSP… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   FALSE   <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> TRB_Uns… GGAGCT… <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>      TRUE    FALSE   <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> TRB_Uns… CTGTAG… CASSEK… FALSE   FALSE   <span style="color: #BB0000;">NA</span>      FALSE   TRUE    <span style="color: #BB0000;">NA</span>    TRBV1…</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 90 more rows, 137 more variables: d_call &lt;chr&gt;, d2_call &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   j_call &lt;chr&gt;, c_call &lt;chr&gt;, sequence_alignment &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   sequence_alignment_aa &lt;chr&gt;, germline_alignment &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   germline_alignment_aa &lt;chr&gt;, junction &lt;chr&gt;, junction_aa &lt;chr&gt;, np1 &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   np1_aa &lt;chr&gt;, np2 &lt;chr&gt;, np2_aa &lt;chr&gt;, np3 &lt;chr&gt;, np3_aa &lt;chr&gt;, cdr1 &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   cdr1_aa &lt;chr&gt;, cdr2 &lt;chr&gt;, cdr2_aa &lt;chr&gt;, cdr3 &lt;chr&gt;, cdr3_aa &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   fwr1 &lt;chr&gt;, fwr1_aa &lt;chr&gt;, fwr2 &lt;chr&gt;, fwr2_aa &lt;chr&gt;, fwr3 &lt;chr&gt;, …</span></span></span></code></pre></div>
<p>After <code>productiveSeq</code> is run, the unproductive sequences
are removed and the <code>duplicate_frequency</code> is recalculated for
each sequence. If there were two identical amino acid sequences that
differed in their nucleotide sequence, they would be combined and their
counts added together.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">aa_table</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">repertoire_id</span> <span class="op">==</span> <span class="st">"TRB_Unsorted_0"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 83 × 11</span></span></span>
<span><span class="co">#&gt;    repert…¹ junct…² v_call d_call j_call v_fam…³ d_fam…⁴ j_fam…⁵ readi…⁶ dupli…⁷</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> TRB_Uns… CAISDL… TRBV1… TRBD2… TRBJ2… TRBV10  TRBD2   TRBJ2   in-fra…      59</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> TRB_Uns… CARPPY… TRBV1… <span style="color: #BB0000;">NA</span>     TRBJ1… TRBV10  <span style="color: #BB0000;">NA</span>      TRBJ1   in-fra…      21</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> TRB_Uns… CASKYG… TRBV7… TRBD2… TRBJ1… TRBV7   TRBD2   TRBJ1   in-fra…      55</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> TRB_Uns… CASREA… TRBV2… TRBD1… TRBJ1… TRBV2   TRBD1   TRBJ1   in-fra…     137</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> TRB_Uns… CASRHR… TRBV2… <span style="color: #BB0000;">NA</span>     TRBJ1… TRBV28  <span style="color: #BB0000;">NA</span>      TRBJ1   in-fra…      22</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> TRB_Uns… CASRPD… TRBV2… TRBD1… TRBJ1… TRBV28  TRBD1   TRBJ1   in-fra…      21</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> TRB_Uns… CASRPG… TRBV1… TRBD1… TRBJ2… TRBV10  TRBD1   TRBJ2   in-fra…      57</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> TRB_Uns… CASRPT… TRBV1… <span style="color: #BB0000;">NA</span>     TRBJ2… TRBV19  <span style="color: #BB0000;">NA</span>      TRBJ2   in-fra…     181</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> TRB_Uns… CASRSG… TRBV2… TRBD2… TRBJ1… TRBV2   TRBD2   TRBJ1   in-fra…      32</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> TRB_Uns… CASSAR… TRBV7… <span style="color: #BB0000;">NA</span>     TRBJ2… TRBV7   <span style="color: #BB0000;">NA</span>      TRBJ2   in-fra…      46</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 73 more rows, 1 more variable: duplicate_frequency &lt;dbl&gt;, and</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   abbreviated variable names ¹​repertoire_id, ²​junction_aa, ³​v_family,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ⁴​d_family, ⁵​j_family, ⁶​reading_frame, ⁷​duplicate_count</span></span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="create-a-table-of-summary-statistics">Create a table of summary statistics<a class="anchor" aria-label="anchor" href="#create-a-table-of-summary-statistics"></a>
</h2>
<p>To create a table summarizing the total number of sequences, number
of unique productive sequences, number of genomes, clonality, Gini
coefficient, and the frequency (%) of the top productive sequence,
Simpson index, Inverse Simpson index, Hill diversity index, Chao1 index
and Kemp index in each imported file, use the function
<code>clonality</code>.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/clonality.html">clonality</a></span><span class="op">(</span>study_table <span class="op">=</span> <span class="va">study_table</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 10 × 8</span></span></span>
<span><span class="co">#&gt;    repertoire_id     total_seq…¹ uniqu…² total…³ clona…⁴ gini_…⁵ top_p…⁶ conve…⁷</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> TRB_CD4_949               100      80   <span style="text-decoration: underline;">23</span>093   0.349   0.791   34.9     1   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> TRB_CD8_949               100      81   <span style="text-decoration: underline;">23</span>072   0.292   0.731   21.8     1   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> TRB_CD8_CMV_369           100      78    <span style="text-decoration: underline;">1</span>456   0.305   0.746   19.3     1.03</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> TRB_Unsorted_0            100      83   <span style="text-decoration: underline;">14</span>776   0.128   0.563    7.20    1   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> TRB_Unsorted_1320         100      83  <span style="text-decoration: underline;">157</span>660   0.279   0.744   16.5     1   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> TRB_Unsorted_1496         100      82   <span style="text-decoration: underline;">28</span>876   0.260   0.721   16.7     1   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> TRB_Unsorted_32           100      82   <span style="text-decoration: underline;">17</span>043   0.105   0.475    9.16    1   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> TRB_Unsorted_369          100      80  <span style="text-decoration: underline;">274</span>812   0.387   0.785   36.6     1   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> TRB_Unsorted_83           100      81  <span style="text-decoration: underline;">170</span>526   0.328   0.729   32.2     1   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> TRB_Unsorted_949          100      82    <span style="text-decoration: underline;">4</span>971   0.247   0.682   18.4     1   </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with abbreviated variable names ¹​total_sequences,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ²​unique_productive_sequences, ³​total_count, ⁴​clonality, ⁵​gini_coefficient,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ⁶​top_productive_sequence, ⁷​convergence</span></span></span></code></pre></div>
<p>The clonality score is derived from the Shannon entropy, which is
calculated from the frequencies of all productive sequences divided by
the logarithm of the total number of unique productive sequences. This
normalized entropy value is then inverted (1 - normalized entropy) to
produce the clonality metric.</p>
<p>The Gini coefficient, Chao1 estimate, Kemp estimate, Hill estimate,
Simpson index and Inverse Simpson index are alternative metric to
measure sequence diversity within the immune repertoire.</p>
<p>The Gini coefficient is an alternative metric used to calculate
repertoire diversity and is derived from the Lorenz curve. The Lorenz
curve is drawn such that x-axis represents the cumulative percentage of
unique sequences and the y-axis represents the cumulative percentage of
reads. A line passing through the origin with a slope of 1 reflects
equal frequencies of all clones. The Gini coefficient is the ratio of
the area between the line of equality and the observed Lorenz curve over
the total area under the line of equality.</p>
</div>
<div class="section level2">
<h2 id="calculate-clonal-relatedness">Calculate clonal relatedness<a class="anchor" aria-label="anchor" href="#calculate-clonal-relatedness"></a>
</h2>
<p>One of the drawbacks of the clonality metric is that it does not take
into account sequence similarity. This is particularly important when
studying affinity maturation or B cell malignancies(Lombardo, K.A., et
al. Blood Advances 2017 1:535-544). Clonal relatedness is a useful
metric that takes into account sequence similarity without regard for
clonal frequency. It is defined as the proportion of nucleotide
sequences that are related by a defined edit distance threshold. The
value ranges from 0 to 1 where 0 indicates no sequences are related and
1 indicates all sequences are related. Edit distance is a way of
quantifying how dissimilar two sequences are to one another by counting
the minimum number of operations required to transform one sequence into
the other. For example, an edit distance of 0 means the sequences are
identical and an edit distance of 1 indicates that the sequences
different by a single amino acid or nucleotide.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">IGH_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"IGH_sequencing"</span>, package <span class="op">=</span> <span class="st">"LymphoSeq2"</span><span class="op">)</span> </span>
<span><span class="va">IGH_table</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/readImmunoSeq.html">readImmunoSeq</a></span><span class="op">(</span>path <span class="op">=</span> <span class="va">IGH_path</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/topSeqs.html">topSeqs</a></span><span class="op">(</span>top <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></span>
<span><span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/clonalRelatedness.html">clonalRelatedness</a></span><span class="op">(</span>study_table <span class="op">=</span> <span class="va">IGH_table</span>, edit_distance <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 10 × 2</span></span></span>
<span><span class="co">#&gt;    repertoire_id     relatedness</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> IGH_MVQ108911A_BL        0.61</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> IGH_MVQ194745A_BL        0.7 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> IGH_MVQ81231A_BL         0.61</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> IGH_MVQ89037A_BL         0.35</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> IGH_MVQ90143A_BL         0.03</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> IGH_MVQ92552A_BL         0.08</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> IGH_MVQ93505A_BL         0.31</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> IGH_MVQ93631A_BL         0.83</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> IGH_MVQ94865A_BL         0.06</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> IGH_MVQ95413A_BL         0.01</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="draw-a-phylogenetic-tree">Draw a phylogenetic tree<a class="anchor" aria-label="anchor" href="#draw-a-phylogenetic-tree"></a>
</h2>
<p>A phylogenetic tree is a useful way to visualize the similarity
between sequences. The <code>phyloTree</code> function create a
phylogenetic tree of a single sample using neighbor joining tree
estimation for amino acid or nucleotide CDR3 sequences. Each leaf in the
tree represents a sequence color coded by the V, D, and J gene usage.
The number next to each leaf refers to the sequence count. A triangle
shaped leaf indicates the most frequent sequence. The distance between
leaves on the horizontal axis corresponds to the sequence similarity
(i.e. the further apart the leaves are horizontally, the less similar
the sequences are to one another).</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>nuc_IGH_table <span class="ot">&lt;-</span> LymphoSeq2<span class="sc">::</span><span class="fu">productiveSeq</span>(<span class="at">study_table =</span> IGH_table, <span class="at">aggregate =</span> <span class="st">"junction"</span>)</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>LymphoSeq2<span class="sc">::</span><span class="fu">phyloTree</span>(<span class="at">study_table =</span> nuc_IGH_table, </span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>                      <span class="at">repertoire_ids =</span> <span class="st">"IGH_MVQ92552A_BL"</span>, </span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>                      <span class="at">type =</span> <span class="st">"junction"</span>, </span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>                      <span class="at">layout =</span> <span class="st">"rectangular"</span>)</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>Warning<span class="sc">:</span>  [1m [22mThe <span class="st">`</span><span class="at">&lt;scale&gt;</span><span class="st">`</span> argument of <span class="st">`</span><span class="at">guides()</span><span class="st">`</span> cannot be <span class="st">`</span><span class="at">FALSE</span><span class="st">`</span>. Use <span class="st">"none"</span> instead as</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>of ggplot2 <span class="dv">3</span>.<span class="dv">3</span>.<span class="fl">4.</span></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a> [36mℹ [39m The deprecated feature was likely used <span class="cf">in</span> the  [34mLymphoSeq2 [39m package.</span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>  Please report the issue at</span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>   [3m [34m<span class="sc">&lt;</span>https<span class="sc">:</span><span class="er">//</span>github.com<span class="sc">/</span>shashidhar22<span class="sc">/</span>LymphoSeq2<span class="sc">/</span>issues<span class="sc">&gt;</span> [39m [23m.</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a> [90mThis warning is displayed once every <span class="dv">8</span> hours. [39m</span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a> [90mCall <span class="st">`</span><span class="at">lifecycle::last_lifecycle_warnings()</span><span class="st">`</span> to see where this warning was [39m</span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a> [90mgenerated. [39m</span></code></pre></div>
<p><img src="LymphoSeq2_files/figure-html/getIGHProd-1.png" width="672"></p>
</div>
<div class="section level2">
<h2 id="multiple-sequence-alignment">Multiple sequence alignment<a class="anchor" aria-label="anchor" href="#multiple-sequence-alignment"></a>
</h2>
<p>In LymphoSeq2, you can perform a multiple sequence alignment using
one of three methods provided by the Bioconductor msa package (ClustalW,
ClustalOmega, or Muscle), the change in functionality however is, now
the function returns a <code>msa</code> S4 object. One may perform the
alignment of all amino acid or nucleotide sequences in a single sample.
Alternatively, one may search for a given sequence within a list of
samples using an edit distance threshold.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">alignment</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/alignSeq.html">alignSeq</a></span><span class="op">(</span>study_table <span class="op">=</span> <span class="va">nuc_IGH_table</span>, </span>
<span>                                  repertoire_ids <span class="op">=</span> <span class="st">"IGH_MVQ92552A_BL"</span>, </span>
<span>                                  type <span class="op">=</span> <span class="st">"junction_aa"</span>, </span>
<span>                                  method <span class="op">=</span> <span class="st">"ClustalW"</span><span class="op">)</span></span></code></pre></div>
<p>use default substitution matrix</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/plotAlignment.html">plotAlignment</a></span><span class="op">(</span><span class="va">alignment</span><span class="op">)</span></span>
<span><span class="co">#&gt; Coordinate system already present. Adding new coordinate system, which will</span></span>
<span><span class="co">#&gt; replace the existing one.</span></span></code></pre></div>
<p><img src="LymphoSeq2_files/figure-html/alignSeq-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="searching-for-sequences">Searching for sequences<a class="anchor" aria-label="anchor" href="#searching-for-sequences"></a>
</h2>
<p>To search for one or more amino acid or nucleotide CDR3 sequences in
a list of data frames, use the function <code>searchSeq</code>. The
function allows sequence search with a edit distance threshold. For
example, an edit distance of 0 means the sequences are identical and an
edit distance of 1 indicates that the sequences differ by a single amino
acid or nucleotide. Match options include “global” matching which
performs end-to-end matching of sequences. “partial” matching allows
searching for sub strings with CDR3 sequences.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/searchSeq.html">searchSeq</a></span><span class="op">(</span>study_table <span class="op">=</span> <span class="va">aa_table</span>, </span>
<span>                      sequence <span class="op">=</span> <span class="st">"CASSPVSNEQFF"</span>, </span>
<span>                      seq_type <span class="op">=</span> <span class="st">"junction_aa"</span>, </span>
<span>                      match <span class="op">=</span> <span class="st">"global"</span>, </span>
<span>                      edit_distance <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 1 × 13</span></span></span>
<span><span class="co">#&gt;   reperto…¹ junct…² v_call d_call j_call v_fam…³ d_fam…⁴ j_fam…⁵ readi…⁶ dupli…⁷</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> TRB_Unso… CASSPV… TRBV2… TRBD2… TRBJ2… TRBV28  TRBD2   TRBJ2   in-fra…     822</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 3 more variables: duplicate_frequency &lt;dbl&gt;, edit_distance &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   searchSequence &lt;chr&gt;, and abbreviated variable names ¹​repertoire_id,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ²​junction_aa, ³​v_family, ⁴​d_family, ⁵​j_family, ⁶​reading_frame,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ⁷​duplicate_count</span></span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="searching-for-published-sequences">Searching for published sequences<a class="anchor" aria-label="anchor" href="#searching-for-published-sequences"></a>
</h2>
<p>To search your entire list of data frames for a published amino acid
CDR3 TCRB sequence with known antigen specificity, use the function
<code>searchPublished</code>.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/searchPublished.html">searchPublished</a></span><span class="op">(</span>study_table <span class="op">=</span> <span class="va">aa_table</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">PMID</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 26 × 16</span></span></span>
<span><span class="co">#&gt;    repert…¹ junct…² v_call d_call j_call v_fam…³ d_fam…⁴ j_fam…⁵ readi…⁶ dupli…⁷</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> TRB_CD4… CASSQD… TRBV4… TRBD1… TRBJ2… TRBV4   TRBD1   TRBJ2   in-fra…      13</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> TRB_CD8… CASSPG… TRBV1… TRBD1… TRBJ1… TRBV10  TRBD1   TRBJ1   in-fra…      67</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> TRB_CD8… CASSPS… TRBV4… TRBD2… TRBJ1… TRBV4   TRBD2   TRBJ1   in-fra…      91</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> TRB_CD8… CASSYS… <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>     TRBJ1… <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>      TRBJ1   in-fra…      38</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> TRB_CD8… CASSPA… TRBV4… <span style="color: #BB0000;">NA</span>     TRBJ1… TRBV4   <span style="color: #BB0000;">NA</span>      TRBJ1   in-fra…      15</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> TRB_CD8… CASSPG… TRBV1… TRBD1… TRBJ1… TRBV10  TRBD1   TRBJ1   in-fra…       9</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> TRB_CD8… CASSPS… TRBV4… TRBD2… TRBJ1… TRBV4   TRBD2   TRBJ1   in-fra…      11</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> TRB_CD8… CASSYS… <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>     TRBJ1… <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>      TRBJ1   in-fra…       2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> TRB_Uns… CASSPQ… TRBV4… TRBD2… TRBJ1… TRBV4   TRBD2   TRBJ1   in-fra…      87</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> TRB_Uns… CASSLE… TRBV5… TRBD1… TRBJ1… TRBV5   TRBD1   TRBJ1   in-fra…     126</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 16 more rows, 6 more variables: duplicate_frequency &lt;dbl&gt;, PMID &lt;fct&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   HLA &lt;fct&gt;, antigen &lt;fct&gt;, epitope &lt;fct&gt;, prevalence &lt;dbl&gt;, and abbreviated</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   variable names ¹​repertoire_id, ²​junction_aa, ³​v_family, ⁴​d_family,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ⁵​j_family, ⁶​reading_frame, ⁷​duplicate_count</span></span></span></code></pre></div>
<p>For each found sequence, a table is provides listing the antigen,
epitope, HLA type, PubMed ID (PMID), and prevalence percentage of the
sequence among 55 healthy donor blood samples.</p>
<p>You can even search of productive CDR3 amino acid sequences from the
repertoires that are found in public databases such as VdjDB, IEDB, and
McPas-TCR using the function <code>searchDB</code>. By specifying
<code>dbname="all"</code> <code>searchDB</code> will look for each CDR3
amino acid sequence in the dataset in all three public databases. You
can also pass a vector with any of the three databases (“VdjDB”, “IEDB”,
“McPAS-TCR”) to search just those databases.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/searchDB.html">searchDB</a></span><span class="op">(</span>study_table <span class="op">=</span> <span class="va">aa_table</span>, dbname <span class="op">=</span> <span class="st">"all"</span>, chain <span class="op">=</span> <span class="st">"trb"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 839 × 26</span></span></span>
<span><span class="co">#&gt;    repert…¹ junct…² v_call d_call j_call v_fam…³ d_fam…⁴ j_fam…⁵ readi…⁶ dupli…⁷</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> TRB_CD4… CAISVG… TRBV1… TRBD2… TRBJ1… TRBV10  TRBD2   TRBJ1   in-fra…      10</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> TRB_CD4… CASDGG… TRBV1… TRBD2… TRBJ1… TRBV19  TRBD2   TRBJ1   in-fra…     458</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> TRB_CD4… CASGGL… <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>     TRBJ1… <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>      TRBJ1   in-fra…      23</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> TRB_CD4… CASGLV… TRBV1… TRBD2… TRBJ2… TRBV12  TRBD2   TRBJ2   in-fra…      29</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> TRB_CD4… CASGTG… TRBV6… TRBD2… TRBJ2… TRBV6   TRBD2   TRBJ2   in-fra…      21</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> TRB_CD4… CASHSS… TRBV6… <span style="color: #BB0000;">NA</span>     TRBJ1… TRBV6   <span style="color: #BB0000;">NA</span>      TRBJ1   in-fra…      11</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> TRB_CD4… CASKPP… TRBV6… TRBD1… TRBJ1… TRBV6   TRBD1   TRBJ1   in-fra…     119</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> TRB_CD4… CASMID… TRBV5… <span style="color: #BB0000;">NA</span>     TRBJ1… TRBV5   <span style="color: #BB0000;">NA</span>      TRBJ1   in-fra…      11</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> TRB_CD4… CASNAR… TRBV6… TRBD1… TRBJ1… TRBV6   TRBD1   TRBJ1   in-fra…      12</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> TRB_CD4… CASRLG… <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>     TRBJ1… <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>      TRBJ1   in-fra…      24</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 829 more rows, 16 more variables: duplicate_frequency &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   tra_cdr3_aa &lt;chr&gt;, gene &lt;chr&gt;, epitope &lt;chr&gt;, pathology &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   antigen &lt;chr&gt;, tra_v_call &lt;chr&gt;, tra_j_call &lt;chr&gt;, mhc_allele &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   reference &lt;chr&gt;, score &lt;dbl&gt;, cell_type &lt;chr&gt;, source &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   trb_v_call &lt;chr&gt;, trb_j_call &lt;chr&gt;, Species &lt;chr&gt;, and abbreviated variable</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   names ¹​repertoire_id, ²​junction_aa, ³​v_family, ⁴​d_family, ⁵​j_family,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ⁶​reading_frame, ⁷​duplicate_count</span></span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="visualizing-repertoire-diversity">Visualizing repertoire diversity<a class="anchor" aria-label="anchor" href="#visualizing-repertoire-diversity"></a>
</h2>
<p>Antigen receptor repertoire diversity can be characterized by a
number such as clonality or Gini coefficient calculated by the
<code>clonality</code> function. Alternatively, you can visualize the
repertoire diversity by plotting the Lorenz curve for each sample as
defined above. In this plot, the more diverse samples will appear near
the dotted diagonal line (the line of equality) whereas the more clonal
samples will appear to have a more bowed shape.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">samples</span> <span class="op">&lt;-</span> <span class="va">aa_table</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>           <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">repertoire_id</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/lorenzCurve.html">lorenzCurve</a></span><span class="op">(</span>repertoire_ids <span class="op">=</span> <span class="va">samples</span>, study_table <span class="op">=</span> <span class="va">aa_table</span><span class="op">)</span></span></code></pre></div>
<p><img src="LymphoSeq2_files/figure-html/plotLorenz-1.png" width="700"></p>
<p>Alternatively, you can get a feel for the repertoire diversity by
plotting the cumulative frequency of a selected number of the top most
frequent clones using the function <code>topSeqsPlot</code>. In this
case, each of the top sequences are represented by a different color and
all less frequent clones will be assigned a single color (violet).</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/topSeqsPlot.html">topSeqsPlot</a></span><span class="op">(</span>study_table <span class="op">=</span> <span class="va">aa_table</span>, top <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<p><img src="LymphoSeq2_files/figure-html/topSeqPlot-1.png" width="700"></p>
<p>Both of these functions are built using the ggplot2 package. You can
reformat the plot using ggplot2 functions. Please refer to the
<code>lorenzCurve</code> and <code>topSeqsPlot</code> manual for
specific examples.</p>
</div>
<div class="section level2">
<h2 id="comparing-samples">Comparing samples<a class="anchor" aria-label="anchor" href="#comparing-samples"></a>
</h2>
<p>To compare the T or B cell repertoires of all samples in a pairwise
fashion, use the <code>bhattacharyyaMatrix</code> or
<code>similarityMatrix</code> functions. Both the Bhattacharyya
coefficient and similarity score are measures of the amount of overlap
between two samples. The value for each ranges from 0 to 1 where 1
indicates the sequence frequencies are identical in the two samples and
0 indicates no shared frequencies. The Bhattacharyya coefficient differs
from the similarity score in that it involves weighting each shared
sequence in the two distributions by the arithmetic mean of the
frequency of each sequence, while calculating the similarity scores
involves weighting each shared sequence in the two distributions by the
geometric mean of the frequency of each sequence in the two
distributions.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bhattacharyya_matrix</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/scoringMatrix.html">scoringMatrix</a></span><span class="op">(</span><span class="va">aa_table</span>, mode <span class="op">=</span> <span class="st">"Bhattacharyya"</span><span class="op">)</span></span>
<span><span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/pairwisePlot.html">pairwisePlot</a></span><span class="op">(</span><span class="va">bhattacharyya_matrix</span><span class="op">)</span></span></code></pre></div>
<p><img src="LymphoSeq2_files/figure-html/bhattacharyya-1.png" width="700"></p>
<p>To view sequences shared between two or more samples, use the
function <code>commonSeqs</code>. This function requires that a
productive amino acid list be specified.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">common</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/commonSeqs.html">commonSeqs</a></span><span class="op">(</span>study_table <span class="op">=</span> <span class="va">aa_table</span>, </span>
<span>                                 repertoire_ids <span class="op">=</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TRB_Unsorted_0"</span>, <span class="st">"TRB_Unsorted_32"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">common</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 1 × 3</span></span></span>
<span><span class="co">#&gt;   junction_aa     TRB_Unsorted_0 TRB_Unsorted_32</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> CASSQDRTGQYGYTF        0.004<span style="text-decoration: underline;">29</span>          0.015<span style="text-decoration: underline;">2</span></span></span></code></pre></div>
<p>To visualize the number of overlapping sequences between two or three
samples in the form of a Venn diagram, use the function
<code>commonSeqVenn</code></p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/commonSeqsVenn.html">commonSeqsVenn</a></span><span class="op">(</span>repertoire_ids <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TRB_Unsorted_32"</span>, <span class="st">"TRB_Unsorted_83"</span><span class="op">)</span>, </span>
<span>                           amino_table <span class="op">=</span> <span class="va">aa_table</span><span class="op">)</span></span></code></pre></div>
<p><img src="LymphoSeq2_files/figure-html/commonVenn-1.png" width="480"></p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/commonSeqsVenn.html">commonSeqsVenn</a></span><span class="op">(</span>repertoire_ids <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TRB_Unsorted_0"</span>, <span class="st">"TRB_Unsorted_32"</span>, <span class="st">"TRB_Unsorted_83"</span><span class="op">)</span>, </span>
<span>                           amino_table <span class="op">=</span> <span class="va">aa_table</span><span class="op">)</span></span></code></pre></div>
<p><img src="LymphoSeq2_files/figure-html/commonVennT-1.png" width="480"></p>
<p>To compare the frequency of sequences between two samples as a
scatter plot, use the function <code>commonSeqsPlot</code>.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/commonSeqsPlot.html">commonSeqsPlot</a></span><span class="op">(</span><span class="st">"TRB_Unsorted_32"</span>, <span class="st">"TRB_Unsorted_83"</span>, </span>
<span>                           amino_table <span class="op">=</span> <span class="va">aa_table</span>, show <span class="op">=</span> <span class="st">"common"</span><span class="op">)</span></span></code></pre></div>
<p><img src="LymphoSeq2_files/figure-html/commonSeqPlot-1.png" width="700"></p>
<p>If you have more than 3 samples to compare, use the
<code>commonSeqBar</code> function. You can chose to color a single
sample with the <code>color.sample</code> argument or a desired
intersection with the <code>color.intersection</code> argument.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/commonSeqsBar.html">commonSeqsBar</a></span><span class="op">(</span>amino_table <span class="op">=</span> <span class="va">aa_table</span>, </span>
<span>                          repertoire_ids <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TRB_CD4_949"</span>, <span class="st">"TRB_CD8_949"</span>, </span>
<span>                                             <span class="st">"TRB_Unsorted_949"</span>, <span class="st">"TRB_Unsorted_1320"</span><span class="op">)</span>, </span>
<span>                          color_sample <span class="op">=</span> <span class="st">"TRB_CD8_949"</span>,</span>
<span>                          labels <span class="op">=</span> <span class="st">"no"</span><span class="op">)</span></span></code></pre></div>
<p><img src="LymphoSeq2_files/figure-html/commonBar-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="differential-abundance">Differential abundance<a class="anchor" aria-label="anchor" href="#differential-abundance"></a>
</h2>
<p>When comparing a sample from two different time points, it is useful
to identify sequences that are significantly more or less abundant in
one versus the other time point (DeWitt, W.S., et al. Journal of
Virology 2015 89(8):4517-4526). The <code>differentialAbundance</code>
function uses a Fisher exact test to calculate differential abundance of
each sequence in two time points and reports the log2 transformed fold
change, P value and adjusted P value.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/differentialAbundance.html">differentialAbundance</a></span><span class="op">(</span>study_table <span class="op">=</span> <span class="va">aa_table</span>, </span>
<span>                                  repertoire_ids <span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TRB_Unsorted_949"</span>, </span>
<span>                                                    <span class="st">"TRB_Unsorted_1320"</span><span class="op">)</span>, </span>
<span>                                  type <span class="op">=</span> <span class="st">"junction_aa"</span>, q <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 107 × 6</span></span></span>
<span><span class="co">#&gt;    junction_aa     TRB_Unsorted_949 TRB_Unsorted_1320        p        q    l2fc</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> CAIKMETPNGEQYF                29               326 1.14<span style="color: #949494;">e</span><span style="color: #BB0000;">- 6</span> 1.14<span style="color: #949494;">e</span><span style="color: #BB0000;">- 6</span>   -<span style="color: #BB0000;">3.49</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> CAISEGQGVKPQHF                 0               167 1.09<span style="color: #949494;">e</span><span style="color: #BB0000;">- 2</span> 1.09<span style="color: #949494;">e</span><span style="color: #BB0000;">- 2</span> -<span style="color: #BB0000;">Inf</span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> CAISESGVLNEKLFF               13               150 1.20<span style="color: #949494;">e</span><span style="color: #BB0000;">- 3</span> 1.20<span style="color: #949494;">e</span><span style="color: #BB0000;">- 3</span>   -<span style="color: #BB0000;">3.53</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> CASDGGFRNTIYF                 17               387 1.40<span style="color: #949494;">e</span><span style="color: #BB0000;">- 1</span> 1.40<span style="color: #949494;">e</span><span style="color: #BB0000;">- 1</span>   -<span style="color: #BB0000;">4.51</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> CASKPPGQGGYGYTF                0               173 1.12<span style="color: #949494;">e</span><span style="color: #BB0000;">- 2</span> 1.12<span style="color: #949494;">e</span><span style="color: #BB0000;">- 2</span> -<span style="color: #BB0000;">Inf</span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> CASNRVPEETQYF                  0               127 5.75<span style="color: #949494;">e</span><span style="color: #BB0000;">- 2</span> 5.75<span style="color: #949494;">e</span><span style="color: #BB0000;">- 2</span> -<span style="color: #BB0000;">Inf</span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> CASNSKADSTDTQYF               21              <span style="text-decoration: underline;">1</span>325 1.52<span style="color: #949494;">e</span><span style="color: #BB0000;">- 3</span> 1.52<span style="color: #949494;">e</span><span style="color: #BB0000;">- 3</span>   -<span style="color: #BB0000;">5.98</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> CASRDGQGSGNTIYF               48               358 6.73<span style="color: #949494;">e</span><span style="color: #BB0000;">-16</span> 6.73<span style="color: #949494;">e</span><span style="color: #BB0000;">-16</span>   -<span style="color: #BB0000;">2.90</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> CASREDRGSSPLHF                 0               147 2.45<span style="color: #949494;">e</span><span style="color: #BB0000;">- 2</span> 2.45<span style="color: #949494;">e</span><span style="color: #BB0000;">- 2</span> -<span style="color: #BB0000;">Inf</span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> CASRLGPGAGDEAFF               12               619 1.26<span style="color: #949494;">e</span><span style="color: #BB0000;">- 1</span> 1.26<span style="color: #949494;">e</span><span style="color: #BB0000;">- 1</span>   -<span style="color: #BB0000;">5.69</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 97 more rows</span></span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="finding-recurring-sequences">Finding recurring sequences<a class="anchor" aria-label="anchor" href="#finding-recurring-sequences"></a>
</h2>
<p>To create a tibble of unique, productive amino acid sequences as rows
and sample names as headers use the <code>seqMatrix</code> function.
Each value in the data frame represents the frequency that each sequence
appears in the sample. You can specify your own list of sequences or all
unique sequences in the list using the output of the function
<code>uniqueSeqs.</code> The <code>uniqueSeqs</code> function creates a
tibble of all unique, productive sequences and reports the total count
in all samples.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">unique_seqs</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/uniqueSeqs.html">uniqueSeqs</a></span><span class="op">(</span>productive_table <span class="op">=</span> <span class="va">aa_table</span><span class="op">)</span></span>
<span><span class="va">unique_seqs</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 438 × 2</span></span></span>
<span><span class="co">#&gt;    junction_aa            duplicate_count</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                            <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> CASSQDWERLGEQFF                  <span style="text-decoration: underline;">99</span>480</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> CASSLQGREKLFF                    <span style="text-decoration: underline;">90</span>563</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> CASSQDLMTVDSLFAGANVLTF           <span style="text-decoration: underline;">68</span>679</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> CASSPAGAYYNEQFF                  <span style="text-decoration: underline;">30</span>418</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> CASSPPTGERDTQYF                  <span style="text-decoration: underline;">24</span>552</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> CASSLAGDSQETQYF                  <span style="text-decoration: underline;">22</span>147</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> CASSESAGSTGELFF                  <span style="text-decoration: underline;">17</span>438</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> CASRDGQGSGNTIYF                  <span style="text-decoration: underline;">11</span>516</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> CASSPSRNTEAFF                     <span style="text-decoration: underline;">8</span>705</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> CASSQDRTGQYGYTF                   <span style="text-decoration: underline;">8</span>017</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 428 more rows</span></span></span></code></pre></div>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sequence_matrix</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/seqMatrix.html">seqMatrix</a></span><span class="op">(</span>amino_table <span class="op">=</span> <span class="va">aa_table</span>, sequences <span class="op">=</span> <span class="va">unique_seqs</span><span class="op">$</span><span class="va">junction_aa</span><span class="op">)</span></span>
<span><span class="va">sequence_matrix</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 438 × 11</span></span></span>
<span><span class="co">#&gt;    junction_aa   TRB_C…¹ TRB_C…² TRB_C…³ TRB_U…⁴ TRB_U…⁵ TRB_U…⁶ TRB_U…⁷ TRB_U…⁸</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> CAISVGGSSPLHF 6.95<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span>       0       0       0 0       0             0       0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> CASDGGFRNTIYF 3.18<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span>       0       0       0 0.003<span style="text-decoration: underline;">01</span> 0.004<span style="text-decoration: underline;">63</span>       0       0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> CASGGLNTEAFF  1.60<span style="color: #949494;">e</span><span style="color: #BB0000;">-3</span>       0       0       0 0       0             0       0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> CASGLVAGSTLG… 2.02<span style="color: #949494;">e</span><span style="color: #BB0000;">-3</span>       0       0       0 0       0             0       0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> CASGTGGETQYF  1.46<span style="color: #949494;">e</span><span style="color: #BB0000;">-3</span>       0       0       0 0       0             0       0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> CASHSSGNTIYF  7.65<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span>       0       0       0 0       0             0       0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> CASKPPGQGGYG… 8.27<span style="color: #949494;">e</span><span style="color: #BB0000;">-3</span>       0       0       0 0.001<span style="text-decoration: underline;">35</span> 0             0       0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> CASMIDPSGNTI… 7.65<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span>       0       0       0 0       0             0       0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> CASNARVDSPLHF 8.34<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span>       0       0       0 0       0             0       0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> CASRLGESPLHF  1.67<span style="color: #949494;">e</span><span style="color: #BB0000;">-3</span>       0       0       0 0       0             0       0</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 428 more rows, 2 more variables: TRB_Unsorted_83 &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   TRB_Unsorted_949 &lt;dbl&gt;, and abbreviated variable names ¹​TRB_CD4_949,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ²​TRB_CD8_949, ³​TRB_CD8_CMV_369, ⁴​TRB_Unsorted_0, ⁵​TRB_Unsorted_1320,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ⁶​TRB_Unsorted_1496, ⁷​TRB_Unsorted_32, ⁸​TRB_Unsorted_369</span></span></span></code></pre></div>
<p>If just the top clones with a frequency greater than a specified
amount are of interest to you, then use the <code>topFreq</code>
function. This creates a tibble of the top productive amino acid
sequences having a minimum specified frequency and reports the minimum,
maximum, and mean frequency that the sequence appears in a list of
samples. For TCRB sequences, the prevalence percentage and the published
antigen specificity of that sequence are also provided.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">top_freq</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/topFreq.html">topFreq</a></span><span class="op">(</span>productive_table <span class="op">=</span> <span class="va">aa_table</span>, frequency <span class="op">=</span> <span class="fl">0.001</span><span class="op">)</span></span>
<span><span class="va">top_freq</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 425 × 7</span></span></span>
<span><span class="co">#&gt;    junction_aa            minFrequency maxFreq…¹ meanF…² numbe…³ preva…⁴ antigen</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> CASSQDRTGQYGYTF             0.004<span style="text-decoration: underline;">29</span>    0.024<span style="text-decoration: underline;">8</span> 0.010<span style="text-decoration: underline;">6</span>        9       0 <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> CASSLQGREKLFF               0.056<span style="text-decoration: underline;">9</span>     0.322  0.113         8       0 <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> CASSQDLMTVDSLFAGANVLTF      0.029<span style="text-decoration: underline;">2</span>     0.166  0.092<span style="text-decoration: underline;">4</span>        8       0 <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> CASSREGDQPQHF               0.001<span style="text-decoration: underline;">57</span>    0.052<span style="text-decoration: underline;">0</span> 0.009<span style="text-decoration: underline;">13</span>       8       0 <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> CASRDGQGSGNTIYF             0.002<span style="text-decoration: underline;">78</span>    0.035<span style="text-decoration: underline;">1</span> 0.015<span style="text-decoration: underline;">5</span>        7       0 <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> CASSPFDRGPDTEAFF            0.005<span style="text-decoration: underline;">08</span>    0.016<span style="text-decoration: underline;">5</span> 0.011<span style="text-decoration: underline;">2</span>        7       0 <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> CASSQDLGQAFSPLHF            0.002<span style="text-decoration: underline;">23</span>    0.023<span style="text-decoration: underline;">5</span> 0.011<span style="text-decoration: underline;">1</span>        7       0 <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> CASSQDSSDTEAFF              0.001<span style="text-decoration: underline;">47</span>    0.048<span style="text-decoration: underline;">8</span> 0.010<span style="text-decoration: underline;">6</span>        7       0 <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> CAIKMETPNGEQYF              0.002<span style="text-decoration: underline;">53</span>    0.011<span style="text-decoration: underline;">8</span> 0.007<span style="text-decoration: underline;">04</span>       7       0 <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> CASSPGTGTYGYTF              0.001<span style="text-decoration: underline;">21</span>    0.015<span style="text-decoration: underline;">6</span> 0.005<span style="text-decoration: underline;">85</span>       7       0 CMV    </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 415 more rows, and abbreviated variable names ¹​maxFrequency,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ²​meanFrequency, ³​numberSamples, ⁴​prevalence</span></span></span></code></pre></div>
<p>One very useful thing to do is merge the output of
<code>seqMatrix</code> and <code>topFreq</code>.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">top_freq_matrix</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">full_join</a></span><span class="op">(</span><span class="va">top_freq</span>, <span class="va">sequence_matrix</span><span class="op">)</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(junction_aa)`</span></span>
<span><span class="va">top_freq_matrix</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 438 × 17</span></span></span>
<span><span class="co">#&gt;    junction_aa   minFr…¹ maxFr…² meanF…³ numbe…⁴ preva…⁵ antigen TRB_C…⁶ TRB_C…⁷</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> CASSQDRTGQYG… 0.004<span style="text-decoration: underline;">29</span>  0.024<span style="text-decoration: underline;">8</span> 0.010<span style="text-decoration: underline;">6</span>        9       0 <span style="color: #BB0000;">NA</span>            0 0.011<span style="text-decoration: underline;">2</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> CASSLQGREKLFF 0.056<span style="text-decoration: underline;">9</span>   0.322  0.113         8       0 <span style="color: #BB0000;">NA</span>            0 0.098<span style="text-decoration: underline;">6</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> CASSQDLMTVDS… 0.029<span style="text-decoration: underline;">2</span>   0.166  0.092<span style="text-decoration: underline;">4</span>        8       0 <span style="color: #BB0000;">NA</span>            0 0.120  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> CASSREGDQPQHF 0.001<span style="text-decoration: underline;">57</span>  0.052<span style="text-decoration: underline;">0</span> 0.009<span style="text-decoration: underline;">13</span>       8       0 <span style="color: #BB0000;">NA</span>            0 0.003<span style="text-decoration: underline;">27</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> CASRDGQGSGNT… 0.002<span style="text-decoration: underline;">78</span>  0.035<span style="text-decoration: underline;">1</span> 0.015<span style="text-decoration: underline;">5</span>        7       0 <span style="color: #BB0000;">NA</span>            0 0.015<span style="text-decoration: underline;">2</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> CASSPFDRGPDT… 0.005<span style="text-decoration: underline;">08</span>  0.016<span style="text-decoration: underline;">5</span> 0.011<span style="text-decoration: underline;">2</span>        7       0 <span style="color: #BB0000;">NA</span>            0 0.014<span style="text-decoration: underline;">9</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> CASSQDLGQAFS… 0.002<span style="text-decoration: underline;">23</span>  0.023<span style="text-decoration: underline;">5</span> 0.011<span style="text-decoration: underline;">1</span>        7       0 <span style="color: #BB0000;">NA</span>            0 0.009<span style="text-decoration: underline;">07</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> CASSQDSSDTEA… 0.001<span style="text-decoration: underline;">47</span>  0.048<span style="text-decoration: underline;">8</span> 0.010<span style="text-decoration: underline;">6</span>        7       0 <span style="color: #BB0000;">NA</span>            0 0.002<span style="text-decoration: underline;">92</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> CAIKMETPNGEQ… 0.002<span style="text-decoration: underline;">53</span>  0.011<span style="text-decoration: underline;">8</span> 0.007<span style="text-decoration: underline;">04</span>       7       0 <span style="color: #BB0000;">NA</span>            0 0.007<span style="text-decoration: underline;">96</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> CASSPGTGTYGY… 0.001<span style="text-decoration: underline;">21</span>  0.015<span style="text-decoration: underline;">6</span> 0.005<span style="text-decoration: underline;">85</span>       7       0 CMV           0 0.003<span style="text-decoration: underline;">38</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 428 more rows, 8 more variables: TRB_CD8_CMV_369 &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   TRB_Unsorted_0 &lt;dbl&gt;, TRB_Unsorted_1320 &lt;dbl&gt;, TRB_Unsorted_1496 &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   TRB_Unsorted_32 &lt;dbl&gt;, TRB_Unsorted_369 &lt;dbl&gt;, TRB_Unsorted_83 &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   TRB_Unsorted_949 &lt;dbl&gt;, and abbreviated variable names ¹​minFrequency,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ²​maxFrequency, ³​meanFrequency, ⁴​numberSamples, ⁵​prevalence, ⁶​TRB_CD4_949,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ⁷​TRB_CD8_949</span></span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="tracking-sequences-across-samples">Tracking sequences across samples<a class="anchor" aria-label="anchor" href="#tracking-sequences-across-samples"></a>
</h2>
<p>To visually track the frequency of sequences across multiple samples,
use the function <code>cloneTrack</code> This function takes the output
from the <code>seqMatrix</code> function. You can specify a character
vector of amino acid sequences using the parameter track to highlight
those sequences with a different color. Alternatively, you can highlight
all of the sequences from a given sample using the parameter map. If the
mapping feature is use, then you must specify a productive amino acid
list and a character vector of labels to title the mapped samples.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ctable</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/cloneTrack.html">cloneTrack</a></span><span class="op">(</span>study_table <span class="op">=</span> <span class="va">aa_table</span>,</span>
<span>                                 sample_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TRB_CD8_949"</span>, <span class="st">"TRB_CD8_CMV_369"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/plotTrack.html">plotTrack</a></span><span class="op">(</span><span class="va">ctable</span><span class="op">)</span></span></code></pre></div>
<p><img src="LymphoSeq2_files/figure-html/cloneTrack-1.png" width="700"></p>
<p>You can track particular sequences across samples by providing an
optional list of CDR3 amino acid sequences.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ttable</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/topSeqs.html">topSeqs</a></span><span class="op">(</span><span class="va">aa_table</span>, top <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">ctable</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/cloneTrack.html">cloneTrack</a></span><span class="op">(</span><span class="va">ttable</span><span class="op">)</span></span>
<span><span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/plotTrack.html">plotTrack</a></span><span class="op">(</span><span class="va">ctable</span>, alist <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CASSESAGSTGELFF"</span>, <span class="st">"CASSLAGDSQETQYF"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"bottom"</span><span class="op">)</span></span></code></pre></div>
<p><img src="LymphoSeq2_files/figure-html/plotCloneSet-1.png" width="700"></p>
<p>Alternatively you can use the function <code>plotTrackSingular</code>
to retrieve a list of alluvial diagrams each tracking one single amino
acid from the clone track table. Considering that a plot is generated
for each unique CDR3 sequence, we recommend running this feature on a
clone track table derived from only the top sequences from each
repertoire as described in the example above.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lalluvial</span> <span class="op">&lt;-</span> <span class="va">ctable</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/topSeqs.html">topSeqs</a></span><span class="op">(</span>top <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/plotTrackSingular.html">plotTrackSingular</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">lalluvial</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<p><img src="LymphoSeq2_files/figure-html/plotSingular-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="comparing-vdj-gene-usage">Comparing V(D)J gene usage<a class="anchor" aria-label="anchor" href="#comparing-vdj-gene-usage"></a>
</h2>
<p>To compare the V, D, and J gene usage across samples, start by
creating a data frame of V, D, and J gene counts and frequencies using
the function <code>geneFreq</code>. You can specify if you are
interested in the “VDJ”, “DJ”, “VJ”, “DJ”, “V”, “D”, or “J” loci using
the locus parameter. Set family to TRUE if you prefer the family names
instead of the gene names as reported by ImmunoSeq.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">vGenes</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/geneFreq.html">geneFreq</a></span><span class="op">(</span><span class="va">nuc_table</span>, locus <span class="op">=</span> <span class="st">"V"</span>, family <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">vGenes</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 167 × 5</span></span></span>
<span><span class="co">#&gt;    repertoire_id gene_name duplicate_count gene_type gene_frequency</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>               <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> TRB_CD4_949   <span style="color: #BB0000;">NA</span>                   <span style="text-decoration: underline;">2</span>945 v_family         0.205  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> TRB_CD4_949   TRBV10               <span style="text-decoration: underline;">5</span>071 v_family         0.353  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> TRB_CD4_949   TRBV11                107 v_family         0.007<span style="text-decoration: underline;">44</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> TRB_CD4_949   TRBV12                 29 v_family         0.002<span style="text-decoration: underline;">02</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> TRB_CD4_949   TRBV18                226 v_family         0.015<span style="text-decoration: underline;">7</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> TRB_CD4_949   TRBV19               <span style="text-decoration: underline;">1</span>643 v_family         0.114  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> TRB_CD4_949   TRBV2                 230 v_family         0.016<span style="text-decoration: underline;">0</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> TRB_CD4_949   TRBV21                 18 v_family         0.001<span style="text-decoration: underline;">25</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> TRB_CD4_949   TRBV27                 84 v_family         0.005<span style="text-decoration: underline;">84</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> TRB_CD4_949   TRBV28                208 v_family         0.014<span style="text-decoration: underline;">5</span> </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 157 more rows</span></span></span></code></pre></div>
<p>To create a chord diagram showing VJ or DJ gene associations from one
or more more samples, combine the output of <code>geneFreq</code> with
the function <code>chordDiagramVDJ</code>. This function works well the
<code>topSeqs</code> function that creates a data frame of a selected
number of top productive sequences. In the example below, a chord
diagram is made showing the association between V and J genes of just
the single dominant clones in each sample. The size of the ribbons
connecting VJ genes correspond to the number of samples that have that
recombination event. The thicker the ribbon, the higher the frequency of
the recombination.</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">top_seqs</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/topSeqs.html">topSeqs</a></span><span class="op">(</span><span class="va">nuc_table</span>, top <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/chordDiagramVDJ.html">chordDiagramVDJ</a></span><span class="op">(</span>study_table <span class="op">=</span> <span class="va">top_seqs</span>, </span>
<span>                            association <span class="op">=</span> <span class="st">"VJ"</span>, </span>
<span>                            colors <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"darkred"</span>, <span class="st">"navyblue"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="LymphoSeq2_files/figure-html/chordDiagram-1.png" width="700"></p>
<p>You can also visualize the results of <code>geneFreq</code> as a heat
map, word cloud, our cumulative frequency bar plot with the support of
additional R packages as shown below.</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">vGenes</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/geneFreq.html">geneFreq</a></span><span class="op">(</span><span class="va">nuc_table</span>, locus <span class="op">=</span> <span class="st">"V"</span>, family <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">RedBlue</span> <span class="op">&lt;-</span> <span class="fu">grDevices</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/grDevices/colorRamp.html" class="external-link">colorRampPalette</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rev.html" class="external-link">rev</a></span><span class="op">(</span><span class="fu">RColorBrewer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html" class="external-link">brewer.pal</a></span><span class="op">(</span><span class="fl">11</span>, <span class="st">"RdBu"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">(</span><span class="fl">256</span><span class="op">)</span></span>
<span><span class="va">vtable</span> <span class="op">&lt;-</span> <span class="va">vGenes</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">repertoire_id</span> <span class="op">==</span> <span class="st">"TRB_Unsorted_83"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">gene_name</span>, <span class="va">gene_frequency</span><span class="op">)</span></span>
<span><span class="fu">wordcloud2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/wordcloud2/man/wordcloud2.html" class="external-link">wordcloud2</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">vtable</span>, </span>
<span>                     color <span class="op">=</span> <span class="va">RedBlue</span><span class="op">)</span></span></code></pre></div>
<div class="wordcloud2 html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-8d32cfd9db97395c4a5f" style="width:700px;height:432.632880098888px;"></div>
<script type="application/json" data-for="htmlwidget-8d32cfd9db97395c4a5f">{"x":{"word":[null,"TRBV10","TRBV11","TRBV12","TRBV18","TRBV19","TRBV2","TRBV24","TRBV27","TRBV28","TRBV29","TRBV4","TRBV5","TRBV6","TRBV7","TRBV9"],"freq":[0.0156101275461641,0.0266711304541904,0.00370231788731562,0.0017723862226511,0.004844522341913,0.0218856876530324,0.0324477966613495,0.00380078378857402,0.323965943926951,0.0230541496812987,0.0141003170602021,0.343324340114352,0.0414344512495323,0.0731470358481525,0.0654601311565805,0.00477887840774073],"fontFamily":"Segoe UI","fontWeight":"bold","color":["#053061","#063263","#073466","#083669","#09386C","#0A3A6F","#0B3C72","#0C3E75","#0D4078","#0E437B","#0F457E","#114781","#124984","#134B87","#144D8A","#154F8D","#165190","#175493","#185695","#195898","#1A5A9B","#1C5C9E","#1D5EA1","#1E60A4","#1F62A7","#2064AA","#2166AC","#2368AD","#246AAE","#256CAF","#276DB0","#286FB0","#2971B1","#2A73B2","#2C75B3","#2D76B4","#2F78B5","#307AB6","#317CB7","#337DB8","#347FB9","#3581B9","#3783BA","#3884BB","#3986BC","#3A88BD","#3C8ABE","#3D8BBF","#3F8DC0","#408FC1","#4191C2","#4393C3","#4694C4","#4996C5","#4C98C6","#4F9AC7","#529CC8","#559EC9","#58A0CA","#5BA2CB","#5EA4CC","#61A6CD","#65A8CE","#68AACF","#6BACD0","#6EAED1","#71B0D2","#74B2D3","#77B4D4","#7AB6D6","#7DB8D7","#80BAD8","#84BCD9","#87BEDA","#8AC0DB","#8DC2DC","#90C4DD","#93C5DE","#95C6DF","#98C8DF","#9AC9E0","#9DCAE1","#9FCBE1","#A2CDE2","#A4CEE3","#A6CFE3","#A9D0E4","#ABD2E5","#AED3E6","#B0D4E6","#B3D5E7","#B5D7E8","#B8D8E8","#BAD9E9","#BDDAEA","#BFDCEB","#C2DDEB","#C4DEEC","#C7DFED","#C9E1ED","#CCE2EE","#CEE3EF","#D1E5F0","#D2E5F0","#D3E6F0","#D5E7F0","#D6E7F1","#D8E8F1","#D9E9F1","#DBE9F1","#DCEAF2","#DEEBF2","#DFECF2","#E1ECF3","#E2EDF3","#E4EEF3","#E5EEF3","#E7EFF4","#E8F0F4","#EAF1F4","#EBF1F4","#EDF2F5","#EEF3F5","#F0F3F5","#F1F4F6","#F3F5F6","#F4F5F6","#F6F6F6","#F7F6F6","#F7F5F4","#F7F4F2","#F7F3F0","#F8F2EE","#F8F0EC","#F8EFEA","#F8EEE8","#F9EDE7","#F9ECE5","#F9EBE3","#F9EAE1","#F9E9DF","#FAE8DD","#FAE7DB","#FAE5D9","#FAE4D7","#FBE3D6","#FBE2D4","#FBE1D2","#FBE0D0","#FCDFCE","#FCDECC","#FCDDCA","#FCDCC8","#FDDBC7","#FCD8C4","#FCD6C1","#FBD4BE","#FBD2BC","#FBD0B9","#FACEB6","#FACCB4","#FACAB1","#F9C7AE","#F9C5AB","#F9C3A9","#F8C1A6","#F8BFA3","#F8BDA1","#F7BB9E","#F7B99B","#F7B799","#F6B496","#F6B293","#F5B090","#F5AE8E","#F5AC8B","#F4AA88","#F4A886","#F4A683","#F3A380","#F2A07E","#F19E7C","#EF9B7A","#EE9878","#ED9676","#EC9374","#EB9072","#EA8E70","#E88B6E","#E7886C","#E6856A","#E58368","#E48065","#E27D63","#E17B61","#E0785F","#DF755D","#DE725B","#DD7059","#DB6D57","#DA6A55","#D96853","#D86551","#D7624F","#D6604D","#D45D4B","#D35A4A","#D15749","#D05447","#CE5146","#CD4F45","#CC4C43","#CA4942","#C94641","#C7433F","#C6403E","#C53E3D","#C33B3B","#C2383A","#C03539","#BF3237","#BE3036","#BC2D35","#BB2A33","#B92732","#B82431","#B6212F","#B51F2E","#B41C2D","#B2192B","#B0172A","#AD162A","#AA1529","#A71429","#A41328","#A11228","#9E1127","#9B1027","#991027","#960F26","#930E26","#900D25","#8D0C25","#8A0B24","#870A24","#840923","#810823","#7E0722","#7B0622","#780521","#750421","#720320","#6F0220","#6C011F","#69001F","#67001F"],"minSize":0,"weightFactor":524.285577713619,"backgroundColor":"white","gridSize":0,"minRotation":-0.785398163397448,"maxRotation":0.785398163397448,"shuffle":true,"rotateRatio":0.4,"shape":"circle","ellipticity":0.65,"figBase64":null,"hover":null},"evals":[],"jsHooks":{"render":[{"code":"function(el,x){\n                        console.log(123);\n                        if(!iii){\n                          window.location.reload();\n                          iii = False;\n\n                        }\n  }","data":null}]}}</script><div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">vGenes</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/geneFreq.html">geneFreq</a></span><span class="op">(</span><span class="va">nuc_table</span>, locus <span class="op">=</span> <span class="st">"V"</span>, family <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>           <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_wider.html" class="external-link">pivot_wider</a></span><span class="op">(</span>id_cols <span class="op">=</span> <span class="va">gene_name</span>, </span>
<span>                              names_from <span class="op">=</span> <span class="va">repertoire_id</span>, </span>
<span>                              values_from <span class="op">=</span> <span class="va">gene_frequency</span>, </span>
<span>                              values_fn <span class="op">=</span> <span class="va">sum</span>,</span>
<span>                              values_fill <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span>
<span><span class="va">gene_names</span> <span class="op">&lt;-</span> <span class="va">vGenes</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>              <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">gene_name</span><span class="op">)</span></span>
<span><span class="va">vGenes</span> <span class="op">&lt;-</span> <span class="va">vGenes</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>          <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">-</span><span class="va">gene_name</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>          <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">vGenes</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">gene_names</span></span>
<span><span class="fu">pheatmap</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html" class="external-link">pheatmap</a></span><span class="op">(</span><span class="va">vGenes</span>, scale <span class="op">=</span> <span class="st">"row"</span><span class="op">)</span></span></code></pre></div>
<p><img src="LymphoSeq2_files/figure-html/heatmap-1.png" width="700"></p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">vGenes</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/geneFreq.html">geneFreq</a></span><span class="op">(</span><span class="va">nuc_table</span>, locus <span class="op">=</span> <span class="st">"V"</span>, family <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">multicolors</span> <span class="op">&lt;-</span> <span class="fu">grDevices</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/grDevices/colorRamp.html" class="external-link">colorRampPalette</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rev.html" class="external-link">rev</a></span><span class="op">(</span><span class="fu">RColorBrewer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html" class="external-link">brewer.pal</a></span><span class="op">(</span><span class="fl">9</span>, <span class="st">"Set1"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">(</span><span class="fl">28</span><span class="op">)</span></span>
<span><span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">vGenes</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">repertoire_id</span>, y <span class="op">=</span> <span class="va">gene_frequency</span>, fill <span class="op">=</span> <span class="va">gene_name</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_y_continuous</a></span><span class="op">(</span>expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html" class="external-link">guides</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guide_legend.html" class="external-link">guide_legend</a></span><span class="op">(</span>ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">multicolors</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>y <span class="op">=</span> <span class="st">"Frequency (%)"</span>, x <span class="op">=</span> <span class="st">""</span>, fill <span class="op">=</span> <span class="st">""</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, vjust <span class="op">=</span> <span class="fl">0.5</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="LymphoSeq2_files/figure-html/geneFreqPlot-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="removing-sequences">Removing sequences<a class="anchor" aria-label="anchor" href="#removing-sequences"></a>
</h2>
<p>Occasionally you may identify one or more sequences in your data set
that appear to be contamination. You can remove an amino acid sequence
from all data frames using the function <code>removeSeq</code> and
recompute <code>frequencyCount</code> for all remaining sequences.</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/searchSeq.html">searchSeq</a></span><span class="op">(</span>study_table <span class="op">=</span> <span class="va">aa_table</span>, sequence <span class="op">=</span> <span class="st">"CASSESAGSTGELFF"</span>, seq_type <span class="op">=</span> <span class="st">"junction_aa"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 4 × 13</span></span></span>
<span><span class="co">#&gt;   reperto…¹ junct…² v_call d_call j_call v_fam…³ d_fam…⁴ j_fam…⁵ readi…⁶ dupli…⁷</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> TRB_CD4_… CASSES… TRBV1… TRBD2… TRBJ2… TRBV10  TRBD2   TRBJ2   in-fra…    <span style="text-decoration: underline;">5</span>019</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> TRB_Unso… CASSES… TRBV1… TRBD2… TRBJ2… TRBV10  TRBD2   TRBJ2   in-fra…   <span style="text-decoration: underline;">10</span>326</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> TRB_Unso… CASSES… TRBV1… TRBD2… TRBJ2… TRBV10  TRBD2   TRBJ2   in-fra…    <span style="text-decoration: underline;">1</span>755</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> TRB_Unso… CASSES… TRBV1… TRBD2… TRBJ2… TRBV10  TRBD2   TRBJ2   in-fra…     338</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 3 more variables: duplicate_frequency &lt;dbl&gt;, edit_distance &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   searchSequence &lt;chr&gt;, and abbreviated variable names ¹​repertoire_id,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ²​junction_aa, ³​v_family, ⁴​d_family, ⁵​j_family, ⁶​reading_frame,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ⁷​duplicate_count</span></span></span></code></pre></div>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cleansed</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/removeSeq.html">removeSeq</a></span><span class="op">(</span>study_table <span class="op">=</span> <span class="va">aa_table</span>, sequence <span class="op">=</span> <span class="st">"CASSESAGSTGELFF"</span><span class="op">)</span></span>
<span><span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/searchSeq.html">searchSeq</a></span><span class="op">(</span>study_table <span class="op">=</span> <span class="va">cleansed</span>, sequence <span class="op">=</span> <span class="st">"CASSESAGSTGELFF"</span>, seq_type <span class="op">=</span> <span class="st">"junction_aa"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 0 × 13</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 13 variables: repertoire_id &lt;chr&gt;, junction_aa &lt;chr&gt;, v_call &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   d_call &lt;chr&gt;, j_call &lt;chr&gt;, v_family &lt;chr&gt;, d_family &lt;chr&gt;, j_family &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   reading_frame &lt;chr&gt;, duplicate_count &lt;dbl&gt;, duplicate_frequency &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   edit_distance &lt;dbl&gt;, searchSequence &lt;chr&gt;</span></span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="rarefaction-curves">Rarefaction curves<a class="anchor" aria-label="anchor" href="#rarefaction-curves"></a>
</h2>
<p>Rarefaction and extrapolation curves allow for comparison of TCR
diversity across repertoires given a ideal sequencing depth. Rarefaction
and extrapolation curves are drawn by sampling a sequencing dataset to
various depths to understand the trajectory of sequence diversity and
then extrapolating the curve to an ideal depth.</p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/plotRarefactionCurve.html">plotRarefactionCurve</a></span><span class="op">(</span>study_table <span class="op">=</span> <span class="va">aa_table</span><span class="op">)</span></span></code></pre></div>
<p><img src="LymphoSeq2_files/figure-html/rarefation-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.2.3 (2023-03-15)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.2 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3</span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] vroom_1.6.1        lubridate_1.9.2    forcats_1.0.0      stringr_1.5.0     </span></span>
<span><span class="co">#&gt;  [5] dplyr_1.1.0        purrr_1.0.1        readr_2.1.4        tidyr_1.3.0       </span></span>
<span><span class="co">#&gt;  [9] tibble_3.2.1       ggplot2_3.4.1      tidyverse_2.0.0    wordcloud2_0.2.1  </span></span>
<span><span class="co">#&gt; [13] RColorBrewer_1.1-3 LymphoSeq2_1.0.0   data.table_1.14.8 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] colorspace_2.1-0       ggtree_3.7.1.003       ellipsis_0.3.2        </span></span>
<span><span class="co">#&gt;   [4] rprojroot_2.0.3        circlize_0.4.15        futile.logger_1.4.3   </span></span>
<span><span class="co">#&gt;   [7] XVector_0.38.0         GlobalOptions_0.1.2    fs_1.6.1              </span></span>
<span><span class="co">#&gt;  [10] aplot_0.1.10           farver_2.1.1           bit64_4.0.5           </span></span>
<span><span class="co">#&gt;  [13] fansi_1.0.4            codetools_0.2-19       extrafont_0.19        </span></span>
<span><span class="co">#&gt;  [16] cachem_1.0.7           knitr_1.42             polyclip_1.10-4       </span></span>
<span><span class="co">#&gt;  [19] jsonlite_1.8.4         Rttf2pt1_1.3.12        pheatmap_1.0.12       </span></span>
<span><span class="co">#&gt;  [22] ineq_0.2-13            ggforce_0.4.1          msa_1.30.1            </span></span>
<span><span class="co">#&gt;  [25] compiler_4.2.3         Matrix_1.5-3           fastmap_1.1.1         </span></span>
<span><span class="co">#&gt;  [28] lazyeval_0.2.2         cli_3.6.0              formatR_1.14          </span></span>
<span><span class="co">#&gt;  [31] tweenr_2.0.2           htmltools_0.5.4        prettyunits_1.1.1     </span></span>
<span><span class="co">#&gt;  [34] tools_4.2.3            igraph_1.4.1           gtable_0.3.2          </span></span>
<span><span class="co">#&gt;  [37] glue_1.6.2             GenomeInfoDbData_1.2.9 reshape2_1.4.4        </span></span>
<span><span class="co">#&gt;  [40] maps_3.4.1             fastmatch_1.1-3        Rcpp_1.0.10           </span></span>
<span><span class="co">#&gt;  [43] jquerylib_0.1.4        pkgdown_2.0.7          vctrs_0.6.0           </span></span>
<span><span class="co">#&gt;  [46] Biostrings_2.66.0      ggalt_0.4.0            ape_5.7-1             </span></span>
<span><span class="co">#&gt;  [49] nlme_3.1-162           extrafontdb_1.0        ggalluvial_0.12.5     </span></span>
<span><span class="co">#&gt;  [52] xfun_0.37              timechange_0.2.0       lifecycle_1.0.3       </span></span>
<span><span class="co">#&gt;  [55] phangorn_2.11.1        stringdist_0.9.10      zlibbioc_1.44.0       </span></span>
<span><span class="co">#&gt;  [58] MASS_7.3-58.2          scales_1.2.1           ragg_1.2.5            </span></span>
<span><span class="co">#&gt;  [61] hms_1.1.2              parallel_4.2.3         proj4_1.0-12          </span></span>
<span><span class="co">#&gt;  [64] lambda.r_1.2.4         yaml_2.3.7             gridExtra_2.3         </span></span>
<span><span class="co">#&gt;  [67] memoise_2.0.1          R4RNA_1.26.0           UpSetR_1.4.0          </span></span>
<span><span class="co">#&gt;  [70] ggfun_0.0.9            dtplyr_1.3.0           seqmagick_0.1.5       </span></span>
<span><span class="co">#&gt;  [73] yulab.utils_0.0.6      sass_0.4.5             stringi_1.7.12        </span></span>
<span><span class="co">#&gt;  [76] highr_0.10             S4Vectors_0.36.2       desc_1.4.2            </span></span>
<span><span class="co">#&gt;  [79] tidytree_0.4.2         ggmsa_1.3.4            BiocGenerics_0.44.0   </span></span>
<span><span class="co">#&gt;  [82] shape_1.4.6            GenomeInfoDb_1.34.9    rlang_1.1.0           </span></span>
<span><span class="co">#&gt;  [85] pkgconfig_2.0.3        systemfonts_1.0.4      bitops_1.0-7          </span></span>
<span><span class="co">#&gt;  [88] iNEXT_3.0.0            evaluate_0.20          lattice_0.20-45       </span></span>
<span><span class="co">#&gt;  [91] labeling_0.4.2         treeio_1.23.1          patchwork_1.1.2       </span></span>
<span><span class="co">#&gt;  [94] htmlwidgets_1.6.2      bit_4.0.5              tidyselect_1.2.0      </span></span>
<span><span class="co">#&gt;  [97] plyr_1.8.8             magrittr_2.0.3         R6_2.5.1              </span></span>
<span><span class="co">#&gt; [100] IRanges_2.32.0         generics_0.1.3         pillar_1.8.1          </span></span>
<span><span class="co">#&gt; [103] withr_2.5.0            RCurl_1.98-1.10        ash_1.0-15            </span></span>
<span><span class="co">#&gt; [106] crayon_1.5.2           futile.options_1.0.1   KernSmooth_2.23-20    </span></span>
<span><span class="co">#&gt; [109] utf8_1.2.3             tzdb_0.3.0             rmarkdown_2.20        </span></span>
<span><span class="co">#&gt; [112] progress_1.2.2         grid_4.2.3             digest_0.6.31         </span></span>
<span><span class="co">#&gt; [115] VennDiagram_1.7.3      gridGraphics_0.5-1     textshaping_0.3.6     </span></span>
<span><span class="co">#&gt; [118] stats4_4.2.3           munsell_0.5.0          ggplotify_0.1.0       </span></span>
<span><span class="co">#&gt; [121] bslib_0.4.2            quadprog_1.5-8</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Wu Elena, Ravishankar Shashidhar, Coffey David.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
