<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="LymphoSeq2">
<title>Analysis of Adaptive Immune Receptor Repertoire Sequencing data with LymphoSeq2 • LymphoSeq2</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Analysis of Adaptive Immune Receptor Repertoire Sequencing data with LymphoSeq2">
<meta property="og:description" content="LymphoSeq2">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">LymphoSeq2</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/LymphoSeq2.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/elulu3/LymphoSeq2/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">



<script src="LymphoSeq2_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <main id="main"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Analysis of Adaptive Immune Receptor Repertoire Sequencing data with LymphoSeq2</h1>
                        <h4 data-toc-skip class="author">Elena Wu</h4>
                        <h4 data-toc-skip class="author">Shashidhar Ravishankar</h4>
                        <h4 data-toc-skip class="author">David Coffey</h4>
            
            <h4 data-toc-skip class="date">2022-02-23</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/elulu3/LymphoSeq2/blob/HEAD/vignettes/LymphoSeq2.Rmd" class="external-link"><code>vignettes/LymphoSeq2.Rmd</code></a></small>
      <div class="d-none name"><code>LymphoSeq2.Rmd</code></div>
    </div>

    
    
<p>Adaptive Immune Receptor Repertoire Sequencing (AIRR-seq) provides a unique opportunity to interrogate the adaptive immune repertoire under various clinical conditions. The utility offered by this technology has quickly garnered interest from a community of clinicians and researchers investigating the immunological landscapes of a large spectrum of health and disease states. LymphoSeq2 is a toolkit that allows users to import, manipulate and visualize AIRR-Seq data from various AIRR-Seq assays such as Adaptive ImmunoSEQ and BGI-IRSeq, with support for 10X VDJ sequencing coming soon. The platform also supports the importing of AIRR-seq data processed using the MiXCR pipeline. The vignette highlights some of the key features of LymphoSeq2.</p>
<div class="section level2">
<h2 id="importing-data">Importing data<a class="anchor" aria-label="anchor" href="#importing-data"></a>
</h2>
<p>The function <code>readImmunoSeq</code> imports AIRR-seq receptor files from Adaptive ImmunoSEQ assay as well well as BGI-IRSeq assay. The sequences can be (.tsv) files processed using one of the three following platforms: Adaptive Biotechnologies ImmunoSEQ analyzer, BGI IR-SEQ iMonitor platform, and the MiXCR pipeline for AIRR-seq data analysis. The function has the ability to identify file type based on the headers provided in the (.tsv) file, accordingly the data is transformed into a format that is compatible AIRR-Community guidelines (<a href="https://github.com/airr-community/airr-standards" class="external-link uri">https://github.com/airr-community/airr-standards</a>).</p>
<p>To explore the features of LymphoSeq, this package includes 2 example data sets. The first is a data set of T cell receptor beta (TCRB) sequencing from 10 blood samples acquired serially from a single patient who underwent a bone marrow transplant (Kanakry, C.G., et al. JCI Insight 2016;1(5):pii: e86252). The second, is a data set of B cell receptor immunoglobulin heavy (IGH) chain sequencing from Burkitt lymphoma tumor biopsies acquired from 10 different individuals (Lombardo, K.A., et al. Blood Advances 2017 1:535-544). To improve performance, both data sets contain only the top 1,000 most frequent sequences. The complete data sets are publicly available through Adapatives’ immuneACCESS portal. As shown in the example below, you can specify the path to the example data sets using the command</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"TCRB_sequencing"</span>, package <span class="op">=</span> <span class="st">"LymphoSeq2"</span><span class="op">)</span> <span class="co">#For the TCRB files</span>
<span class="co">#&gt; [1] "/home/runner/work/_temp/Library/LymphoSeq2/extdata/TCRB_sequencing"</span>
<span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"IGH_sequencing"</span>, package <span class="op">=</span> <span class="st">"LymphoSeq2"</span><span class="op">)</span> <span class="co">#For the IGH files.</span>
<span class="co">#&gt; [1] "/home/runner/work/_temp/Library/LymphoSeq2/extdata/IGH_sequencing"</span></code></pre></div>
<p><code>readImmunoSeq</code> can take as input, a single file name, a list of files or the path to a directory containing AIRR-seq data. The columns are renamed to follow AIRR-community guidelines based on the input file type. The function returns a tibble with individual file names set as the <code>repertoire_id</code>. The CDR3 nucleotide and amino acid sequences are denoted by the <code>junction</code> and <code>junction_aa</code> fields respectively. The counts of the CDR3 sequeneces observed, and their frequency in each individual repertoire is denoted by the <code>duplicate_count</code> and <code>duplicate_frequency</code> field respectively.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/elulu3/LymphoSeq2" class="external-link">LymphoSeq2</a></span><span class="op">)</span>
<span class="co">#&gt; Loading required package: data.table</span>
<span class="co">#&gt; Warning: replacing previous import 'Biostrings::collapse' by 'dplyr::collapse'</span>
<span class="co">#&gt; when loading 'LymphoSeq2'</span>
<span class="co">#&gt; Warning: replacing previous import 'Biostrings::union' by 'dplyr::union' when</span>
<span class="co">#&gt; loading 'LymphoSeq2'</span>
<span class="co">#&gt; Warning: replacing previous import 'Biostrings::intersect' by 'dplyr::intersect'</span>
<span class="co">#&gt; when loading 'LymphoSeq2'</span>
<span class="co">#&gt; Warning: replacing previous import 'Biostrings::setdiff' by 'dplyr::setdiff'</span>
<span class="co">#&gt; when loading 'LymphoSeq2'</span>
<span class="co">#&gt; Warning: replacing previous import 'Biostrings::setequal' by 'dplyr::setequal'</span>
<span class="co">#&gt; when loading 'LymphoSeq2'</span>
<span class="co">#&gt; Warning: replacing previous import 'jsonlite::flatten' by 'purrr::flatten' when</span>
<span class="co">#&gt; loading 'LymphoSeq2'</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span>
<span class="co">#&gt; ── <span style="font-weight: bold;">Attaching packages</span> ─────────────────────────────────────── tidyverse 1.3.1 ──</span>
<span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">ggplot2</span> 3.3.5     <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">purrr  </span> 0.3.4</span>
<span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">tibble </span> 3.1.6     <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">dplyr  </span> 1.0.8</span>
<span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">tidyr  </span> 1.2.0     <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">stringr</span> 1.4.0</span>
<span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">readr  </span> 2.1.2     <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">forcats</span> 0.5.1</span>
<span class="co">#&gt; ── <span style="font-weight: bold;">Conflicts</span> ────────────────────────────────────────── tidyverse_conflicts() ──</span>
<span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">dplyr</span>::<span style="color: #00BB00;">between()</span>   masks <span style="color: #0000BB;">data.table</span>::between()</span>
<span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">dplyr</span>::<span style="color: #00BB00;">filter()</span>    masks <span style="color: #0000BB;">stats</span>::filter()</span>
<span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">dplyr</span>::<span style="color: #00BB00;">first()</span>     masks <span style="color: #0000BB;">data.table</span>::first()</span>
<span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">dplyr</span>::<span style="color: #00BB00;">lag()</span>       masks <span style="color: #0000BB;">stats</span>::lag()</span>
<span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">dplyr</span>::<span style="color: #00BB00;">last()</span>      masks <span style="color: #0000BB;">data.table</span>::last()</span>
<span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">purrr</span>::<span style="color: #00BB00;">transpose()</span> masks <span style="color: #0000BB;">data.table</span>::transpose()</span>
<span class="va">study_files</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"TCRB_sequencing"</span>, package <span class="op">=</span> <span class="st">"LymphoSeq2"</span><span class="op">)</span> 
<span class="va">study_table</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/readImmunoSeq.html">readImmunoSeq</a></span><span class="op">(</span><span class="va">study_files</span><span class="op">)</span>
<span class="co">#&gt; <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">1</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">144</span></span>
<span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="co">#&gt; <span style="font-weight: bold;">Delimiter:</span> ","</span>
<span class="co">#&gt; <span style="color: #BB0000;">chr</span> (69): sequence_id, sequence, sequence_aa, locus, v_call, d_call, d2_call...</span>
<span class="co">#&gt; <span style="color: #00BB00;">dbl</span> (70): v_score, v_identity, v_support, d_score, d_identity, d_support, d2...</span>
<span class="co">#&gt; <span style="color: #BBBB00;">lgl</span>  (5): rev_comp, productive, vj_in_frame, stop_codon, complete_vdj</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="co">#&gt; <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">1000</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">52</span></span>
<span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="co">#&gt; <span style="font-weight: bold;">Delimiter:</span> "\t"</span>
<span class="co">#&gt; <span style="color: #BB0000;">chr</span> (33): nucleotide, aminoAcid, vMaxResolved, vFamilyName, vGeneName, vGene...</span>
<span class="co">#&gt; <span style="color: #00BB00;">dbl</span> (15): count (templates/reads), frequencyCount (%), cdr3Length, vDeletion...</span>
<span class="co">#&gt; <span style="color: #BBBB00;">lgl</span>  (4): vFamilyTies, jFamilyTies, jGeneNameTies, jGeneAlleleTies</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="co">#&gt; Joining, by = c("sequence", "sequence_aa", "v_call", "d_call", "d2_call", "j_call", "junction", "junction_aa", "duplicate_count", "clone_id", "repertoire_id")</span>
<span class="co">#&gt; <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">1000</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">52</span></span>
<span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="co">#&gt; <span style="font-weight: bold;">Delimiter:</span> "\t"</span>
<span class="co">#&gt; <span style="color: #BB0000;">chr</span> (34): nucleotide, aminoAcid, vMaxResolved, vFamilyName, vGeneName, vGene...</span>
<span class="co">#&gt; <span style="color: #00BB00;">dbl</span> (15): count (templates/reads), frequencyCount (%), cdr3Length, vDeletion...</span>
<span class="co">#&gt; <span style="color: #BBBB00;">lgl</span>  (3): jFamilyTies, jGeneNameTies, jGeneAlleleTies</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="co">#&gt; Joining, by = c("sequence", "sequence_aa", "v_call", "d_call", "d2_call", "j_call", "junction", "junction_aa", "duplicate_count", "clone_id", "repertoire_id")</span>
<span class="co">#&gt; <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">414</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">52</span></span>
<span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="co">#&gt; <span style="font-weight: bold;">Delimiter:</span> "\t"</span>
<span class="co">#&gt; <span style="color: #BB0000;">chr</span> (34): nucleotide, aminoAcid, vMaxResolved, vFamilyName, vGeneName, vGene...</span>
<span class="co">#&gt; <span style="color: #00BB00;">dbl</span> (15): count (templates/reads), frequencyCount (%), cdr3Length, vDeletion...</span>
<span class="co">#&gt; <span style="color: #BBBB00;">lgl</span>  (3): jFamilyTies, jGeneNameTies, jGeneAlleleTies</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="co">#&gt; Joining, by = c("sequence", "sequence_aa", "v_call", "d_call", "d2_call", "j_call", "junction", "junction_aa", "duplicate_count", "clone_id", "repertoire_id")</span>
<span class="co">#&gt; <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">1000</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">52</span></span>
<span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="co">#&gt; <span style="font-weight: bold;">Delimiter:</span> "\t"</span>
<span class="co">#&gt; <span style="color: #BB0000;">chr</span> (34): nucleotide, aminoAcid, vMaxResolved, vFamilyName, vGeneName, vGene...</span>
<span class="co">#&gt; <span style="color: #00BB00;">dbl</span> (15): count (templates/reads), frequencyCount (%), cdr3Length, vDeletion...</span>
<span class="co">#&gt; <span style="color: #BBBB00;">lgl</span>  (3): jFamilyTies, jGeneNameTies, jGeneAlleleTies</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="co">#&gt; Joining, by = c("sequence", "sequence_aa", "v_call", "d_call", "d2_call", "j_call", "junction", "junction_aa", "duplicate_count", "clone_id", "repertoire_id")</span>
<span class="co">#&gt; <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">1000</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">52</span></span>
<span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="co">#&gt; <span style="font-weight: bold;">Delimiter:</span> "\t"</span>
<span class="co">#&gt; <span style="color: #BB0000;">chr</span> (34): nucleotide, aminoAcid, vMaxResolved, vFamilyName, vGeneName, vGene...</span>
<span class="co">#&gt; <span style="color: #00BB00;">dbl</span> (15): count (templates/reads), frequencyCount (%), cdr3Length, vDeletion...</span>
<span class="co">#&gt; <span style="color: #BBBB00;">lgl</span>  (3): jFamilyTies, jGeneNameTies, jGeneAlleleTies</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="co">#&gt; Joining, by = c("sequence", "sequence_aa", "v_call", "d_call", "d2_call", "j_call", "junction", "junction_aa", "duplicate_count", "clone_id", "repertoire_id")</span>
<span class="co">#&gt; <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">1000</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">52</span></span>
<span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="co">#&gt; <span style="font-weight: bold;">Delimiter:</span> "\t"</span>
<span class="co">#&gt; <span style="color: #BB0000;">chr</span> (35): nucleotide, aminoAcid, vMaxResolved, vFamilyName, vGeneName, vGene...</span>
<span class="co">#&gt; <span style="color: #00BB00;">dbl</span> (15): count (templates/reads), frequencyCount (%), cdr3Length, vDeletion...</span>
<span class="co">#&gt; <span style="color: #BBBB00;">lgl</span>  (2): jFamilyTies, jGeneAlleleTies</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="co">#&gt; Joining, by = c("sequence", "sequence_aa", "v_call", "d_call", "d2_call", "j_call", "junction", "junction_aa", "duplicate_count", "clone_id", "repertoire_id")</span>
<span class="co">#&gt; <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">920</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">52</span></span>
<span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="co">#&gt; <span style="font-weight: bold;">Delimiter:</span> "\t"</span>
<span class="co">#&gt; <span style="color: #BB0000;">chr</span> (29): nucleotide, aminoAcid, vMaxResolved, vFamilyName, vGeneName, vFami...</span>
<span class="co">#&gt; <span style="color: #00BB00;">dbl</span> (14): count (templates/reads), frequencyCount (%), cdr3Length, vDeletion...</span>
<span class="co">#&gt; <span style="color: #BBBB00;">lgl</span>  (9): vGeneAllele, vGeneAlleleTies, dGeneAllele, dFamilyTies, dGeneAllel...</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="co">#&gt; Joining, by = c("sequence", "sequence_aa", "v_call", "d_call", "d2_call", "j_call", "junction", "junction_aa", "duplicate_count", "clone_id", "repertoire_id")</span>
<span class="co">#&gt; <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">1000</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">52</span></span>
<span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="co">#&gt; <span style="font-weight: bold;">Delimiter:</span> "\t"</span>
<span class="co">#&gt; <span style="color: #BB0000;">chr</span> (29): nucleotide, aminoAcid, vMaxResolved, vFamilyName, vGeneName, vFami...</span>
<span class="co">#&gt; <span style="color: #00BB00;">dbl</span> (14): count (templates/reads), frequencyCount (%), cdr3Length, vDeletion...</span>
<span class="co">#&gt; <span style="color: #BBBB00;">lgl</span>  (9): vGeneAllele, vGeneAlleleTies, dGeneAllele, dFamilyTies, dGeneAllel...</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="co">#&gt; Joining, by = c("sequence", "sequence_aa", "v_call", "d_call", "d2_call", "j_call", "junction", "junction_aa", "duplicate_count", "clone_id", "repertoire_id")</span>
<span class="co">#&gt; <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">1000</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">52</span></span>
<span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="co">#&gt; <span style="font-weight: bold;">Delimiter:</span> "\t"</span>
<span class="co">#&gt; <span style="color: #BB0000;">chr</span> (29): nucleotide, aminoAcid, vMaxResolved, vFamilyName, vGeneName, vFami...</span>
<span class="co">#&gt; <span style="color: #00BB00;">dbl</span> (14): count (templates/reads), frequencyCount (%), cdr3Length, vDeletion...</span>
<span class="co">#&gt; <span style="color: #BBBB00;">lgl</span>  (9): vGeneAllele, vGeneAlleleTies, dGeneAllele, dFamilyTies, dGeneAllel...</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="co">#&gt; Joining, by = c("sequence", "sequence_aa", "v_call", "d_call", "d2_call", "j_call", "junction", "junction_aa", "duplicate_count", "clone_id", "repertoire_id")</span>
<span class="co">#&gt; <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">1000</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">52</span></span>
<span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="co">#&gt; <span style="font-weight: bold;">Delimiter:</span> "\t"</span>
<span class="co">#&gt; <span style="color: #BB0000;">chr</span> (34): nucleotide, aminoAcid, vMaxResolved, vFamilyName, vGeneName, vGene...</span>
<span class="co">#&gt; <span style="color: #00BB00;">dbl</span> (15): count (templates/reads), frequencyCount (%), cdr3Length, vDeletion...</span>
<span class="co">#&gt; <span style="color: #BBBB00;">lgl</span>  (3): jFamilyTies, jGeneNameTies, jGeneAlleleTies</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="co">#&gt; Joining, by = c("sequence", "sequence_aa", "v_call", "d_call", "d2_call", "j_call", "junction", "junction_aa", "duplicate_count", "clone_id", "repertoire_id")</span></code></pre></div>
<p>Looking at the <code>study_table</code> we see a tibble with 12 columns and 9334 rows</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">study_table</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 9,334 × 144</span></span>
<span class="co">#&gt;    sequence_id sequence   sequence_aa rev_comp productive vj_in_frame stop_codon</span>
<span class="co">#&gt;          <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span>           1 CCTCTCACT… CASSIHQPQHF FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span>           2 ATTTCCCGC… <span style="color: #BB0000;">NA</span>          FALSE    FALSE      <span style="color: #BB0000;">NA</span>          TRUE      </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span>           3 ATCCTGAGT… CAWSAQETQYF FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span>           4 GAGTCAGCT… CASSESAGST… FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span>           5 GCCCTCAGA… <span style="color: #BB0000;">NA</span>          FALSE    FALSE      <span style="color: #BB0000;">NA</span>          TRUE      </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span>           6 ATTCCCTGG… <span style="color: #BB0000;">NA</span>          FALSE    FALSE      <span style="color: #BB0000;">NA</span>          TRUE      </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span>           7 GTGACATCG… CASSPRQGES… FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span>           8 ACCTTGGAG… CASSLDGQGQ… FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span>           9 GTGACCAGT… CSAKTSGITY… FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span>          10 ACCCTGCAG… CASSQD*ASS… FALSE    FALSE      <span style="color: #BB0000;">NA</span>          TRUE      </span>
<span class="co">#&gt; <span style="color: #949494;"># … with 9,324 more rows, and 137 more variables: complete_vdj &lt;lgl&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   locus &lt;chr&gt;, v_call &lt;chr&gt;, d_call &lt;chr&gt;, d2_call &lt;chr&gt;, j_call &lt;chr&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   c_call &lt;chr&gt;, sequence_alignment &lt;chr&gt;, sequence_alignment_aa &lt;chr&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   germline_alignment &lt;chr&gt;, germline_alignment_aa &lt;chr&gt;, junction &lt;chr&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   junction_aa &lt;chr&gt;, np1 &lt;chr&gt;, np1_aa &lt;chr&gt;, np2 &lt;chr&gt;, np2_aa &lt;chr&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   np3 &lt;chr&gt;, np3_aa &lt;chr&gt;, cdr1 &lt;chr&gt;, cdr1_aa &lt;chr&gt;, cdr2 &lt;chr&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   cdr2_aa &lt;chr&gt;, cdr3 &lt;chr&gt;, cdr3_aa &lt;chr&gt;, fwr1 &lt;chr&gt;, fwr1_aa &lt;chr&gt;, …</span></span></code></pre></div>
<p>Since the study table is a tibble, we can use tidyverse syntax to extract a list of sample names</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">study_table</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
<span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">repertoire_id</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
<span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/duplicated.html" class="external-link">unique</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt;  [1] "TRB_CD4_949"       "TRB_CD8_949"       "TRB_CD8_CMV_369"  </span>
<span class="co">#&gt;  [4] "TRB_Unsorted_0"    "TRB_Unsorted_1320" "TRB_Unsorted_1496"</span>
<span class="co">#&gt;  [7] "TRB_Unsorted_32"   "TRB_Unsorted_369"  "TRB_Unsorted_83"  </span>
<span class="co">#&gt; [10] "TRB_Unsorted_949"</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="subsetting-data">Subsetting Data<a class="anchor" aria-label="anchor" href="#subsetting-data"></a>
</h2>
<p>The tibble structure of the TCR data allows for easy subsampling of data. To select the TCR sequences from any given samples in the dataset, the <code>filter</code> function from the <code>dplyr</code> package can be used.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">TRB_Unsorted_0</span> <span class="op">&lt;-</span> <span class="va">study_table</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">repertoire_id</span> <span class="op">==</span> <span class="st">"TRB_Unsorted_0"</span><span class="op">)</span>
<span class="va">TRB_Unsorted_0</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 1,000 × 144</span></span>
<span class="co">#&gt;    sequence_id sequence   sequence_aa rev_comp productive vj_in_frame stop_codon</span>
<span class="co">#&gt;          <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span>           1 TCAATTCCC… <span style="color: #BB0000;">NA</span>          FALSE    FALSE      <span style="color: #BB0000;">NA</span>          TRUE      </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span>           2 CTGATTCTG… CASSPVSNEQ… FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span>           3 ATCAATTCC… CASSQEVPPY… FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span>           4 TGCCATCCC… <span style="color: #BB0000;">NA</span>          FALSE    FALSE      <span style="color: #BB0000;">NA</span>          TRUE      </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span>           5 CACACCCTG… CASSQEASGR… FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span>           6 GCCAGCACC… CASSLEHTGA… FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span>           7 CCCCTGACC… CASSPGDEQYF FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span>           8 AGTGCCCAT… CSARSPSTGT… FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span>           9 GGAGCTTGG… <span style="color: #BB0000;">NA</span>          FALSE    FALSE      <span style="color: #BB0000;">NA</span>          TRUE      </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span>          10 CTGTAGTCT… CASSEKREGH… FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #949494;"># … with 990 more rows, and 137 more variables: complete_vdj &lt;lgl&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   locus &lt;chr&gt;, v_call &lt;chr&gt;, d_call &lt;chr&gt;, d2_call &lt;chr&gt;, j_call &lt;chr&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   c_call &lt;chr&gt;, sequence_alignment &lt;chr&gt;, sequence_alignment_aa &lt;chr&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   germline_alignment &lt;chr&gt;, germline_alignment_aa &lt;chr&gt;, junction &lt;chr&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   junction_aa &lt;chr&gt;, np1 &lt;chr&gt;, np1_aa &lt;chr&gt;, np2 &lt;chr&gt;, np2_aa &lt;chr&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   np3 &lt;chr&gt;, np3_aa &lt;chr&gt;, cdr1 &lt;chr&gt;, cdr1_aa &lt;chr&gt;, cdr2 &lt;chr&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   cdr2_aa &lt;chr&gt;, cdr3 &lt;chr&gt;, cdr3_aa &lt;chr&gt;, fwr1 &lt;chr&gt;, fwr1_aa &lt;chr&gt;, …</span></span></code></pre></div>
<p>The <code>str_detect</code> function from <code>stringr</code> package can be used in conjunction with the <code>filter</code> to find samples using a pattern</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">CMV</span> <span class="op">&lt;-</span> <span class="va">study_table</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
       <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_detect.html" class="external-link">str_detect</a></span><span class="op">(</span><span class="va">repertoire_id</span>, <span class="st">"CMV"</span><span class="op">)</span><span class="op">)</span>
<span class="va">CMV</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 414 × 144</span></span>
<span class="co">#&gt;    sequence_id sequence   sequence_aa rev_comp productive vj_in_frame stop_codon</span>
<span class="co">#&gt;          <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span>           1 CAGCGCACA… CASSPPTGER… FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span>           2 CAGCCCTCA… CASSPAGAYY… FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span>           3 CAGCCTGCA… CASSQDWERL… FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span>           4 TCGGCCCTG… CASSQDLMTV… FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span>           5 ATCCTGGAG… CASSLQGREK… FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span>           6 GAGGATCCA… <span style="color: #BB0000;">NA</span>          FALSE    FALSE      <span style="color: #BB0000;">NA</span>          TRUE      </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span>           7 ACCCTGCAG… CASSQDLGQA… FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span>           8 CTCCTCACT… CAISDTGELFF FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span>           9 GAGTCCGCC… CASSLAGDSQ… FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span>          10 GAGTTGGCT… CASRDGQGSG… FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #949494;"># … with 404 more rows, and 137 more variables: complete_vdj &lt;lgl&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   locus &lt;chr&gt;, v_call &lt;chr&gt;, d_call &lt;chr&gt;, d2_call &lt;chr&gt;, j_call &lt;chr&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   c_call &lt;chr&gt;, sequence_alignment &lt;chr&gt;, sequence_alignment_aa &lt;chr&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   germline_alignment &lt;chr&gt;, germline_alignment_aa &lt;chr&gt;, junction &lt;chr&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   junction_aa &lt;chr&gt;, np1 &lt;chr&gt;, np1_aa &lt;chr&gt;, np2 &lt;chr&gt;, np2_aa &lt;chr&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   np3 &lt;chr&gt;, np3_aa &lt;chr&gt;, cdr1 &lt;chr&gt;, cdr1_aa &lt;chr&gt;, cdr2 &lt;chr&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   cdr2_aa &lt;chr&gt;, cdr3 &lt;chr&gt;, cdr3_aa &lt;chr&gt;, fwr1 &lt;chr&gt;, fwr1_aa &lt;chr&gt;, …</span></span></code></pre></div>
<p>A metadata file for the TCR sequencing samples can easily be combined with the <code>study_table</code> by reading in the metadata file as a tibble and using the <code><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">dplyr::left_join</a></code> function to merge the two tables. In the example below, a metadata file is imported for the example TCRB data set which contains information on the number of days post bone marrow transplant the sample was collected and the cellular phenotype the blood sample was sorted for prior to sequencing.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">TCRB_metadata</span> <span class="op">&lt;-</span> <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link">read_csv</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"TCRB_metadata.csv"</span>, package <span class="op">=</span> <span class="st">"LymphoSeq2"</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">10</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">3</span></span>
<span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="co">#&gt; <span style="font-weight: bold;">Delimiter:</span> ","</span>
<span class="co">#&gt; <span style="color: #BB0000;">chr</span> (2): samples, phenotype</span>
<span class="co">#&gt; <span style="color: #00BB00;">dbl</span> (1): day</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="va">TCRB_metadata</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 10 × 3</span></span>
<span class="co">#&gt;    samples             day phenotype</span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> TRB_Unsorted_0        0 Unsorted </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> TRB_Unsorted_32      32 Unsorted </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> TRB_Unsorted_83      82 Unsorted </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> TRB_CD8_CMV_369     369 CD8+CMV+ </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> TRB_Unsorted_369    369 Unsorted </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> TRB_CD4_949         949 CD4+     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> TRB_CD8_949         949 CD8+     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> TRB_Unsorted_949    949 Unsorted </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> TRB_Unsorted_1320  <span style="text-decoration: underline;">1</span>320 Unsorted </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> TRB_Unsorted_1496  <span style="text-decoration: underline;">1</span>496 Unsorted</span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">study_table</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">study_table</span>, <span class="va">TCRB_metadata</span>, by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"repertoire_id"</span> <span class="op">=</span> <span class="st">"samples"</span><span class="op">)</span><span class="op">)</span>
<span class="va">study_table</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 9,334 × 146</span></span>
<span class="co">#&gt;    sequence_id sequence   sequence_aa rev_comp productive vj_in_frame stop_codon</span>
<span class="co">#&gt;          <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span>           1 CCTCTCACT… CASSIHQPQHF FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span>           2 ATTTCCCGC… <span style="color: #BB0000;">NA</span>          FALSE    FALSE      <span style="color: #BB0000;">NA</span>          TRUE      </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span>           3 ATCCTGAGT… CAWSAQETQYF FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span>           4 GAGTCAGCT… CASSESAGST… FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span>           5 GCCCTCAGA… <span style="color: #BB0000;">NA</span>          FALSE    FALSE      <span style="color: #BB0000;">NA</span>          TRUE      </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span>           6 ATTCCCTGG… <span style="color: #BB0000;">NA</span>          FALSE    FALSE      <span style="color: #BB0000;">NA</span>          TRUE      </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span>           7 GTGACATCG… CASSPRQGES… FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span>           8 ACCTTGGAG… CASSLDGQGQ… FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span>           9 GTGACCAGT… CSAKTSGITY… FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span>          10 ACCCTGCAG… CASSQD*ASS… FALSE    FALSE      <span style="color: #BB0000;">NA</span>          TRUE      </span>
<span class="co">#&gt; <span style="color: #949494;"># … with 9,324 more rows, and 139 more variables: complete_vdj &lt;lgl&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   locus &lt;chr&gt;, v_call &lt;chr&gt;, d_call &lt;chr&gt;, d2_call &lt;chr&gt;, j_call &lt;chr&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   c_call &lt;chr&gt;, sequence_alignment &lt;chr&gt;, sequence_alignment_aa &lt;chr&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   germline_alignment &lt;chr&gt;, germline_alignment_aa &lt;chr&gt;, junction &lt;chr&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   junction_aa &lt;chr&gt;, np1 &lt;chr&gt;, np1_aa &lt;chr&gt;, np2 &lt;chr&gt;, np2_aa &lt;chr&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   np3 &lt;chr&gt;, np3_aa &lt;chr&gt;, cdr1 &lt;chr&gt;, cdr1_aa &lt;chr&gt;, cdr2 &lt;chr&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   cdr2_aa &lt;chr&gt;, cdr3 &lt;chr&gt;, cdr3_aa &lt;chr&gt;, fwr1 &lt;chr&gt;, fwr1_aa &lt;chr&gt;, …</span></span></code></pre></div>
<p>Now the metadata information can be used to further subset the data. For instance to select all “Unsorted” samples collected more than 300 days after bone marrow transplant, we would use the following code</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">unsorted_300</span> <span class="op">&lt;-</span> <span class="va">study_table</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
                <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">day</span> <span class="op">&gt;</span> <span class="fl">300</span> <span class="op">&amp;</span> <span class="va">phenotype</span> <span class="op">==</span> <span class="st">"Unsorted"</span><span class="op">)</span>
<span class="va">unsorted_300</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 4,000 × 146</span></span>
<span class="co">#&gt;    sequence_id sequence   sequence_aa rev_comp productive vj_in_frame stop_codon</span>
<span class="co">#&gt;          <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span>           1 CACCCTGCA… <span style="color: #BB0000;">NA</span>          FALSE    FALSE      <span style="color: #BB0000;">NA</span>          TRUE      </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span>           2 AGTGCCCAT… CSALLGLAGR… FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span>           3 CAGCGCACA… CASSFRIGGS… FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span>           4 TCGGCTGCT… CASSKGTSDL… FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span>           5 GCTGCTCCC… CASSAPGTSG… FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span>           6 ACCAGTGCC… CSARDLWGPT… FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span>           7 ATCCGGTCC… CASSEVSGAY… FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span>           8 AAGATCCAG… CASSSPTRDT… FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span>           9 ATCCAGCGC… CASSLDGPLD… FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span>          10 AGCCCTGAA… CSVEDPHGTS… FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #949494;"># … with 3,990 more rows, and 139 more variables: complete_vdj &lt;lgl&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   locus &lt;chr&gt;, v_call &lt;chr&gt;, d_call &lt;chr&gt;, d2_call &lt;chr&gt;, j_call &lt;chr&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   c_call &lt;chr&gt;, sequence_alignment &lt;chr&gt;, sequence_alignment_aa &lt;chr&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   germline_alignment &lt;chr&gt;, germline_alignment_aa &lt;chr&gt;, junction &lt;chr&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   junction_aa &lt;chr&gt;, np1 &lt;chr&gt;, np1_aa &lt;chr&gt;, np2 &lt;chr&gt;, np2_aa &lt;chr&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   np3 &lt;chr&gt;, np3_aa &lt;chr&gt;, cdr1 &lt;chr&gt;, cdr1_aa &lt;chr&gt;, cdr2 &lt;chr&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   cdr2_aa &lt;chr&gt;, cdr3 &lt;chr&gt;, cdr3_aa &lt;chr&gt;, fwr1 &lt;chr&gt;, fwr1_aa &lt;chr&gt;, …</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="extracting-productive-sequences">Extracting productive sequences<a class="anchor" aria-label="anchor" href="#extracting-productive-sequences"></a>
</h2>
<p>When AIRR-seq samples are derived from genomic DNA rather than complimentary DNA made from RNA, then you will find productive and unproductive sequences. Productive sequences are defined as in-frame sequences without any early stop codons. To filter out these productive sequences, you can use the <code>productiveSeq</code> to remove unproductive sequences and recompute the duplicate_frequency to reflect the productive amino acid or nucleotide sequence frequencies.</p>
<p>If you are interested in just the complementarity determining region 3 (CDR3) amino acid sequences, then set aggregate to <code>junction_aa</code> and the <code>duplicate_count</code> for duplicate amino acid sequences will be summed. The resulting tibble will have <code>junction_aa</code>, <code>duplicate_count</code>, <code>duplicate_frequency</code>, <code>reading_frame</code>, and the most frequent VDJ gene combinations for each of the duplicated amino acid sequences and the corresponding gene family names. These gene names are only kept for consistency of the tibble structure, but since a single amino acid sequence can be generated from different VDJ combinations, it is inadvisable to use these values for downstream analysis</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">aa_table</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/productiveSeq.html">productiveSeq</a></span><span class="op">(</span>study_table <span class="op">=</span> <span class="va">study_table</span>, aggregate <span class="op">=</span> <span class="st">"junction_aa"</span>, 
                          prevalence <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
  <span class="va">aa_table</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 7,533 × 11</span></span>
<span class="co">#&gt;    repertoire_id junction_aa     v_call d_call j_call v_family d_family j_family</span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> TRB_CD4_949   CAIKPGQGASNSPL… TCRBV… TCRBD… TCRBJ… TCRBV10  TCRBD01  TCRBJ01 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> TRB_CD4_949   CAIRAGTSTDTQYF  TCRBV… TCRBD… TCRBJ… TCRBV10  TCRBD02  TCRBJ02 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> TRB_CD4_949   CAISDETPGELFF   TCRBV… TCRBD… TCRBJ… TCRBV10  TCRBD01  TCRBJ02 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> TRB_CD4_949   CAISDLGRGDTEAFF TCRBV… TCRBD… TCRBJ… TCRBV10  TCRBD01  TCRBJ01 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> TRB_CD4_949   CAISDLKEQPQHF   TCRBV… TCRBD… TCRBJ… TCRBV10  TCRBD02  TCRBJ01 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> TRB_CD4_949   CAISDQGGDQPQHF  TCRBV… TCRBD… TCRBJ… TCRBV10  TCRBD02  TCRBJ01 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> TRB_CD4_949   CAISEREQGAFF    TCRBV… TCRBD… TCRBJ… TCRBV10  TCRBD01  TCRBJ01 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> TRB_CD4_949   CAISEWSGSSYNEQ… TCRBV… TCRBD… TCRBJ… TCRBV10  TCRBD01  TCRBJ02 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> TRB_CD4_949   CAISGQGSTEAFF   TCRBV… TCRBD… TCRBJ… TCRBV10  TCRBD01  TCRBJ01 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> TRB_CD4_949   CAISLNSGGAYEQYF TCRBV… TCRBD… TCRBJ… TCRBV10  TCRBD02  TCRBJ02 </span>
<span class="co">#&gt; <span style="color: #949494;"># … with 7,523 more rows, and 3 more variables: reading_frame &lt;chr&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   duplicate_count &lt;dbl&gt;, duplicate_frequency &lt;dbl&gt;</span></span></code></pre></div>
<p>Alternatively you can aggregate by <code>junction</code> to group sequences by CDR3 nucleotide sequences. This option will produce a tibble similar to the output of <code>readImmunoSeq</code>. Many of the functions within LymphoSeq2 use the results from <code>productiveSeq</code> function. Please be sure to check the function documentation.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">nuc_table</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/productiveSeq.html">productiveSeq</a></span><span class="op">(</span>study_table <span class="op">=</span> <span class="va">study_table</span>, aggregate <span class="op">=</span> <span class="st">"junction"</span>, 
                          prevalence <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">nuc_table</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 7,679 × 12</span></span>
<span class="co">#&gt;    repertoire_id junction     junction_aa v_call d_call j_call v_family d_family</span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> TRB_CD4_949   AACATGAGCTC… CASSLGGGYG… TCRBV… TCRBD… TCRBJ… TCRBV13  TCRBD02 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> TRB_CD4_949   AACCCGACAGC… CASSINPVQG… TCRBV… TCRBD… TCRBJ… TCRBV19  TCRBD01 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> TRB_CD4_949   AACCCGACAGC… CASSIPTGLA… TCRBV… TCRBD… TCRBJ… TCRBV19  TCRBD02 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> TRB_CD4_949   AACCTGAGCTC… CASSVEVGSA… TCRBV… TCRBD… TCRBJ… TCRBV09  TCRBD01 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> TRB_CD4_949   AACCTGAGCTC… CASSVEVAGW… TCRBV… TCRBD… TCRBJ… TCRBV09  TCRBD01 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> TRB_CD4_949   AACCTGAGCTC… CASSVMVGTE… TCRBV… TCRBD… TCRBJ… TCRBV09  TCRBD01 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> TRB_CD4_949   AACGCCTTGGA… CASSPGGTGM… TCRBV… TCRBD… TCRBJ… TCRBV05  TCRBD01 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> TRB_CD4_949   AACGCCTTGGA… CASSDSGVPG… TCRBV… TCRBD… TCRBJ… TCRBV05  TCRBD01 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> TRB_CD4_949   AACGCCTTGGA… CASSFLRGGS… TCRBV… TCRBD… TCRBJ… TCRBV05  TCRBD02 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> TRB_CD4_949   AACGCCTTGGA… CASSLATKNT… TCRBV… TCRBD… TCRBJ… TCRBV05  TCRBD01 </span>
<span class="co">#&gt; <span style="color: #949494;"># … with 7,669 more rows, and 4 more variables: j_family &lt;chr&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   reading_frame &lt;chr&gt;, duplicate_count &lt;dbl&gt;, duplicate_frequency &lt;dbl&gt;</span></span></code></pre></div>
<p>If the parameter prevalence is set to TRUE, then a new column is added to each of the data frames giving the prevalence (%) of each TCR beta CDR3 amino acid sequence in 55 healthy donor peripheral blood samples. Values range from 0 to 100% where 100% means the sequence appeared in the blood of all 55 individuals.</p>
<p>Notice in the example below that there are no amino acid sequences given in the first and fourth row of the study_table table for sample “TRB_Unsorted_949”. This is because the nucleotide sequence is out of frame and does not produce a productively transcribed amino acid sequence. If an asterisk (*) appears in the amino acid sequences, this would indicate an early stop codon.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">study_table</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
<span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">repertoire_id</span> <span class="op">==</span> <span class="st">"TRB_Unsorted_0"</span><span class="op">)</span> 
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 1,000 × 146</span></span>
<span class="co">#&gt;    sequence_id sequence   sequence_aa rev_comp productive vj_in_frame stop_codon</span>
<span class="co">#&gt;          <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span>           1 TCAATTCCC… <span style="color: #BB0000;">NA</span>          FALSE    FALSE      <span style="color: #BB0000;">NA</span>          TRUE      </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span>           2 CTGATTCTG… CASSPVSNEQ… FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span>           3 ATCAATTCC… CASSQEVPPY… FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span>           4 TGCCATCCC… <span style="color: #BB0000;">NA</span>          FALSE    FALSE      <span style="color: #BB0000;">NA</span>          TRUE      </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span>           5 CACACCCTG… CASSQEASGR… FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span>           6 GCCAGCACC… CASSLEHTGA… FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span>           7 CCCCTGACC… CASSPGDEQYF FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span>           8 AGTGCCCAT… CSARSPSTGT… FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span>           9 GGAGCTTGG… <span style="color: #BB0000;">NA</span>          FALSE    FALSE      <span style="color: #BB0000;">NA</span>          TRUE      </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span>          10 CTGTAGTCT… CASSEKREGH… FALSE    TRUE       <span style="color: #BB0000;">NA</span>          FALSE     </span>
<span class="co">#&gt; <span style="color: #949494;"># … with 990 more rows, and 139 more variables: complete_vdj &lt;lgl&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   locus &lt;chr&gt;, v_call &lt;chr&gt;, d_call &lt;chr&gt;, d2_call &lt;chr&gt;, j_call &lt;chr&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   c_call &lt;chr&gt;, sequence_alignment &lt;chr&gt;, sequence_alignment_aa &lt;chr&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   germline_alignment &lt;chr&gt;, germline_alignment_aa &lt;chr&gt;, junction &lt;chr&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   junction_aa &lt;chr&gt;, np1 &lt;chr&gt;, np1_aa &lt;chr&gt;, np2 &lt;chr&gt;, np2_aa &lt;chr&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   np3 &lt;chr&gt;, np3_aa &lt;chr&gt;, cdr1 &lt;chr&gt;, cdr1_aa &lt;chr&gt;, cdr2 &lt;chr&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   cdr2_aa &lt;chr&gt;, cdr3 &lt;chr&gt;, cdr3_aa &lt;chr&gt;, fwr1 &lt;chr&gt;, fwr1_aa &lt;chr&gt;, …</span></span></code></pre></div>
<p>After <code>productiveSeq</code> is run, the unproductive sequences are removed and the <code>duplicate_frequency</code> is recalculated for each sequence. If there were two identical amino acid sequences that differed in their nucleotide sequence, they would be combined and their counts added together.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">aa_table</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
<span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">repertoire_id</span> <span class="op">==</span> <span class="st">"TRB_Unsorted_0"</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 833 × 11</span></span>
<span class="co">#&gt;    repertoire_id  junction_aa    v_call d_call j_call v_family d_family j_family</span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> TRB_Unsorted_0 CAAGDTTLYEQYF  TCRBV… TCRBD… TCRBJ… TCRBV06  TCRBD02  TCRBJ02 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> TRB_Unsorted_0 CAARGGGESYEQYF TCRBV… TCRBD… TCRBJ… TCRBV06  TCRBD02  TCRBJ02 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> TRB_Unsorted_0 CAGGRGLEEAFF   TCRBV… TCRBD… TCRBJ… TCRBV28  TCRBD01  TCRBJ01 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> TRB_Unsorted_0 CAGQGVGYTEAFF  TCRBV… TCRBD… TCRBJ… TCRBV02  TCRBD01  TCRBJ01 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> TRB_Unsorted_0 CAGRVGYTF      TCRBV… TCRBD… TCRBJ… TCRBV02  TCRBD01  TCRBJ01 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> TRB_Unsorted_0 CAIAQPNEKLFF   TCRBV… TCRBD… TCRBJ… TCRBV10  TCRBD01  TCRBJ01 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> TRB_Unsorted_0 CAIGAAYEQYF    TCRBV… TCRBD… TCRBJ… TCRBV10  TCRBD02  TCRBJ02 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> TRB_Unsorted_0 CAIGYNQPQHF    TCRBV… TCRBD… TCRBJ… TCRBV10  TCRBD01  TCRBJ01 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> TRB_Unsorted_0 CAINPGGTGNSPL… TCRBV… TCRBD… TCRBJ… TCRBV10  TCRBD01  TCRBJ01 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> TRB_Unsorted_0 CAIRDGYNSPLHF  TCRBV… TCRBD… TCRBJ… TCRBV10  TCRBD01  TCRBJ01 </span>
<span class="co">#&gt; <span style="color: #949494;"># … with 823 more rows, and 3 more variables: reading_frame &lt;chr&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   duplicate_count &lt;dbl&gt;, duplicate_frequency &lt;dbl&gt;</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="create-a-table-of-summary-statistics">Create a table of summary statistics<a class="anchor" aria-label="anchor" href="#create-a-table-of-summary-statistics"></a>
</h2>
<p>To create a table summarizing the total number of sequences, number of unique productive sequences, number of genomes, clonality, Gini coefficient, and the frequency (%) of the top productive sequence, Simpson index, Inverse Simpson index, Hill diversity index, Chao1 index and Kemp index in each imported file, use the function <code>clonality</code>.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/clonality.html">clonality</a></span><span class="op">(</span>study_table <span class="op">=</span> <span class="va">study_table</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 10 × 12</span></span>
<span class="co">#&gt;    repertoire_id     total_sequences unique_productive_se… total_count clonality</span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                       <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> TRB_CD4_949                  <span style="text-decoration: underline;">1</span>000                   845       <span style="text-decoration: underline;">25</span>769     0.443</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> TRB_CD8_949                  <span style="text-decoration: underline;">1</span>000                   794       <span style="text-decoration: underline;">26</span>239     0.431</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> TRB_CD8_CMV_369               414                   281        <span style="text-decoration: underline;">1</span>794     0.332</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> TRB_Unsorted_0               <span style="text-decoration: underline;">1</span>000                   838       <span style="text-decoration: underline;">18</span>161     0.281</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> TRB_Unsorted_1320            <span style="text-decoration: underline;">1</span>000                   838      <span style="text-decoration: underline;">178</span>190     0.422</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> TRB_Unsorted_1496            <span style="text-decoration: underline;">1</span>000                   832       <span style="text-decoration: underline;">33</span>669     0.389</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> TRB_Unsorted_32               920                   767       <span style="text-decoration: underline;">31</span>078     0.134</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> TRB_Unsorted_369             <span style="text-decoration: underline;">1</span>000                   830      <span style="text-decoration: underline;">339</span>413     0.426</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> TRB_Unsorted_83              <span style="text-decoration: underline;">1</span>000                   823      <span style="text-decoration: underline;">236</span>732     0.338</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> TRB_Unsorted_949             <span style="text-decoration: underline;">1</span>000                   831        <span style="text-decoration: underline;">6</span>549     0.306</span>
<span class="co">#&gt; <span style="color: #949494;"># … with 7 more variables: simpson_index &lt;dbl&gt;, inverse_simpson &lt;dbl&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   gini_coefficient &lt;dbl&gt;, top_productive_sequence &lt;dbl&gt;, chao_estimate &lt;dbl&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   kemp_estimate &lt;dbl&gt;, hill_estimate &lt;dbl&gt;</span></span></code></pre></div>
<p>The clonality score is derived from the Shannon entropy, which is calculated from the frequencies of all productive sequences divided by the logarithm of the total number of unique productive sequences. This normalized entropy value is then inverted (1 - normalized entropy) to produce the clonality metric.</p>
<p>The Gini coefficient, Chao1 estimate, Kemp estimate, Hill estimate, Simpson index and Inverse Simpson index are alternative metric to measure sequence diversity within the immune repertoire.</p>
<p>The Gini coefficient is an alternative metric used to calculate repertoire diversity and is derived from the Lorenz curve. The Lorenz curve is drawn such that x-axis represents the cumulative percentage of unique sequences and the y-axis represents the cumulative percentage of reads. A line passing through the origin with a slope of 1 reflects equal frequencies of all clones. The Gini coefficient is the ratio of the area between the line of equality and the observed Lorenz curve over the total area under the line of equality.</p>
</div>
<div class="section level2">
<h2 id="calculate-clonal-relatedness">Calculate clonal relatedness<a class="anchor" aria-label="anchor" href="#calculate-clonal-relatedness"></a>
</h2>
<p>One of the drawbacks of the clonality metric is that it does not take into account sequence similarity. This is particularly important when studying affinity maturation or B cell malignancies(Lombardo, K.A., et al. Blood Advances 2017 1:535-544). Clonal relatedness is a useful metric that takes into account sequence similarity without regard for clonal frequency. It is defined as the proportion of nucleotide sequences that are related by a defined edit distance threshold. The value ranges from 0 to 1 where 0 indicates no sequences are related and 1 indicates all sequences are related. Edit distance is a way of quantifying how dissimilar two sequences are to one another by counting the minimum number of operations required to transform one sequence into the other. For example, an edit distance of 0 means the sequences are identical and an edit distance of 1 indicates that the sequences different by a single amino acid or nucleotide.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">IGH_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"IGH_sequencing"</span>, package <span class="op">=</span> <span class="st">"LymphoSeq2"</span><span class="op">)</span>
<span class="va">IGH_table</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/readImmunoSeq.html">readImmunoSeq</a></span><span class="op">(</span>path <span class="op">=</span> <span class="va">IGH_path</span><span class="op">)</span>
<span class="co">#&gt; <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">1</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">144</span></span>
<span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="co">#&gt; <span style="font-weight: bold;">Delimiter:</span> ","</span>
<span class="co">#&gt; <span style="color: #BB0000;">chr</span> (69): sequence_id, sequence, sequence_aa, locus, v_call, d_call, d2_call...</span>
<span class="co">#&gt; <span style="color: #00BB00;">dbl</span> (70): v_score, v_identity, v_support, d_score, d_identity, d_support, d2...</span>
<span class="co">#&gt; <span style="color: #BBBB00;">lgl</span>  (5): rev_comp, productive, vj_in_frame, stop_codon, complete_vdj</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="co">#&gt; <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">694</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">52</span></span>
<span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span>
<span class="co">#&gt; <span style="font-weight: bold;">Delimiter:</span> "\t"</span>
<span class="co">#&gt; <span style="color: #BB0000;">chr</span> (25): nucleotide, aminoAcid, vMaxResolved, vFamilyName, vGeneName, vGene...</span>
<span class="co">#&gt; <span style="color: #00BB00;">dbl</span> (17): count (reads), frequencyCount (%), cdr3Length, vDeletion, n1Insert...</span>
<span class="co">#&gt; <span style="color: #BBBB00;">lgl</span> (10): vFamilyTies, vOrphon, dOrphon, jOrphon, vFunction, dFunction, jFun...</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="co">#&gt; Warning: Expected 2 pieces. Additional pieces discarded in 41 rows [14, 15, 33,</span>
<span class="co">#&gt; 36, 48, 78, 119, 123, 130, 135, 149, 167, 176, 190, 198, 210, 245, 247, 250,</span>
<span class="co">#&gt; 262, ...].</span>
<span class="co">#&gt; Joining, by = c("sequence", "sequence_aa", "v_call", "d_call", "d2_call",</span>
<span class="co">#&gt; "j_call", "junction", "junction_aa", "duplicate_count", "clone_id",</span>
<span class="co">#&gt; "repertoire_id")</span>
<span class="co">#&gt; <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">1000</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">52</span></span>
<span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span></span>
<span class="co">#&gt; <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span> <span style="font-weight: bold;">Delimiter:</span> "\t" <span style="color: #BB0000;">chr</span></span>
<span class="co">#&gt; (25): nucleotide, aminoAcid, vMaxResolved, vFamilyName, vGeneName, vGene... <span style="color: #00BB00;">dbl</span></span>
<span class="co">#&gt; (17): count (reads), frequencyCount (%), cdr3Length, vDeletion, n1Insert... <span style="color: #BBBB00;">lgl</span></span>
<span class="co">#&gt; (8): vFamilyTies, vOrphon, dOrphon, jOrphon, vFunction, dFunction, jFun...</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data. <span style="color: #00BBBB;">ℹ</span></span>
<span class="co">#&gt; Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="co">#&gt; Warning: Expected 2 pieces. Additional pieces discarded in 58 rows [31, 33, 40,</span>
<span class="co">#&gt; 41, 90, 96, 109, 117, 146, 154, 178, 189, 238, 252, 255, 260, 270, 278, 315,</span>
<span class="co">#&gt; 320, ...].</span>
<span class="co">#&gt; Joining, by = c("sequence", "sequence_aa", "v_call", "d_call", "d2_call",</span>
<span class="co">#&gt; "j_call", "junction", "junction_aa", "duplicate_count", "clone_id",</span>
<span class="co">#&gt; "repertoire_id")</span>
<span class="co">#&gt; <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">694</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">52</span></span>
<span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span></span>
<span class="co">#&gt; <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span> <span style="font-weight: bold;">Delimiter:</span> "\t" <span style="color: #BB0000;">chr</span></span>
<span class="co">#&gt; (25): nucleotide, aminoAcid, vMaxResolved, vFamilyName, vGeneName, vGene... <span style="color: #00BB00;">dbl</span></span>
<span class="co">#&gt; (17): count (reads), frequencyCount (%), cdr3Length, vDeletion, n1Insert... <span style="color: #BBBB00;">lgl</span></span>
<span class="co">#&gt; (10): vFamilyTies, vOrphon, dOrphon, jOrphon, vFunction, dFunction, jFun...</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data. <span style="color: #00BBBB;">ℹ</span></span>
<span class="co">#&gt; Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="co">#&gt; Warning: Expected 2 pieces. Additional pieces discarded in 41 rows [14, 15, 33,</span>
<span class="co">#&gt; 36, 48, 78, 119, 123, 130, 135, 149, 167, 176, 190, 198, 210, 245, 247, 250,</span>
<span class="co">#&gt; 262, ...].</span>
<span class="co">#&gt; Joining, by = c("sequence", "sequence_aa", "v_call", "d_call", "d2_call",</span>
<span class="co">#&gt; "j_call", "junction", "junction_aa", "duplicate_count", "clone_id",</span>
<span class="co">#&gt; "repertoire_id")</span>
<span class="co">#&gt; <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">694</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">52</span></span>
<span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span></span>
<span class="co">#&gt; <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span> <span style="font-weight: bold;">Delimiter:</span> "\t" <span style="color: #BB0000;">chr</span></span>
<span class="co">#&gt; (26): nucleotide, aminoAcid, vMaxResolved, vFamilyName, vGeneName, vGene... <span style="color: #00BB00;">dbl</span></span>
<span class="co">#&gt; (17): count (reads), frequencyCount (%), cdr3Length, vDeletion, n1Insert... <span style="color: #BBBB00;">lgl</span></span>
<span class="co">#&gt; (9): vOrphon, dOrphon, jOrphon, vFunction, dFunction, jFunction, fracti...</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data. <span style="color: #00BBBB;">ℹ</span></span>
<span class="co">#&gt; Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="co">#&gt; Warning: Expected 2 pieces. Additional pieces discarded in 10 rows [204, 206,</span>
<span class="co">#&gt; 265, 347, 410, 411, 419, 512, 582, 608].</span>
<span class="co">#&gt; Joining, by = c("sequence", "sequence_aa", "v_call", "d_call", "d2_call",</span>
<span class="co">#&gt; "j_call", "junction", "junction_aa", "duplicate_count", "clone_id",</span>
<span class="co">#&gt; "repertoire_id")</span>
<span class="co">#&gt; <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">492</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">52</span></span>
<span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span></span>
<span class="co">#&gt; <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span> <span style="font-weight: bold;">Delimiter:</span> "\t" <span style="color: #BB0000;">chr</span></span>
<span class="co">#&gt; (25): nucleotide, aminoAcid, vMaxResolved, vFamilyName, vGeneName, vGene... <span style="color: #00BB00;">dbl</span></span>
<span class="co">#&gt; (18): count (reads), frequencyCount (%), cdr3Length, vDeletion, n1Insert... <span style="color: #BBBB00;">lgl</span></span>
<span class="co">#&gt; (9): jGeneAlleleTies, vOrphon, dOrphon, jOrphon, vFunction, dFunction, ...</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data. <span style="color: #00BBBB;">ℹ</span></span>
<span class="co">#&gt; Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="co">#&gt; Warning: Expected 2 pieces. Additional pieces discarded in 3 rows [134, 143,</span>
<span class="co">#&gt; 251].</span>
<span class="co">#&gt; Joining, by = c("sequence", "sequence_aa", "v_call", "d_call", "d2_call",</span>
<span class="co">#&gt; "j_call", "junction", "junction_aa", "duplicate_count", "clone_id",</span>
<span class="co">#&gt; "repertoire_id")</span>
<span class="co">#&gt; <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">209</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">52</span></span>
<span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span></span>
<span class="co">#&gt; <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span> <span style="font-weight: bold;">Delimiter:</span> "\t" <span style="color: #BB0000;">chr</span></span>
<span class="co">#&gt; (25): nucleotide, aminoAcid, vMaxResolved, vFamilyName, vGeneName, vGene... <span style="color: #00BB00;">dbl</span></span>
<span class="co">#&gt; (17): count (reads), frequencyCount (%), cdr3Length, vDeletion, n1Insert... <span style="color: #BBBB00;">lgl</span></span>
<span class="co">#&gt; (10): jGeneAlleleTies, vOrphon, dOrphon, jOrphon, vFunction, dFunction, ...</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data. <span style="color: #00BBBB;">ℹ</span></span>
<span class="co">#&gt; Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="co">#&gt; Warning: Expected 2 pieces. Additional pieces discarded in 20 rows [4, 27, 34,</span>
<span class="co">#&gt; 37, 52, 53, 55, 69, 81, 87, 88, 90, 95, 108, 111, 131, 151, 158, 160, 200].</span>
<span class="co">#&gt; Joining, by = c("sequence", "sequence_aa", "v_call", "d_call", "d2_call",</span>
<span class="co">#&gt; "j_call", "junction", "junction_aa", "duplicate_count", "clone_id",</span>
<span class="co">#&gt; "repertoire_id")</span>
<span class="co">#&gt; <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">436</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">52</span></span>
<span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span></span>
<span class="co">#&gt; <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span> <span style="font-weight: bold;">Delimiter:</span> "\t" <span style="color: #BB0000;">chr</span></span>
<span class="co">#&gt; (25): nucleotide, aminoAcid, vMaxResolved, vFamilyName, vGeneName, vGene... <span style="color: #00BB00;">dbl</span></span>
<span class="co">#&gt; (17): count (reads), frequencyCount (%), cdr3Length, vDeletion, n1Insert... <span style="color: #BBBB00;">lgl</span></span>
<span class="co">#&gt; (10): jGeneAlleleTies, vOrphon, dOrphon, jOrphon, vFunction, dFunction, ...</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data. <span style="color: #00BBBB;">ℹ</span></span>
<span class="co">#&gt; Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="co">#&gt; Warning: Expected 2 pieces. Additional pieces discarded in 47 rows [21, 22,</span>
<span class="co">#&gt; 28, 59, 63, 69, 78, 79, 82, 87, 90, 91, 116, 121, 149, 170, 182, 188, 216,</span>
<span class="co">#&gt; 237, ...].</span>
<span class="co">#&gt; Joining, by = c("sequence", "sequence_aa", "v_call", "d_call", "d2_call",</span>
<span class="co">#&gt; "j_call", "junction", "junction_aa", "duplicate_count", "clone_id",</span>
<span class="co">#&gt; "repertoire_id")</span>
<span class="co">#&gt; <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">1000</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">52</span></span>
<span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span></span>
<span class="co">#&gt; <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span> <span style="font-weight: bold;">Delimiter:</span> "\t" <span style="color: #BB0000;">chr</span></span>
<span class="co">#&gt; (26): nucleotide, aminoAcid, vMaxResolved, vFamilyName, vGeneName, vGene... <span style="color: #00BB00;">dbl</span></span>
<span class="co">#&gt; (17): count (reads), frequencyCount (%), cdr3Length, vDeletion, n1Insert... <span style="color: #BBBB00;">lgl</span></span>
<span class="co">#&gt; (9): vOrphon, dOrphon, jOrphon, vFunction, dFunction, jFunction, fracti...</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data. <span style="color: #00BBBB;">ℹ</span></span>
<span class="co">#&gt; Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="co">#&gt; Warning: Expected 2 pieces. Additional pieces discarded in 27 rows [117, 121,</span>
<span class="co">#&gt; 146, 157, 178, 199, 296, 310, 322, 323, 324, 325, 349, 351, 363, 420, 421, 467,</span>
<span class="co">#&gt; 468, 484, ...].</span>
<span class="co">#&gt; Joining, by = c("sequence", "sequence_aa", "v_call", "d_call", "d2_call",</span>
<span class="co">#&gt; "j_call", "junction", "junction_aa", "duplicate_count", "clone_id",</span>
<span class="co">#&gt; "repertoire_id")</span>
<span class="co">#&gt; <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">1000</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">52</span></span>
<span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span></span>
<span class="co">#&gt; <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span> <span style="font-weight: bold;">Delimiter:</span> "\t" <span style="color: #BB0000;">chr</span></span>
<span class="co">#&gt; (26): nucleotide, aminoAcid, vMaxResolved, vFamilyName, vGeneName, vGene... <span style="color: #00BB00;">dbl</span></span>
<span class="co">#&gt; (18): count (reads), frequencyCount (%), cdr3Length, vDeletion, n1Insert... <span style="color: #BBBB00;">lgl</span></span>
<span class="co">#&gt; (8): vOrphon, dOrphon, jOrphon, vFunction, dFunction, jFunction, vAlign...</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data. <span style="color: #00BBBB;">ℹ</span></span>
<span class="co">#&gt; Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="co">#&gt; Warning: Expected 2 pieces. Additional pieces discarded in 85 rows [38, 58, 79,</span>
<span class="co">#&gt; 83, 92, 119, 127, 145, 149, 161, 162, 169, 187, 191, 199, 237, 250, 272, 275,</span>
<span class="co">#&gt; 283, ...].</span>
<span class="co">#&gt; Joining, by = c("sequence", "sequence_aa", "v_call", "d_call", "d2_call",</span>
<span class="co">#&gt; "j_call", "junction", "junction_aa", "duplicate_count", "clone_id",</span>
<span class="co">#&gt; "repertoire_id")</span>
<span class="co">#&gt; <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">275</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">52</span></span>
<span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span></span>
<span class="co">#&gt; <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span> <span style="font-weight: bold;">Delimiter:</span> "\t" <span style="color: #BB0000;">chr</span></span>
<span class="co">#&gt; (24): nucleotide, aminoAcid, vMaxResolved, vFamilyName, vGeneName, vGene... <span style="color: #00BB00;">dbl</span></span>
<span class="co">#&gt; (18): count (reads), frequencyCount (%), cdr3Length, vDeletion, n1Insert... <span style="color: #BBBB00;">lgl</span></span>
<span class="co">#&gt; (10): vFamilyTies, jGeneAlleleTies, vOrphon, dOrphon, jOrphon, vFunction...</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data. <span style="color: #00BBBB;">ℹ</span></span>
<span class="co">#&gt; Specify the column types or set `show_col_types = FALSE` to quiet this message.</span>
<span class="co">#&gt; Warning: Expected 2 pieces. Additional pieces discarded in 24 rows [9, 29, 40,</span>
<span class="co">#&gt; 42, 61, 84, 87, 101, 104, 106, 108, 119, 146, 170, 177, 192, 201, 206, 214,</span>
<span class="co">#&gt; 248, ...].</span>
<span class="co">#&gt; Joining, by = c("sequence", "sequence_aa", "v_call", "d_call", "d2_call",</span>
<span class="co">#&gt; "j_call", "junction", "junction_aa", "duplicate_count", "clone_id",</span>
<span class="co">#&gt; "repertoire_id")</span>
<span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/clonalRelatedness.html">clonalRelatedness</a></span><span class="op">(</span>study_table <span class="op">=</span> <span class="va">IGH_table</span>, editDistance <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 10 × 2</span></span>
<span class="co">#&gt;    repertoire_id     clonalRelatedness</span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> IGH_MVQ108911A_BL           0.631  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> IGH_MVQ194745A_BL           0.847  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> IGH_MVQ81231A_BL            0.631  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> IGH_MVQ89037A_BL            0.290  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> IGH_MVQ90143A_BL            0.006<span style="text-decoration: underline;">10</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> IGH_MVQ92552A_BL            0.273  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> IGH_MVQ93505A_BL            0.408  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> IGH_MVQ93631A_BL            0.757  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> IGH_MVQ94865A_BL            0.007  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> IGH_MVQ95413A_BL            0.003<span style="text-decoration: underline;">64</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="draw-a-phylogenetic-tree">Draw a phylogenetic tree<a class="anchor" aria-label="anchor" href="#draw-a-phylogenetic-tree"></a>
</h2>
<p>A phylogenetic tree is a useful way to visualize the similarity between sequences. The <code>phyloTree</code> function create a phylogenetic tree of a single sample using neighbor joining tree estimation for amino acid or nucleotide CDR3 sequences. Each leaf in the tree represents a sequence color coded by the V, D, and J gene usage. The number next to each leaf refers to the sequence count. A triangle shaped leaf indicates the most frequent sequence. The distance between leaves on the horizontal axis corresponds to the sequence similarity (i.e. the further apart the leaves are horizontally, the less similar the sequences are to one another).</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1"></a>nuc_IGH_table &lt;-<span class="st"> </span>LymphoSeq2<span class="op">::</span><span class="kw">productiveSeq</span>(<span class="dt">study_table =</span> IGH_table, <span class="dt">aggregate =</span> <span class="st">"junction"</span>)</span>
<span id="cb16-2"><a href="#cb16-2"></a>LymphoSeq2<span class="op">::</span><span class="kw">phyloTree</span>(<span class="dt">study_table =</span> nuc_IGH_table, </span>
<span id="cb16-3"><a href="#cb16-3"></a>                      <span class="dt">repertoire_ids =</span> <span class="st">"IGH_MVQ92552A_BL"</span>, </span>
<span id="cb16-4"><a href="#cb16-4"></a>                      <span class="dt">type =</span> <span class="st">"junction"</span>, </span>
<span id="cb16-5"><a href="#cb16-5"></a>                      <span class="dt">layout =</span> <span class="st">"rectangular"</span>)</span>
<span id="cb16-6"><a href="#cb16-6"></a>Found more than one class <span class="st">"phylo"</span> <span class="cf">in</span> cache; using the first, from namespace <span class="st">'phyloseq'</span></span>
<span id="cb16-7"><a href="#cb16-7"></a>Also defined by <span class="st">'tidytree'</span></span>
<span id="cb16-8"><a href="#cb16-8"></a>Warning<span class="op">:</span><span class="st"> `</span><span class="dt">guides(&lt;scale&gt; = FALSE)</span><span class="st">`</span> is deprecated. Please use <span class="st">`</span><span class="dt">guides(&lt;scale&gt; =</span></span>
<span id="cb16-9"><a href="#cb16-9"></a><span class="dt">"none")</span><span class="st">`</span> instead.</span></code></pre></div>
<p><img src="LymphoSeq2_files/figure-html/getIGHProd-1.png" width="672"></p>
</div>
<div class="section level2">
<h2 id="multiple-sequence-alignment">Multiple sequence alignment<a class="anchor" aria-label="anchor" href="#multiple-sequence-alignment"></a>
</h2>
<p>In LymphoSeq2, you can perform a multiple sequence alignment using one of three methods provided by the Bioconductor msa package (ClustalW, ClustalOmega, or Muscle), the chaange in functionality however is, now the function returns a <code>msa</code> S4 object. One may perform the alignment of all amino acid or nucleotide sequences in a single sample. Alternatively, one may search for a given sequence within a list of samples using an edit distance threshold.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">alignment</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/alignSeq.html">alignSeq</a></span><span class="op">(</span>study_table <span class="op">=</span> <span class="va">nuc_IGH_table</span>, 
                                  repertoire_ids <span class="op">=</span> <span class="st">"IGH_MVQ92552A_BL"</span>, 
                                  type <span class="op">=</span> <span class="st">"junction_aa"</span>, 
                                  method <span class="op">=</span> <span class="st">"ClustalW"</span><span class="op">)</span></code></pre></div>
<p>use default substitution matrix</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">msa</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/msa/man/msaPrettyPrint.html" class="external-link">msaPrettyPrint</a></span><span class="op">(</span><span class="va">alignment</span>, output <span class="op">=</span> <span class="st">"asis"</span><span class="op">)</span></code></pre></div>

</div>
<div class="section level2">
<h2 id="searching-for-sequences">Searching for sequences<a class="anchor" aria-label="anchor" href="#searching-for-sequences"></a>
</h2>
<p>To search for one or more amino acid or nucleotide CDR3 sequences in a list of data frames, use the function <code>searchSeq</code>. The function allows sequence search with a edit distance threshold. For example, an edit distance of 0 means the sequences are identical and an edit distance of 1 indicates that the sequences differ by a single amino acid or nucleotide. Match options include “global” matching which performs end-to-end matching of sequences. “partial” matching allows searching for sub strings with CDR3 sequences.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/searchSeq.html">searchSeq</a></span><span class="op">(</span>study_table <span class="op">=</span> <span class="va">aa_table</span>, 
                      sequence <span class="op">=</span> <span class="st">"CASSPVSNEQFF"</span>, 
                      seq_type <span class="op">=</span> <span class="st">"junction_aa"</span>, 
                      match <span class="op">=</span> <span class="st">"global"</span>, 
                      edit_distance <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 2 × 13</span></span>
<span class="co">#&gt;   repertoire_id  junction_aa  v_call    d_call j_call v_family d_family j_family</span>
<span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">1</span> TRB_CD8_949    CASSPVSNEQFF TCRBV28-… TCRBD… TCRBJ… TCRBV28  TCRBD02  TCRBJ02 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">2</span> TRB_Unsorted_0 CASSPVSNEQFF TCRBV28-… TCRBD… TCRBJ… TCRBV28  TCRBD02  TCRBJ02 </span>
<span class="co">#&gt; <span style="color: #949494;"># … with 5 more variables: reading_frame &lt;chr&gt;, duplicate_count &lt;dbl&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   duplicate_frequency &lt;dbl&gt;, edit_distance &lt;dbl&gt;, searchSequence &lt;chr&gt;</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="searching-for-published-sequences">Searching for published sequences<a class="anchor" aria-label="anchor" href="#searching-for-published-sequences"></a>
</h2>
<p>To search your entire list of data frames for a published amino acid CDR3 TCRB sequence with known antigen specificity, use the function <code>searchPublished</code>.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/searchPublished.html">searchPublished</a></span><span class="op">(</span>study_table <span class="op">=</span> <span class="va">aa_table</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
<span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">PMID</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 106 × 16</span></span>
<span class="co">#&gt;    repertoire_id junction_aa     v_call d_call j_call v_family d_family j_family</span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> TRB_CD4_949   CASSLYGEQYF     TCRBV… TCRBD… TCRBJ… TCRBV28  TCRBD01  TCRBJ02 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> TRB_CD4_949   CASSQDPGYEQYF   TCRBV… TCRBD… TCRBJ… TCRBV04  TCRBD01  TCRBJ02 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> TRB_CD4_949   CASSYLGNTGELFF  TCRBV… TCRBD… TCRBJ… TCRBV06  TCRBD01  TCRBJ02 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> TRB_CD8_949   CASSGQGAYGYTF   TCRBV… TCRBD… TCRBJ… TCRBV05  TCRBD01  TCRBJ01 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> TRB_CD8_949   CASSIGLYGYTF    TCRBV… TCRBD… TCRBJ… TCRBV19  TCRBD02  TCRBJ01 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> TRB_CD8_949   CASSIGPLEGNEQFF TCRBV… TCRBD… TCRBJ… TCRBV19  TCRBD02  TCRBJ02 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> TRB_CD8_949   CASSIGTYGYTF    TCRBV… TCRBD… TCRBJ… TCRBV19  TCRBD01  TCRBJ01 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> TRB_CD8_949   CASSIGVYGYTF    TCRBV… TCRBD… TCRBJ… TCRBV19  TCRBD02  TCRBJ01 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> TRB_CD8_949   CASSIRSAYEQYF   TCRBV… TCRBD… TCRBJ… TCRBV19  TCRBD02  TCRBJ02 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> TRB_CD8_949   CASSLEGDQPQHF   TCRBV… TCRBD… TCRBJ… TCRBV05  TCRBD01  TCRBJ01 </span>
<span class="co">#&gt; <span style="color: #949494;"># … with 96 more rows, and 8 more variables: reading_frame &lt;chr&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   duplicate_count &lt;dbl&gt;, duplicate_frequency &lt;dbl&gt;, PMID &lt;fct&gt;, HLA &lt;fct&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   antigen &lt;fct&gt;, epitope &lt;fct&gt;, prevalence &lt;dbl&gt;</span></span></code></pre></div>
<p>For each found sequence, a table is provides listing the antigen, epitope, HLA type, PubMed ID (PMID), and prevalence (%) of the sequence among 55 healthy donor blood samples.</p>
</div>
<div class="section level2">
<h2 id="visualizing-repertoire-diversity">Visualizing repertoire diversity<a class="anchor" aria-label="anchor" href="#visualizing-repertoire-diversity"></a>
</h2>
<p>Antigen receptor repertoire diversity can be characterized by a number such as clonality or Gini coefficient calculated by the <code>clonality</code> function. Alternatively, you can visualize the repertoire diversity by plotting the Lorenz curve for each sample as defined above. In this plot, the more diverse samples will appear near the dotted diagonal line (the line of equality) whereas the more clonal samples will appear to have a more bowed shape.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">samples</span> <span class="op">&lt;-</span> <span class="va">aa_table</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
           <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">repertoire_id</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/duplicated.html" class="external-link">unique</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/lorenzCurve.html">lorenzCurve</a></span><span class="op">(</span>repertoire_ids <span class="op">=</span> <span class="va">samples</span>, study_table <span class="op">=</span> <span class="va">aa_table</span><span class="op">)</span></code></pre></div>
<p><img src="LymphoSeq2_files/figure-html/plotLorenz-1.png" width="700"></p>
<p>Alternatively, you can get a feel for the repertoire diversity by plotting the cumulative frequency of a selected number of the top most frequent clones using the function <code>topSeqsPlot</code>. In this case, each of the top sequences are represented by a different color and all less frequent clones will be assigned a single color (violet).</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/topSeqsPlot.html">topSeqsPlot</a></span><span class="op">(</span>study_table <span class="op">=</span> <span class="va">aa_table</span>, top <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></code></pre></div>
<p><img src="LymphoSeq2_files/figure-html/topSeqPlot-1.png" width="700"></p>
<p>Both of these functions are built using the ggplot2 package. You can reformat the plot using ggplot2 functions. Please refer to the <code>lorenzCurve</code> and <code>topSeqsPlot</code> manual for specific examples.</p>
</div>
<div class="section level2">
<h2 id="comparing-samples">Comparing samples<a class="anchor" aria-label="anchor" href="#comparing-samples"></a>
</h2>
<p>To compare the T or B cell repertoires of all samples in a pairwise fashion, use the <code>bhattacharyyaMatrix</code> or <code>similarityMatrix</code> functions. Both the Bhattacharyya coefficient and similarity score are measures of the amount of overlap between two samples. The value for each ranges from 0 to 1 where 1 indicates the sequence frequencies are identical in the two samples and 0 indicates no shared frequencies. The Bhattacharyya coefficient differs from the similarity score in that it involves weighting each shared sequence in the two distributions by the arithmetic mean of the frequency of each sequence, while calculating the similarity scores involves weighting each shared sequence in the two distributions by the geometric mean of the frequency of each sequence in the two distributions.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bhattacharyya_matrix</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/scoringMatrix.html">scoringMatrix</a></span><span class="op">(</span><span class="va">aa_table</span>, mode <span class="op">=</span> <span class="st">"Bhattacharyya"</span><span class="op">)</span>
<span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/pairwisePlot.html">pairwisePlot</a></span><span class="op">(</span><span class="va">bhattacharyya_matrix</span><span class="op">)</span></code></pre></div>
<p><img src="LymphoSeq2_files/figure-html/bhattacharyya-1.png" width="700"></p>
<p>To view sequences shared between two or more samples, use the function <code>commonSeqs</code>. This function requires that a productive amino acid list be specified.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">common</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/commonSeqs.html">commonSeqs</a></span><span class="op">(</span>study_table <span class="op">=</span> <span class="va">aa_table</span>, 
                                 repertoire_ids <span class="op">=</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TRB_Unsorted_0"</span>, <span class="st">"TRB_Unsorted_32"</span><span class="op">)</span><span class="op">)</span>
<span class="va">common</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 4 × 3</span></span>
<span class="co">#&gt;   junction_aa     TRB_Unsorted_0 TRB_Unsorted_32</span>
<span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;">1</span> CAISEGNYGYTF          0.000<span style="text-decoration: underline;">351</span>        0.001<span style="text-decoration: underline;">87</span> </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">2</span> CASSFGIQETQYF         0.000<span style="text-decoration: underline;">140</span>        0.000<span style="text-decoration: underline;">957</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;">3</span> CASSQDRTGQYGYTF       0.004<span style="text-decoration: underline;">71</span>         0.008<span style="text-decoration: underline;">06</span> </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">4</span> CAWTGGTTEAFF          0.001<span style="text-decoration: underline;">05</span>         0.001<span style="text-decoration: underline;">52</span></span></code></pre></div>
<p>To visualize the number of overlapping sequences between two or three samples in the form of a Venn diagram, use the function <code>commonSeqVenn</code></p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/commonSeqsVenn.html">commonSeqsVenn</a></span><span class="op">(</span>repertoire_ids <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TRB_Unsorted_32"</span>, <span class="st">"TRB_Unsorted_83"</span><span class="op">)</span>, 
                           productive_aa <span class="op">=</span> <span class="va">aa_table</span><span class="op">)</span></code></pre></div>
<p><img src="LymphoSeq2_files/figure-html/commonVenn-1.png" width="480"></p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/commonSeqsVenn.html">commonSeqsVenn</a></span><span class="op">(</span>repertoire_ids <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TRB_Unsorted_0"</span>, <span class="st">"TRB_Unsorted_32"</span>, <span class="st">"TRB_Unsorted_83"</span><span class="op">)</span>, 
                           productive_aa <span class="op">=</span> <span class="va">aa_table</span><span class="op">)</span></code></pre></div>
<p><img src="LymphoSeq2_files/figure-html/commonVennT-1.png" width="480"></p>
<p>To compare the frequency of sequences between two samples as a scatter plot, use the function <code>commonSeqsPlot</code>.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/commonSeqsPlot.html">commonSeqsPlot</a></span><span class="op">(</span><span class="st">"TRB_Unsorted_32"</span>, <span class="st">"TRB_Unsorted_83"</span>, 
                           productive_aa <span class="op">=</span> <span class="va">aa_table</span>, show <span class="op">=</span> <span class="st">"common"</span><span class="op">)</span></code></pre></div>
<p><img src="LymphoSeq2_files/figure-html/commonSeqPlot-1.png" width="700"></p>
<p>If you have more than 3 samples to compare, use the <code>commonSeqBar</code> function. You can chose to color a single sample with the <code>color.sample</code> argument or a desired intersection with the <code>color.intersection</code> argument.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/commonSeqsBar.html">commonSeqsBar</a></span><span class="op">(</span>productive_aa <span class="op">=</span> <span class="va">aa_table</span>, 
                          repertoire_ids <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TRB_CD4_949"</span>, <span class="st">"TRB_CD8_949"</span>, 
                                             <span class="st">"TRB_Unsorted_949"</span>, <span class="st">"TRB_Unsorted_1320"</span><span class="op">)</span>, 
                          color_sample <span class="op">=</span> <span class="st">"TRB_CD8_949"</span>,
                          labels <span class="op">=</span> <span class="st">"no"</span><span class="op">)</span></code></pre></div>
<p><img src="LymphoSeq2_files/figure-html/commonBar-1.png" width="700"> # Differential abundance</p>
<p>When comparing a sample from two different time points, it is useful to identify sequences that are significantly more or less abundant in one versus the other time point (DeWitt, W.S., et al. Journal of Virology 2015 89(8):4517-4526). The <code>differentialAbundance</code> function uses a Fisher exact test to calculate differential abundance of each sequence in two time points and reports the log2 transformed fold change, P value and adjusted P value.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/differentialAbundance.html">differentialAbundance</a></span><span class="op">(</span>study_table <span class="op">=</span> <span class="va">aa_table</span>, 
                                  repertoire_ids <span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TRB_Unsorted_949"</span>, 
                                                    <span class="st">"TRB_Unsorted_1320"</span><span class="op">)</span>, 
                                  type <span class="op">=</span> <span class="st">"junction_aa"</span>, q <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 1,332 × 6</span></span>
<span class="co">#&gt;    junction_aa       TRB_Unsorted_949 TRB_Unsorted_1320        p       q    l2fc</span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> CACALGDGYTF                      0                 2  1   <span style="color: #949494;">e</span>+0 1   <span style="color: #949494;">e</span>+0 -<span style="color: #BB0000;">Inf</span>   </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> CACQRTGSSYEQYF                   0                 3  1   <span style="color: #949494;">e</span>+0 1   <span style="color: #949494;">e</span>+0 -<span style="color: #BB0000;">Inf</span>   </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> CAIGLSNQPQHF                     2                67  1   <span style="color: #949494;">e</span>+0 1   <span style="color: #949494;">e</span>+0   -<span style="color: #BB0000;">5.07</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> CAIKMETPNGEQYF                  29               326  2.66<span style="color: #949494;">e</span><span style="color: #BB0000;">-5</span> 2.66<span style="color: #949494;">e</span><span style="color: #BB0000;">-5</span>   -<span style="color: #BB0000;">3.49</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> CAIRGTEDNNSPLHF                  0                16  1   <span style="color: #949494;">e</span>+0 1   <span style="color: #949494;">e</span>+0 -<span style="color: #BB0000;">Inf</span>   </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> CAISDSSYEQYF                     1                23  5.71<span style="color: #949494;">e</span><span style="color: #BB0000;">-1</span> 5.71<span style="color: #949494;">e</span><span style="color: #BB0000;">-1</span>   -<span style="color: #BB0000;">4.52</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> CAISDTGELFF                      8                83  1.39<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> 1.39<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span>   -<span style="color: #BB0000;">3.38</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> CAISEFGLMAREYGYTF                0                 1  1   <span style="color: #949494;">e</span>+0 1   <span style="color: #949494;">e</span>+0 -<span style="color: #BB0000;">Inf</span>   </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> CAISEGQGVKPQHF                   0               167  4.92<span style="color: #949494;">e</span><span style="color: #BB0000;">-3</span> 4.92<span style="color: #949494;">e</span><span style="color: #BB0000;">-3</span> -<span style="color: #BB0000;">Inf</span>   </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> CAISESGVLNEKLFF                 13               150  4.69<span style="color: #949494;">e</span><span style="color: #BB0000;">-3</span> 4.69<span style="color: #949494;">e</span><span style="color: #BB0000;">-3</span>   -<span style="color: #BB0000;">3.53</span></span>
<span class="co">#&gt; <span style="color: #949494;"># … with 1,322 more rows</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="finding-recurring-sequences">Finding recurring sequences<a class="anchor" aria-label="anchor" href="#finding-recurring-sequences"></a>
</h2>
<p>To create a tibble of unique, productive amino acid sequences as rows and sample names as headers use the <code>seqMatrix</code> function. Each value in the data frame represents the frequency that each sequence appears in the sample. You can specify your own list of sequences or all unique sequences in the list using the output of the function <code>uniqueSeqs.</code> The <code>uniqueSeqs</code> function creates a tibble of all unique, productive sequences and reports the total count in all samples.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">unique_seqs</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/uniqueSeqs.html">uniqueSeqs</a></span><span class="op">(</span>productive_table <span class="op">=</span> <span class="va">aa_table</span><span class="op">)</span>
<span class="va">unique_seqs</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 5,615 × 2</span></span>
<span class="co">#&gt;    junction_aa            duplicate_count</span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                            <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> CASSQDWERLGEQFF                  <span style="text-decoration: underline;">99</span>480</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> CASSLQGREKLFF                    <span style="text-decoration: underline;">90</span>567</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> CASSQDLMTVDSLFAGANVLTF           <span style="text-decoration: underline;">68</span>682</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> CASSPAGAYYNEQFF                  <span style="text-decoration: underline;">30</span>454</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> CASSPPTGERDTQYF                  <span style="text-decoration: underline;">24</span>703</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> CASSLAGDSQETQYF                  <span style="text-decoration: underline;">22</span>147</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> CASSESAGSTGELFF                  <span style="text-decoration: underline;">17</span>438</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> CASRDGQGSGNTIYF                  <span style="text-decoration: underline;">11</span>533</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> CASSPSRNTEAFF                     <span style="text-decoration: underline;">8</span>759</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> CASSQDRTGQYGYTF                   <span style="text-decoration: underline;">8</span>118</span>
<span class="co">#&gt; <span style="color: #949494;"># … with 5,605 more rows</span></span></code></pre></div>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sequence_matrix</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/seqMatrix.html">seqMatrix</a></span><span class="op">(</span>productive_aa <span class="op">=</span> <span class="va">aa_table</span>, sequences <span class="op">=</span> <span class="va">unique_seqs</span><span class="op">$</span><span class="va">junction_aa</span>, <span class="op">)</span>
<span class="va">sequence_matrix</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 5,615 × 11</span></span>
<span class="co">#&gt;    junction_aa      TRB_CD4_949 TRB_CD8_949 TRB_CD8_CMV_369 TRB_Unsorted_0</span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> CAIKPGQGASNSPLHF    0.000<span style="text-decoration: underline;">180</span>           0               0              0</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> CAIRAGTSTDTQYF      0.000<span style="text-decoration: underline;">180</span>           0               0              0</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> CAISDETPGELFF       0.000<span style="text-decoration: underline;">120</span>           0               0              0</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> CAISDLGRGDTEAFF     0.000<span style="text-decoration: underline;">120</span>           0               0              0</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> CAISDLKEQPQHF       0.000<span style="text-decoration: underline;">120</span>           0               0              0</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> CAISDQGGDQPQHF      0.000<span style="text-decoration: underline;">120</span>           0               0              0</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> CAISEREQGAFF        0.000<span style="text-decoration: underline;">120</span>           0               0              0</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> CAISEWSGSSYNEQFF    0.000<span style="text-decoration: underline;">240</span>           0               0              0</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> CAISGQGSTEAFF       0.000<span style="text-decoration: underline;">120</span>           0               0              0</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> CAISLNSGGAYEQYF     0.000<span style="text-decoration: underline;">120</span>           0               0              0</span>
<span class="co">#&gt; <span style="color: #949494;"># … with 5,605 more rows, and 6 more variables: TRB_Unsorted_1320 &lt;dbl&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   TRB_Unsorted_1496 &lt;dbl&gt;, TRB_Unsorted_32 &lt;dbl&gt;, TRB_Unsorted_369 &lt;dbl&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   TRB_Unsorted_83 &lt;dbl&gt;, TRB_Unsorted_949 &lt;dbl&gt;</span></span></code></pre></div>
<p>If just the top clones with a frequency greater than a specified amount are of interest to you, then use the <code>topFreq</code> function. This creates a tibble of the top productive amino acid sequences having a minimum specified frequency and reports the minimum, maximum, and mean frequency that the sequence appears in a list of samples. For TCRB sequences, the prevalence (%) and the published antigen specificity of that sequence are also provided.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">top_freq</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/topFreq.html">topFreq</a></span><span class="op">(</span>productive_table <span class="op">=</span> <span class="va">aa_table</span>, frequency <span class="op">=</span> <span class="fl">0.001</span><span class="op">)</span>
<span class="va">top_freq</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 610 × 7</span></span>
<span class="co">#&gt;    junction_aa  minFrequency maxFrequency meanFrequency numberSamples prevalence</span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>               <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> CASSQDRTGQY…      0.004<span style="text-decoration: underline;">32</span>       0.018<span style="text-decoration: underline;">6</span>       0.008<span style="text-decoration: underline;">33</span>             9          0</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> CASSLQGREKL…      0.048<span style="text-decoration: underline;">6</span>        0.236        0.087<span style="text-decoration: underline;">3</span>              8          0</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> CASSQDLMTVD…      0.015<span style="text-decoration: underline;">5</span>        0.122        0.073<span style="text-decoration: underline;">9</span>              8          0</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> CASSREGDQPQ…      0.001<span style="text-decoration: underline;">34</span>       0.028<span style="text-decoration: underline;">5</span>       0.005<span style="text-decoration: underline;">77</span>             8          0</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> CASRDGQGSGN…      0.002<span style="text-decoration: underline;">56</span>       0.025<span style="text-decoration: underline;">8</span>       0.012<span style="text-decoration: underline;">5</span>              7          0</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> CASSQDLGQAF…      0.001<span style="text-decoration: underline;">64</span>       0.020<span style="text-decoration: underline;">8</span>       0.009<span style="text-decoration: underline;">37</span>             7          0</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> CASSPFDRGPD…      0.004<span style="text-decoration: underline;">50</span>       0.013<span style="text-decoration: underline;">4</span>       0.009<span style="text-decoration: underline;">23</span>             7          0</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> CASSQDSSDTE…      0.001<span style="text-decoration: underline;">30</span>       0.035<span style="text-decoration: underline;">9</span>       0.007<span style="text-decoration: underline;">65</span>             7          0</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> CAIKMETPNGE…      0.002<span style="text-decoration: underline;">24</span>       0.010<span style="text-decoration: underline;">1</span>       0.005<span style="text-decoration: underline;">77</span>             7          0</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> CASSPGTGTYG…      0.001<span style="text-decoration: underline;">07</span>       0.012<span style="text-decoration: underline;">7</span>       0.004<span style="text-decoration: underline;">73</span>             7          0</span>
<span class="co">#&gt; <span style="color: #949494;"># … with 600 more rows, and 1 more variable: antigen &lt;fct&gt;</span></span></code></pre></div>
<p>One very useful thing to do is merge the output of <code>seqMatrix</code> and <code>topFreq</code>.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">top_freq_matrix</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">full_join</a></span><span class="op">(</span><span class="va">top_freq</span>, <span class="va">sequence_matrix</span><span class="op">)</span>
<span class="co">#&gt; Joining, by = "junction_aa"</span>
<span class="va">top_freq_matrix</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 5,615 × 17</span></span>
<span class="co">#&gt;    junction_aa  minFrequency maxFrequency meanFrequency numberSamples prevalence</span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>               <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> CASSQDRTGQY…      0.004<span style="text-decoration: underline;">32</span>       0.018<span style="text-decoration: underline;">6</span>       0.008<span style="text-decoration: underline;">33</span>             9          0</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> CASSLQGREKL…      0.048<span style="text-decoration: underline;">6</span>        0.236        0.087<span style="text-decoration: underline;">3</span>              8          0</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> CASSQDLMTVD…      0.015<span style="text-decoration: underline;">5</span>        0.122        0.073<span style="text-decoration: underline;">9</span>              8          0</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> CASSREGDQPQ…      0.001<span style="text-decoration: underline;">34</span>       0.028<span style="text-decoration: underline;">5</span>       0.005<span style="text-decoration: underline;">77</span>             8          0</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> CASRDGQGSGN…      0.002<span style="text-decoration: underline;">56</span>       0.025<span style="text-decoration: underline;">8</span>       0.012<span style="text-decoration: underline;">5</span>              7          0</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> CASSQDLGQAF…      0.001<span style="text-decoration: underline;">64</span>       0.020<span style="text-decoration: underline;">8</span>       0.009<span style="text-decoration: underline;">37</span>             7          0</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> CASSPFDRGPD…      0.004<span style="text-decoration: underline;">50</span>       0.013<span style="text-decoration: underline;">4</span>       0.009<span style="text-decoration: underline;">23</span>             7          0</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> CASSQDSSDTE…      0.001<span style="text-decoration: underline;">30</span>       0.035<span style="text-decoration: underline;">9</span>       0.007<span style="text-decoration: underline;">65</span>             7          0</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> CAIKMETPNGE…      0.002<span style="text-decoration: underline;">24</span>       0.010<span style="text-decoration: underline;">1</span>       0.005<span style="text-decoration: underline;">77</span>             7          0</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> CASSPGTGTYG…      0.001<span style="text-decoration: underline;">07</span>       0.012<span style="text-decoration: underline;">7</span>       0.004<span style="text-decoration: underline;">73</span>             7          0</span>
<span class="co">#&gt; <span style="color: #949494;"># … with 5,605 more rows, and 11 more variables: antigen &lt;fct&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   TRB_CD4_949 &lt;dbl&gt;, TRB_CD8_949 &lt;dbl&gt;, TRB_CD8_CMV_369 &lt;dbl&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   TRB_Unsorted_0 &lt;dbl&gt;, TRB_Unsorted_1320 &lt;dbl&gt;, TRB_Unsorted_1496 &lt;dbl&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   TRB_Unsorted_32 &lt;dbl&gt;, TRB_Unsorted_369 &lt;dbl&gt;, TRB_Unsorted_83 &lt;dbl&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   TRB_Unsorted_949 &lt;dbl&gt;</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="tracking-sequences-across-samples">Tracking sequences across samples<a class="anchor" aria-label="anchor" href="#tracking-sequences-across-samples"></a>
</h2>
<p>To visually track the frequency of sequences across multiple samples, use the function <code>cloneTrack</code> This function takes the output from the <code>seqMatrix</code> function. You can specify a character vector of amino acid sequences using the parameter track to highlight those sequences with a different color. Alternatively, you can highlight all of the sequences from a given sample using the parameter map. If the mapping feature is use, then you must specify a productive amino acid list and a character vector of labels to title the mapped samples.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ctable</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/cloneTrack.html">cloneTrack</a></span><span class="op">(</span>study_table <span class="op">=</span> <span class="va">aa_table</span>,
                                 sample_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TRB_CD8_949"</span>, <span class="st">"TRB_CD8_CMV_369"</span><span class="op">)</span><span class="op">)</span>
<span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/plotTrack.html">plotTrack</a></span><span class="op">(</span><span class="va">ctable</span><span class="op">)</span></code></pre></div>
<p><img src="LymphoSeq2_files/figure-html/cloneTrack-1.png" width="700"></p>
<p>You can track particular sequences across samples by providing an optional list of CDR3 amino acid sequences.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ttable</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/topSeqs.html">topSeqs</a></span><span class="op">(</span><span class="va">aa_table</span>, top <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>
<span class="va">ctable</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/cloneTrack.html">cloneTrack</a></span><span class="op">(</span><span class="va">ttable</span><span class="op">)</span>
<span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/plotTrack.html">plotTrack</a></span><span class="op">(</span><span class="va">ctable</span>, alist <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CASSESAGSTGELFF"</span>, <span class="st">"CASSLAGDSQETQYF"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"bottom"</span><span class="op">)</span></code></pre></div>
<p><img src="LymphoSeq2_files/figure-html/plotCloneSet-1.png" width="700"></p>
<p>Alternatively you can use the function <code>plotTrackSingular</code> to retrieve a list of alluvial diagrams each tracking one single amino acid from the clone track table. Considering that a plot is generated for each unique CDR3 sequence, we recommend running this feature on a clone track table derived from only the top sequences from each repertoire as described in the example above.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">lalluvial</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/plotTrackSingular.html">plotTrackSingular</a></span><span class="op">(</span><span class="va">ctable</span><span class="op">)</span>
<span class="va">lalluvial</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<p><img src="LymphoSeq2_files/figure-html/plotSingular-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="comparing-vdj-gene-usage">Comparing V(D)J gene usage<a class="anchor" aria-label="anchor" href="#comparing-vdj-gene-usage"></a>
</h2>
<p>To compare the V, D, and J gene usage across samples, start by creating a data frame of V, D, and J gene counts and frequencies using the function <code>geneFreq</code>. You can specify if you are interested in the “VDJ”, “DJ”, “VJ”, “DJ”, “V”, “D”, or “J” loci using the locus parameter. Set family to TRUE if you prefer the family names instead of the gene names as reported by ImmunoSeq.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">vGenes</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/geneFreq.html">geneFreq</a></span><span class="op">(</span>productive_nt <span class="op">=</span> <span class="va">nuc_table</span>, locus <span class="op">=</span> <span class="st">"V"</span>, family <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">vGenes</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 235 × 5</span></span>
<span class="co">#&gt;    repertoire_id gene_name duplicate_count gene_type gene_frequency</span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>               <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> TRB_CD4_949   <span style="color: #BB0000;">NA</span>                   <span style="text-decoration: underline;">3</span>675 v_family        0.220   </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> TRB_CD4_949   TCRBV01                 5 v_family        0.000<span style="text-decoration: underline;">300</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> TRB_CD4_949   TCRBV02               309 v_family        0.018<span style="text-decoration: underline;">5</span>  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> TRB_CD4_949   TCRBV04               451 v_family        0.027<span style="text-decoration: underline;">0</span>  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> TRB_CD4_949   TCRBV05              <span style="text-decoration: underline;">1</span>472 v_family        0.088<span style="text-decoration: underline;">3</span>  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> TRB_CD4_949   TCRBV06               476 v_family        0.028<span style="text-decoration: underline;">5</span>  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> TRB_CD4_949   TCRBV07              <span style="text-decoration: underline;">1</span>220 v_family        0.073<span style="text-decoration: underline;">1</span>  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> TRB_CD4_949   TCRBV09               145 v_family        0.008<span style="text-decoration: underline;">69</span> </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> TRB_CD4_949   TCRBV10              <span style="text-decoration: underline;">5</span>128 v_family        0.307   </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> TRB_CD4_949   TCRBV11               154 v_family        0.009<span style="text-decoration: underline;">23</span> </span>
<span class="co">#&gt; <span style="color: #949494;"># … with 225 more rows</span></span></code></pre></div>
<p>To create a chord diagram showing VJ or DJ gene associations from one or more more samples, combine the output of <code>geneFreq</code> with the function <code>chordDiagramVDJ</code>. This function works well the <code>topSeqs</code> function that creates a data frame of a selected number of top productive sequences. In the example below, a chord diagram is made showing the association between V and J genes of just the single dominant clones in each sample. The size of the ribbons connecting VJ genes correspond to the number of samples that have that recombination event. The thicker the ribbon, the higher the frequency of the recombination.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">top_seqs</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/topSeqs.html">topSeqs</a></span><span class="op">(</span>productive_table <span class="op">=</span> <span class="va">nuc_table</span>, top <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
<span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/chordDiagramVDJ.html">chordDiagramVDJ</a></span><span class="op">(</span>study_table <span class="op">=</span> <span class="va">top_seqs</span>, 
                            association <span class="op">=</span> <span class="st">"VJ"</span>, 
                            colors <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"darkred"</span>, <span class="st">"navyblue"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="LymphoSeq2_files/figure-html/chordDiagram-1.png" width="700"></p>
<p>You can also visualize the results of <code>geneFreq</code> as a heat map, word cloud, our cumulative frequency bar plot with the support of additional R packages as shown below.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">vGenes</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/geneFreq.html">geneFreq</a></span><span class="op">(</span>productive_nt <span class="op">=</span> <span class="va">nuc_table</span>, locus <span class="op">=</span> <span class="st">"V"</span>, family <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">RColorBrewer</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">grDevices</span><span class="op">)</span>
<span class="va">RedBlue</span> <span class="op">&lt;-</span> <span class="fu">grDevices</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/grDevices/colorRamp.html" class="external-link">colorRampPalette</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rev.html" class="external-link">rev</a></span><span class="op">(</span><span class="fu">RColorBrewer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html" class="external-link">brewer.pal</a></span><span class="op">(</span><span class="fl">11</span>, <span class="st">"RdBu"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">(</span><span class="fl">256</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://blog.fellstat.com/?cat=11" class="external-link">wordcloud</a></span><span class="op">)</span>
<span class="va">vtable</span> <span class="op">&lt;-</span> <span class="va">vGenes</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">repertoire_id</span> <span class="op">==</span> <span class="st">"TRB_Unsorted_83"</span><span class="op">)</span>
<span class="fu">wordcloud</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/wordcloud/man/wordcloud.html" class="external-link">wordcloud</a></span><span class="op">(</span>words <span class="op">=</span> <span class="va">vtable</span><span class="op">$</span><span class="va">gene_name</span>, 
                     freq <span class="op">=</span> <span class="va">vtable</span><span class="op">$</span><span class="va">gene_frequency</span>, 
                     colors <span class="op">=</span> <span class="va">RedBlue</span><span class="op">)</span></code></pre></div>
<p><img src="LymphoSeq2_files/figure-html/wordcloud-1.png" width="700"></p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">vGenes</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/geneFreq.html">geneFreq</a></span><span class="op">(</span><span class="va">nuc_table</span>, locus <span class="op">=</span> <span class="st">"V"</span>, family <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
           <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_wider.html" class="external-link">pivot_wider</a></span><span class="op">(</span>id_cols <span class="op">=</span> <span class="va">gene_name</span>, 
                              names_from <span class="op">=</span> <span class="va">repertoire_id</span>, 
                              values_from <span class="op">=</span> <span class="va">gene_frequency</span>, 
                              values_fn <span class="op">=</span> <span class="va">sum</span>,
                              values_fill <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>
<span class="va">gene_names</span> <span class="op">&lt;-</span> <span class="va">vGenes</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
              <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">gene_name</span><span class="op">)</span>
<span class="va">vGenes</span> <span class="op">&lt;-</span> <span class="va">vGenes</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
          <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">-</span><span class="va">gene_name</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
          <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/as.matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">vGenes</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">gene_names</span>
<span class="fu">pheatmap</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html" class="external-link">pheatmap</a></span><span class="op">(</span><span class="va">vGenes</span>, scale <span class="op">=</span> <span class="st">"row"</span><span class="op">)</span></code></pre></div>
<p><img src="LymphoSeq2_files/figure-html/heatmap-1.png" width="700"></p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">vGenes</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/geneFreq.html">geneFreq</a></span><span class="op">(</span>productive_nt <span class="op">=</span> <span class="va">nuc_table</span>, locus <span class="op">=</span> <span class="st">"V"</span>, family <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">multicolors</span> <span class="op">&lt;-</span> <span class="fu">grDevices</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/grDevices/colorRamp.html" class="external-link">colorRampPalette</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rev.html" class="external-link">rev</a></span><span class="op">(</span><span class="fu">RColorBrewer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html" class="external-link">brewer.pal</a></span><span class="op">(</span><span class="fl">9</span>, <span class="st">"Set1"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">(</span><span class="fl">28</span><span class="op">)</span>
<span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">vGenes</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">repertoire_id</span>, y <span class="op">=</span> <span class="va">gene_frequency</span>, fill <span class="op">=</span> <span class="va">gene_name</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_y_continuous</a></span><span class="op">(</span>expand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html" class="external-link">guides</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guide_legend.html" class="external-link">guide_legend</a></span><span class="op">(</span>ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">multicolors</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>y <span class="op">=</span> <span class="st">"Frequency (%)"</span>, x <span class="op">=</span> <span class="st">""</span>, fill <span class="op">=</span> <span class="st">""</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, vjust <span class="op">=</span> <span class="fl">0.5</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="LymphoSeq2_files/figure-html/geneFreqPlot-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="removing-sequences">Removing sequences<a class="anchor" aria-label="anchor" href="#removing-sequences"></a>
</h2>
<p>Occasionally you may identify one or more sequences in your data set that appear to be contamination. You can remove an amino acid sequence from all data frames using the function removeSeq and recompute <code>frequencyCount</code> for all remaining sequences.</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/searchSeq.html">searchSeq</a></span><span class="op">(</span>study_table <span class="op">=</span> <span class="va">aa_table</span>, sequence <span class="op">=</span> <span class="st">"CASSESAGSTGELFF"</span>, seq_type <span class="op">=</span> <span class="st">"junction_aa"</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 4 × 13</span></span>
<span class="co">#&gt;   repertoire_id     junction_aa  v_call d_call j_call v_family d_family j_family</span>
<span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">1</span> TRB_CD4_949       CASSESAGSTG… TCRBV… TCRBD… TCRBJ… TCRBV10  TCRBD02  TCRBJ02 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">2</span> TRB_Unsorted_1320 CASSESAGSTG… TCRBV… TCRBD… TCRBJ… TCRBV10  TCRBD02  TCRBJ02 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">3</span> TRB_Unsorted_1496 CASSESAGSTG… TCRBV… TCRBD… TCRBJ… TCRBV10  TCRBD02  TCRBJ02 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">4</span> TRB_Unsorted_949  CASSESAGSTG… TCRBV… TCRBD… TCRBJ… TCRBV10  TCRBD02  TCRBJ02 </span>
<span class="co">#&gt; <span style="color: #949494;"># … with 5 more variables: reading_frame &lt;chr&gt;, duplicate_count &lt;dbl&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   duplicate_frequency &lt;dbl&gt;, edit_distance &lt;dbl&gt;, searchSequence &lt;chr&gt;</span></span></code></pre></div>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cleansed</span> <span class="op">&lt;-</span> <span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/removeSeq.html">removeSeq</a></span><span class="op">(</span>study_table <span class="op">=</span> <span class="va">aa_table</span>, sequence <span class="op">=</span> <span class="st">"CASSESAGSTGELFF"</span><span class="op">)</span>
<span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/searchSeq.html">searchSeq</a></span><span class="op">(</span>study_table <span class="op">=</span> <span class="va">cleansed</span>, sequence <span class="op">=</span> <span class="st">"CASSESAGSTGELFF"</span>, seq_type <span class="op">=</span> <span class="st">"junction_aa"</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 0 × 13</span></span>
<span class="co">#&gt; <span style="color: #949494;"># … with 13 variables: repertoire_id &lt;chr&gt;, junction_aa &lt;chr&gt;, v_call &lt;chr&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   d_call &lt;chr&gt;, j_call &lt;chr&gt;, v_family &lt;chr&gt;, d_family &lt;chr&gt;, j_family &lt;chr&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   reading_frame &lt;chr&gt;, duplicate_count &lt;dbl&gt;, duplicate_frequency &lt;dbl&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   edit_distance &lt;dbl&gt;, searchSequence &lt;chr&gt;</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="rarefaction-curves">Rarefaction curves<a class="anchor" aria-label="anchor" href="#rarefaction-curves"></a>
</h2>
<p>Rarefaction and extrapolation curves allow for comparison of TCR diversity across repertoires given a ideal sequencing depth. Rarefaction and extrapolation curves are drawn by sampling a sequencing dataset to various depths to understand the trajectory of sequence diversity and then extrapolating the curve to an ideal depth.</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">LymphoSeq2</span><span class="fu">::</span><span class="fu"><a href="../reference/plotRarefactionCurve.html">plotRarefactionCurve</a></span><span class="op">(</span>study_table <span class="op">=</span> <span class="va">aa_table</span><span class="op">)</span></code></pre></div>
<p><img src="LymphoSeq2_files/figure-html/rarefation-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.1.2 (2021-11-01)</span>
<span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">#&gt; Running under: Ubuntu 20.04.3 LTS</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0</span>
<span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span>
<span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span>
<span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span>
<span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt;  [1] wordcloud_2.6      RColorBrewer_1.1-2 forcats_0.5.1      stringr_1.4.0     </span>
<span class="co">#&gt;  [5] dplyr_1.0.8        purrr_0.3.4        readr_2.1.2        tidyr_1.2.0       </span>
<span class="co">#&gt;  [9] tibble_3.1.6       ggplot2_3.3.5      tidyverse_1.3.1    LymphoSeq2_1.0.0  </span>
<span class="co">#&gt; [13] data.table_1.14.2 </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;   [1] readxl_1.3.1           backports_1.4.1        circlize_0.4.14       </span>
<span class="co">#&gt;   [4] fastmatch_1.1-3        systemfonts_1.0.4      plyr_1.8.6            </span>
<span class="co">#&gt;   [7] igraph_1.2.11          lazyeval_0.2.2         splines_4.1.2         </span>
<span class="co">#&gt;  [10] GenomeInfoDb_1.30.1    digest_0.6.29          foreach_1.5.2         </span>
<span class="co">#&gt;  [13] yulab.utils_0.0.4      htmltools_0.5.2        ggalluvial_0.12.3     </span>
<span class="co">#&gt;  [16] fansi_1.0.2            magrittr_2.0.2         memoise_2.0.1         </span>
<span class="co">#&gt;  [19] ineq_0.2-13            cluster_2.1.2          tzdb_0.2.0            </span>
<span class="co">#&gt;  [22] Biostrings_2.62.0      modelr_0.1.8           dtplyr_1.2.1          </span>
<span class="co">#&gt;  [25] vroom_1.5.7            pkgdown_2.0.2          prettyunits_1.1.1     </span>
<span class="co">#&gt;  [28] colorspace_2.0-3       rvest_1.0.2            textshaping_0.3.6     </span>
<span class="co">#&gt;  [31] haven_2.4.3            xfun_0.29              crayon_1.5.0          </span>
<span class="co">#&gt;  [34] RCurl_1.98-1.6         jsonlite_1.8.0         lme4_1.1-28           </span>
<span class="co">#&gt;  [37] phangorn_2.8.1         survival_3.2-13        iterators_1.0.14      </span>
<span class="co">#&gt;  [40] ape_5.6-1              glue_1.6.1             gtable_0.3.0          </span>
<span class="co">#&gt;  [43] zlibbioc_1.40.0        XVector_0.34.0         UpSetR_1.4.0          </span>
<span class="co">#&gt;  [46] phyloseq_1.38.0        Rhdf5lib_1.16.0        shape_1.4.6           </span>
<span class="co">#&gt;  [49] BiocGenerics_0.40.0    scales_1.1.1           pheatmap_1.0.12       </span>
<span class="co">#&gt;  [52] futile.options_1.0.1   DBI_1.1.2              Rcpp_1.0.8            </span>
<span class="co">#&gt;  [55] progress_1.2.2         gridGraphics_0.5-1     tidytree_0.3.8        </span>
<span class="co">#&gt;  [58] bit_4.0.4              stats4_4.1.2           httr_1.4.2            </span>
<span class="co">#&gt;  [61] ellipsis_0.3.2         farver_2.1.0           pkgconfig_2.0.3       </span>
<span class="co">#&gt;  [64] sass_0.4.0             dbplyr_2.1.1           utf8_1.2.2            </span>
<span class="co">#&gt;  [67] labeling_0.4.2         ggplotify_0.1.0        tidyselect_1.1.2      </span>
<span class="co">#&gt;  [70] rlang_1.0.1            reshape2_1.4.4         munsell_0.5.0         </span>
<span class="co">#&gt;  [73] breakaway_4.7.6        cellranger_1.1.0       tools_4.1.2           </span>
<span class="co">#&gt;  [76] cachem_1.0.6           cli_3.2.0              generics_0.1.2        </span>
<span class="co">#&gt;  [79] ade4_1.7-18            broom_0.7.12           evaluate_0.15         </span>
<span class="co">#&gt;  [82] biomformat_1.22.0      fastmap_1.1.0          yaml_2.3.5            </span>
<span class="co">#&gt;  [85] ragg_1.2.2             ggtree_3.2.1           knitr_1.37            </span>
<span class="co">#&gt;  [88] bit64_4.0.5            fs_1.5.2               nlme_3.1-155          </span>
<span class="co">#&gt;  [91] formatR_1.11           aplot_0.1.2            tictoc_1.0.1          </span>
<span class="co">#&gt;  [94] xml2_1.3.3             compiler_4.1.2         rstudioapi_0.13       </span>
<span class="co">#&gt;  [97] msa_1.26.0             reprex_2.0.1           treeio_1.18.1         </span>
<span class="co">#&gt; [100] bslib_0.3.1            stringi_1.7.6          futile.logger_1.4.3   </span>
<span class="co">#&gt; [103] highr_0.9              desc_1.4.0             lattice_0.20-45       </span>
<span class="co">#&gt; [106] Matrix_1.4-0           nloptr_2.0.0           vegan_2.5-7           </span>
<span class="co">#&gt; [109] permute_0.9-7          multtest_2.50.0        vctrs_0.3.8           </span>
<span class="co">#&gt; [112] stringdist_0.9.8       pillar_1.7.0           lifecycle_1.0.1       </span>
<span class="co">#&gt; [115] rhdf5filters_1.6.0     jquerylib_0.1.4        GlobalOptions_0.1.2   </span>
<span class="co">#&gt; [118] bitops_1.0-7           patchwork_1.1.1        R6_2.5.1              </span>
<span class="co">#&gt; [121] gridExtra_2.3          IRanges_2.28.0         codetools_0.2-18      </span>
<span class="co">#&gt; [124] lambda.r_1.2.4         boot_1.3-28            MASS_7.3-55           </span>
<span class="co">#&gt; [127] assertthat_0.2.1       rhdf5_2.38.0           rprojroot_2.0.2       </span>
<span class="co">#&gt; [130] withr_2.4.3            iNEXT_2.0.20           S4Vectors_0.32.3      </span>
<span class="co">#&gt; [133] GenomeInfoDbData_1.2.7 mgcv_1.8-38            parallel_4.1.2        </span>
<span class="co">#&gt; [136] hms_1.1.1              VennDiagram_1.7.1      quadprog_1.5-8        </span>
<span class="co">#&gt; [139] grid_4.1.2             ggfun_0.0.5            minqa_1.2.4           </span>
<span class="co">#&gt; [142] rmarkdown_2.11         Biobase_2.54.0         lubridate_1.8.0</span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by First Last.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
