#' Create phylogenetic tree
#' 
#' Create a phylogenetic tree using neighbor joining tree estimation for amino acid or 
#' nucleotide CDR3 sequences in a list of data frames.
#' 
#' @param study_table A tibble of unproductive nucleotide sequences or productive 
#' nucleotide sequences generated by the LymphoSeq function productiveSeq.  
#' vFamilyName, dFamilyName, jFamilyName, and count are required columns.
#' @param sample A character vector indicating the name of the sample in the study table. 
#' @param type A character vector indicating whether "aminoAcid" or "nucleotide" sequences
#' should be compared.
#' @param layout A character vector indicating the tree layout.  Options include 
#' "rectangular", "slanted", "fan", "circular", "radial" and "unrooted".
#' @param label A Boolean indicating if the sequencing count should be shown next to the leaves.
#' @return Returns a phylogenetic tree where each leaf represents a sequence color coded by the
#' V, D, and J gene usage.  The number next to each leaf refers to the sequence count.  A triangle 
#' shaped leaf indicates the dominant sequence.  Refer to the ggtree Bioconductor package 
#' documentation for details on how to manipulate the tree.
#' @examples
#' file.path <- system.file("extdata", "IGH_sequencing", package = "LymphoSeq")
#' 
#' study_table <- readImmunoSeq(path = file.path)
#' 
#' productive_nt <- productiveSeq(study_table = study_table, aggregate = "nucleotide")
#' 
#' phyloTree(study_table = productive_nt, sample = "IGH_MVQ92552A_BL", type = "nucleotide", 
#'          layout = "rectangular")
#' 
#' phyloTree(study_table = productive_nt, sample = "IGH_MVQ92552A_BL", type = "aminoAcid", 
#'          layout = "circular")
#'          
#' # Add scale and title to figure
#' library(ggtree)
#' library(ggplot2)
#' phyloTree(study_table = productive_nt, sample = "IGH_MVQ92552A_BL", type = "aminoAcid", 
#'          layout = "rectangular") +
#'          ggtree::theme_tree2() +
#'          ggplot2::theme(legend.position = "right", legend.key = element_rect(colour = "white")) +
#'          ggplot2::ggtitle("Title")
#'          
#' # Hide legend and leaf labels
#' phyloTree(study_table = productive_nt, sample = "IGH_MVQ92552A_BL", type = "nucleotide", 
#'          layout = "rectangular", label = FALSE) +
#'          ggplot2::theme(legend.position="none")
#'          
#' @export
#' @import ggtree
#' @import tidyverse
#' @import ggplot2
#' @importFrom phangorn NJ
phyloTree <- function(study_table, sample, type = "nucleotide", layout = "rectangular", label = TRUE) {
    sample_table <- study_table %>% filter(sample == sample)
    if(length(grep(pattern = "vFamilyName|dFamilyName|jFamilyName|count", names(study_table))) != 4){
        stop("vFamilyName, dFamilyName, jFamilyName, nucleotide, and count are required columns.  Try aggregating the sequences by 'nucleotide' usng the function productiveSeq.", call. = FALSE)
    }
    if(nrow(study_table) < 3){
        stop("Cannot draw phlogenetic tree with less than 3 sequences.", call. = FALSE)
    }
    sample_table <- sample_table %>% mutate_all(funs(str_replace(., "", "unresolved")))
    if(type == "nucleotide") {
        sample_table <- sample_table %>% filter(nchar(nucleotide) >= 9)
        distance <- adist(sample_table$nucleotide, sample_table$nucleotide)
        colnames(distance) <- rownames(distance) <- sample_table$nucleotide
        names <- sample_table$nucleotide
    }
    if(type == "aminoAcid") {
        sample_table <- sample_table %>% filter(nchar(aminoAcid) >= 9)
        distance <- adist(sample_table$aminoAcid, sample_table$aminoAcid)
        colnames(distance) <- rownames(distance) <- sample_table$aminoAcid
        names <- sample_table$aminoAcid
    }
    tree <- phangorn::NJ(distance)
    geneFamilies <- paste(sample_table$vFamilyName, sample_table$dFamilyName, sample_table$jFamilyName)
    geneFamilies <- gsub(geneFamilies, pattern = "IGH|IGL|IGK|TCRB|TCRA", replacement = "")
    geneFamilies <- gsub(geneFamilies, pattern = "unresolved", replacement = "UNR")
    tree_annotation <- tibble(names = names, count = sample_table$count, 
                                  geneFamilies = geneFamilies, aminoAcid = sample_table$aminoAcid,
                                  frequencyCount = round(sample_table$frequencyCount, digits = 2), 
                                  dominant = c("Yes", rep("No", nrow(sample_table) - 1)))
    getPalette <- grDevices::colorRampPalette(RColorBrewer::brewer.pal(9, "Set1"))
    plot <- ggtree::ggtree(tree, layout = layout) %<+% tree_annotation+ 
        ggtree::geom_tippoint(aes_string(color = "geneFamilies", shape = "dominant"), size = 3) + 
        ggplot2::scale_color_manual(values = getPalette(length(unique(geneFamilies)))) + 
        ggplot2::guides(shape = FALSE) + 
        ggplot2::theme(legend.position = "bottom", legend.key = element_rect(colour = "white")) + 
        ggplot2::labs(color = "")
    if(label == TRUE){
        plot <- plot + ggplot2::geom_text(aes_string(label="count"), nudge_x = 0.5, hjust = 0, size = 2.5, na.rm = TRUE, check_overlap = TRUE)
    }
    return(plot)
}