#' Search for a sequence
#' 
#' Search for one or more amino acid or nucleotide CDR3 sequences in a study tibble.
#' 
#' @param study_table A tibble generated by the LymphoSeq2 functions readImmunoSeq 
#' or productiveSeq.  "aminoAcid" or "nucleotide", "frequencyCount", and 
#' "count" are required columns.
#' @param sequence A character vector of one ore more amino acid or nucleotide 
#' CDR3 sequences to search.
#' @param type A character vector specifying the type of sequence(s) to be 
#' searched.  Available options are "aminoAcid" or "nucleotide". 
#' @param match A character vector specifying whether an exact partial or exact global
#'  match of the searched sequence(s) is desired.  Available options are 
#' "partial" and "global".
#' @param editDistance An integer giving the minimum edit distance that the 
#' sequence must be less than or equal to.  See details below.
#' @details An exact partial match means the searched sequence is contained within 
#' target sequence.  An exact global match means the searched sequence is identical to 
#' the target sequence.
#' 
#' Edit distance is a way of quantifying how dissimilar two sequences 
#' are to one another by counting the minimum number of operations required to 
#' transform one sequence into the other.  For example, an edit distance of 0 
#' means the sequences are identical and an edit distance of 1 indicates that 
#' the sequences different by a single amino acid or nucleotide.
#' @return Returns the rows for every instance in the list of data frames where 
#' the searched sequence(s) appeared.
#' @examples
#' file.path <- system.file("extdata", "TCRB_sequencing", package = "LymphoSeq")
#' 
#' study_table <- readImmunoSeq(path = file.path)
#' 
#' aa1 <- "CASSPVSNEQFF"
#' 
#' aa2 <- "CASSQEVPPYQAFF"
#' 
#' searchSeq(study_table = study_table, sequence = aa1, type = "aminoAcid", 
#'    match = "global", editDistance = 0)
#' 
#' searchSeq(study_table = study_table, sequence = c(aa1, aa2), 
#'    type = "aminoAcid", match = "global", editDistance = 0)
#' 
#' searchSeq(study_table = study_table, sequence = aa1, type = "aminoAcid", editDistance = 1)
#' 
#' nt <- "CTGATTCTGGAGTCCGCCAGCACCAACCAGACATCTATGTACCTCTGTGCCAGCAGTCCGGTAAGCAATGAGCAGTTCTTCGGGCCA"
#' 
#' searchSeq(study_table = study_table, sequence = nt, type = "nucleotide", editDistance = 3)
#' 
#' searchSeq(study_table = study_table, sequence = "CASSPVS", type = "aminoAcid", 
#'    match = "partial", editDistance = 0)
#' 
#' searchSeq(study_table = study_table, sequence = nt, type = "nucleotide", editDistance = 0)
#' @export
#' @importFrom plyr llply
#' @importFrom utils adist
#' @import tidyverse
searchSeq <- function(study_table, sequence, seq_type = "nucleotide", match = "global", editDistance = 0) {
    if (match == "global") {
        partial=FALSE
    } else if (match == "partial") {
        partial=TRUE
    }
    query_list <- study_table %>% select(all_of(seq_type)) %>% filter(!is.na(seq_type)) %>%  as_vector()
    query_list <- query_list[!is.na(query_list)]
    search_tables <- sequence %>% map(~findSeq(.x, query_list, partial, editDistance, seq_type)) %>% reduce(rbind)
    return(search_tables)
}

findSeq <- function(sequence, query_list, mode, editDistance, seq_type){
    edist <- adist(sequence, query_list, partial = mode)
    match_list <- query_list[(edist <= editDistance)]
    edist_list <- edist[(edist <= editDistance)]
    sequence_table <- tibble(c1 = match_list, c2 = edist_list, c3 = sequence, .name_repair= ~ c(seq_type, "editDistance", "searchSequence"))
    search_table <- left_join(sequence_table, study_table, by=seq_type)
    return(search_table)
}
